<html>

<head>
<style type="text/css">
.inline {
  background-color: #f7f7f7;
  border:solid 1px #B0B0B0;
}
.error {
	font-weight: bold;
	color: #FF0000;
}
.warning {
	font-weight: bold;
}
.message {
	font-style: italic;
}
.source, .output, .warning, .error, .message {
	padding: 0 1em;
  border:solid 1px #F7F7F7;
}
.source {
  background-color: #f5f5f5;
}
.left {
  text-align: left;
}
.right {
  text-align: right;
}
.center {
  text-align: center;
}
.hl.num {
  color: #AF0F91;
}
.hl.sng {
  color: #317ECC;
}
.hl.com {
  color: #AD95AF;
  font-style: italic;
}
.hl.opt {
  color: #000000;
}
.hl.def {
  color: #585858;
}
.hl.kwa {
  color: #295F94;
  font-weight: bold;
}
.hl.kwb {
  color: #B05A65;
}
.hl.kwc {
  color: #55aa55;
}
.hl.kwd {
  color: #BC5A65;
  font-weight: bold;
}
</style>
<title>Avijit Podder Date: Aug 12, 2024</title>

<head color="blue">

<h1><font color="blue">Individual Pathway Specific Z score Calculation using TMT Proteomic Data from the ROSMAP Cohort (comparative Pathway Profiling - AD-male vs AD-female)</font></h1>

</head>

<body>

<p>Percentage of individuals in each group with associated pathways (KEGG)</p>

<div class="chunk" id="unnamed-chunk-1"><div class="rcode"><div class="source"><pre class="knitr r"><span class="hl kwd">library</span><span class="hl def">(gplots)</span>
</pre></div>
<div class="message"><pre class="knitr r">## 
## Attaching package: 'gplots'
</pre></div>
<div class="message"><pre class="knitr r">## The following object is masked from 'package:stats':
## 
##     lowess
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl kwd">library</span><span class="hl def">(</span><span class="hl sng">&quot;ggplot2&quot;</span><span class="hl def">)</span>
</pre></div>
<div class="warning"><pre class="knitr r">## Warning: package 'ggplot2' was built under R version 4.5.2
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl kwd">library</span><span class="hl def">(reshape2)</span>
<span class="hl kwd">library</span><span class="hl def">(RColorBrewer)</span>
<span class="hl kwd">library</span><span class="hl def">(dplyr)</span>
</pre></div>
<div class="message"><pre class="knitr r">## 
## Attaching package: 'dplyr'
</pre></div>
<div class="message"><pre class="knitr r">## The following objects are masked from 'package:stats':
## 
##     filter, lag
</pre></div>
<div class="message"><pre class="knitr r">## The following objects are masked from 'package:base':
## 
##     intersect, setdiff, setequal, union
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl kwd">library</span><span class="hl def">(viridis)</span>
</pre></div>
<div class="message"><pre class="knitr r">## Loading required package: viridisLite
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl kwd">library</span><span class="hl def">(ggrepel)</span>
<span class="hl kwd">library</span><span class="hl def">(corrplot)</span>
</pre></div>
<div class="message"><pre class="knitr r">## corrplot 0.95 loaded
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl kwd">library</span><span class="hl def">(plotly)</span>
</pre></div>
<div class="message"><pre class="knitr r">## 
## Attaching package: 'plotly'
</pre></div>
<div class="message"><pre class="knitr r">## The following object is masked from 'package:ggplot2':
## 
##     last_plot
</pre></div>
<div class="message"><pre class="knitr r">## The following object is masked from 'package:stats':
## 
##     filter
</pre></div>
<div class="message"><pre class="knitr r">## The following object is masked from 'package:graphics':
## 
##     layout
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl com"># Open pathways comparison file</span>
<span class="hl com">#data_KEGG &lt;- read.csv(&quot;comparative_pathway_results_proteomics_selected_KEGG.csv&quot;, sep =&quot;,&quot;, header = TRUE, stringsAsFactors = FALSE)</span>

<span class="hl com">#data_KEGG &lt;- read.csv(&quot;comparative_pathway_results.csv&quot;, sep =&quot;,&quot;, header = TRUE, stringsAsFactors = FALSE)</span>

<span class="hl def">data_KEGG</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">read.csv</span><span class="hl def">(</span><span class="hl sng">&quot;comparative_pathway_results_proteomics_selected_KEGG_050525.csv&quot;</span><span class="hl def">,</span> <span class="hl kwc">sep</span> <span class="hl def">=</span><span class="hl sng">&quot;,&quot;</span><span class="hl def">,</span> <span class="hl kwc">header</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span> <span class="hl kwc">stringsAsFactors</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">)</span>

<span class="hl kwd">dim</span><span class="hl def">(data_KEGG)</span>
</pre></div>
<div class="output"><pre class="knitr r">## [1] 186  11
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl kwd">head</span><span class="hl def">(data_KEGG)</span>
</pre></div>
<div class="output"><pre class="knitr r">##   Sno Pathway_ID                              Pathway_name UP_ratio_AD_female
## 1   1   hsa00010             Glycolysis / Gluconeogenesis           0.0937500
## 2   3   hsa00030                Pentose phosphate pathway           0.0390625
## 3   4   hsa00040 Pentose and glucuronate interconversions           0.0625000
## 4   5   hsa00051          Fructose and mannose metabolism           0.0859375
## 5   6   hsa00052                     Galactose metabolism           0.1093750
## 6   7   hsa00053        Ascorbate and aldarate metabolism           0.0781250
##   UP_ratio_AD_male DOWN_ratio_AD_female DOWN_ratio_AD_male
## 1       0.04255319            0.0390625         0.04255319
## 2       0.10638298            0.0000000         0.04255319
## 3       0.10638298            0.0234375         0.02127660
## 4       0.08510638            0.0234375         0.04255319
## 5       0.10638298            0.0234375         0.02127660
## 6       0.04255319            0.0156250         0.04255319
##   UP_ratio_AsymAD_female UP_ratio_AsymAD_male DOWN_ratio_AsymAD_female
## 1             0.06289308           0.06557377               0.02515723
## 2             0.01886793           0.06557377               0.01886793
## 3             0.02515723           0.04918033               0.01886793
## 4             0.03144654           0.06557377               0.01886793
## 5             0.05031447           0.04918033               0.02515723
## 6             0.01257862           0.01639344               0.02515723
##   DOWN_ratio_AsymAD_male
## 1             0.06557377
## 2             0.04918033
## 3             0.03278689
## 4             0.03278689
## 5             0.00000000
## 6             0.03278689
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl com"># Distribution of Gene_ratio over the pathways</span>
<span class="hl com">#ggplot(data_KEGG, aes(x = Gene_ratio)) +  </span>
  <span class="hl com">#geom_histogram(binwidth = 0.05, fill = &quot;blue&quot;, color = &quot;black&quot;, alpha = 0.7) + </span>
  <span class="hl com">#geom_text(stat='bin', binwidth=0.05, aes(label=..count..), vjust=-0.5) + </span>
  <span class="hl com">#labs(title = &quot;Histogram of Gene Ratio&quot;, x = &quot;Gene Ratio&quot;, y = &quot;Frequency&quot;) + </span>
  <span class="hl com">#theme_minimal() +  </span>
  <span class="hl com">#theme(text = element_text(size = 16))</span>




<span class="hl com"># Removing the 1st and 2nd columns</span>
<span class="hl com">#data_KEGG_new &lt;- data_KEGG[, -c(1, 2, 4, 5, 6)]</span>

<span class="hl def">data_KEGG_new</span> <span class="hl kwb">&lt;-</span> <span class="hl def">data_KEGG[,</span> <span class="hl opt">-</span><span class="hl kwd">c</span><span class="hl def">(</span><span class="hl num">1</span><span class="hl def">,</span> <span class="hl num">2</span><span class="hl def">)]</span>

<span class="hl kwd">head</span><span class="hl def">(data_KEGG_new)</span>
</pre></div>
<div class="output"><pre class="knitr r">##                                Pathway_name UP_ratio_AD_female UP_ratio_AD_male
## 1             Glycolysis / Gluconeogenesis           0.0937500       0.04255319
## 2                Pentose phosphate pathway           0.0390625       0.10638298
## 3 Pentose and glucuronate interconversions           0.0625000       0.10638298
## 4          Fructose and mannose metabolism           0.0859375       0.08510638
## 5                     Galactose metabolism           0.1093750       0.10638298
## 6        Ascorbate and aldarate metabolism           0.0781250       0.04255319
##   DOWN_ratio_AD_female DOWN_ratio_AD_male UP_ratio_AsymAD_female
## 1            0.0390625         0.04255319             0.06289308
## 2            0.0000000         0.04255319             0.01886793
## 3            0.0234375         0.02127660             0.02515723
## 4            0.0234375         0.04255319             0.03144654
## 5            0.0234375         0.02127660             0.05031447
## 6            0.0156250         0.04255319             0.01257862
##   UP_ratio_AsymAD_male DOWN_ratio_AsymAD_female DOWN_ratio_AsymAD_male
## 1           0.06557377               0.02515723             0.06557377
## 2           0.06557377               0.01886793             0.04918033
## 3           0.04918033               0.01886793             0.03278689
## 4           0.06557377               0.01886793             0.03278689
## 5           0.04918033               0.02515723             0.00000000
## 6           0.01639344               0.02515723             0.03278689
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl def">data_matrix_KEGG</span> <span class="hl kwb">&lt;-</span> <span class="hl def">data_KEGG_new[,</span> <span class="hl opt">-</span><span class="hl num">1</span><span class="hl def">]</span>

<span class="hl com"># Replace NA values with 0 (or any other specific value)</span>
<span class="hl def">data_matrix_KEGG[</span><span class="hl kwd">is.na</span><span class="hl def">(data_matrix_KEGG)]</span> <span class="hl kwb">&lt;-</span> <span class="hl num">0</span>

<span class="hl com"># Set up the color scale using RColorBrewer</span>
<span class="hl def">color_scale</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">colorRampPalette</span><span class="hl def">(</span><span class="hl kwd">brewer.pal</span><span class="hl def">(</span><span class="hl num">9</span><span class="hl def">,</span> <span class="hl sng">&quot;RdYlGn&quot;</span><span class="hl def">))</span>
<span class="hl def">color_scale</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">colorRampPalette</span><span class="hl def">(</span><span class="hl kwd">brewer.pal</span><span class="hl def">(</span><span class="hl num">9</span><span class="hl def">,</span> <span class="hl sng">&quot;RdYlGn&quot;</span><span class="hl def">))</span>
<span class="hl def">color_scale1</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">colorRampPalette</span><span class="hl def">(</span><span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;white&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;lightblue&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;blue&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;darkblue&quot;</span><span class="hl def">))</span>
<span class="hl def">color_scale2</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">colorRampPalette</span><span class="hl def">(</span><span class="hl kwd">brewer.pal</span><span class="hl def">(</span><span class="hl num">9</span><span class="hl def">,</span> <span class="hl sng">&quot;RdGy&quot;</span><span class="hl def">))</span>
<span class="hl def">color_scale3</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">colorRampPalette</span><span class="hl def">(</span><span class="hl kwd">brewer.pal</span><span class="hl def">(</span><span class="hl num">9</span><span class="hl def">,</span> <span class="hl sng">&quot;Greens&quot;</span><span class="hl def">))</span>
<span class="hl def">color_scale4</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">colorRampPalette</span><span class="hl def">(</span><span class="hl kwd">brewer.pal</span><span class="hl def">(</span><span class="hl num">9</span><span class="hl def">,</span> <span class="hl sng">&quot;Blues&quot;</span><span class="hl def">))</span>
<span class="hl def">color_scale5</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">colorRampPalette</span><span class="hl def">(</span><span class="hl kwd">brewer.pal</span><span class="hl def">(</span><span class="hl num">9</span><span class="hl def">,</span> <span class="hl sng">&quot;Purples&quot;</span><span class="hl def">))</span>
<span class="hl def">color_scale6</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">colorRampPalette</span><span class="hl def">(</span><span class="hl kwd">brewer.pal</span><span class="hl def">(</span><span class="hl num">9</span><span class="hl def">,</span> <span class="hl sng">&quot;Oranges&quot;</span><span class="hl def">))</span>

<span class="hl kwd">heatmap.2</span><span class="hl def">(</span>
<span class="hl kwd">as.matrix</span><span class="hl def">(data_matrix_KEGG),</span>
<span class="hl kwc">Rowv</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>
<span class="hl kwc">Colv</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">,</span>
<span class="hl kwc">col</span> <span class="hl def">= color_scale,</span>
<span class="hl kwc">scale</span> <span class="hl def">=</span> <span class="hl sng">&quot;none&quot;</span><span class="hl def">,</span>  <span class="hl com"># Scale rows (genes) to Z-scores</span>
<span class="hl kwc">trace</span> <span class="hl def">=</span> <span class="hl sng">&quot;none&quot;</span><span class="hl def">,</span>  <span class="hl com"># Turn off row and column annotations</span>
<span class="hl kwc">margins</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl num">15</span><span class="hl def">,</span> <span class="hl num">30</span><span class="hl def">),</span>  <span class="hl com"># Set margins for row and column labels</span>
<span class="hl kwc">key</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>  <span class="hl com"># Include a color key</span>
<span class="hl kwc">keysize</span> <span class="hl def">=</span> <span class="hl num">.5</span><span class="hl def">,</span>  <span class="hl com"># Size of the color key</span>
<span class="hl kwc">cexCol</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">,</span>  <span class="hl com"># Set column label size</span>
<span class="hl kwc">cexRow</span> <span class="hl def">=</span> <span class="hl num">1.5</span><span class="hl def">,</span>  <span class="hl com"># Set row label size</span>
<span class="hl kwc">cellnote</span><span class="hl def">=</span><span class="hl kwd">round</span><span class="hl def">(data_matrix_KEGG,</span> <span class="hl num">2</span><span class="hl def">),</span>
<span class="hl kwc">notecol</span><span class="hl def">=</span><span class="hl sng">&quot;black&quot;</span><span class="hl def">,</span>
<span class="hl kwc">labRow</span> <span class="hl def">= data_KEGG</span><span class="hl opt">$</span><span class="hl def">Pathway_name,</span>  <span class="hl com"># Row labels (Pathway name)</span>
<span class="hl kwc">dendrogram</span> <span class="hl def">=</span> <span class="hl sng">&quot;row&quot;</span><span class="hl def">,</span>  <span class="hl com"># Show both row and column dendrograms</span>
<span class="hl kwc">main</span><span class="hl def">=</span><span class="hl sng">&quot;Individuals (ratio) in each group exhibit a significant pathway (KEGG)&quot;</span>
<span class="hl def">)</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-1-1.png" alt="plot of chunk unnamed-chunk-1" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com"># subset the data where the value is equal to or greater than 0.10 in any row </span>
<span class="hl def">subset_data_KEGG</span> <span class="hl kwb">&lt;-</span> <span class="hl def">data_KEGG_new</span> <span class="hl opt">%&gt;%</span> <span class="hl kwd">filter</span><span class="hl def">(UP_ratio_AD_female</span> <span class="hl opt">&gt;=</span> <span class="hl num">0.10</span> <span class="hl opt">|</span> <span class="hl def">DOWN_ratio_AD_female</span> <span class="hl opt">&gt;=</span> <span class="hl num">0.10</span> <span class="hl opt">|</span> <span class="hl def">UP_ratio_AD_male</span> <span class="hl opt">&gt;=</span> <span class="hl num">0.10</span> <span class="hl opt">|</span> <span class="hl def">DOWN_ratio_AD_male</span> <span class="hl opt">&gt;=</span> <span class="hl num">0.10</span> <span class="hl opt">|</span> <span class="hl def">UP_ratio_AsymAD_female</span> <span class="hl opt">&gt;=</span> <span class="hl num">0.10</span> <span class="hl opt">|</span> <span class="hl def">DOWN_ratio_AsymAD_female</span> <span class="hl opt">&gt;=</span> <span class="hl num">0.10</span> <span class="hl opt">|</span> <span class="hl def">UP_ratio_AsymAD_male</span> <span class="hl opt">&gt;=</span> <span class="hl num">0.10</span> <span class="hl opt">|</span> <span class="hl def">DOWN_ratio_AsymAD_male</span> <span class="hl opt">&gt;=</span> <span class="hl num">0.10</span><span class="hl def">)</span>

<span class="hl kwd">write.table</span><span class="hl def">(subset_data_KEGG,</span><span class="hl sng">&quot;subset_data_KEGG_atleast_10percent_ind.txt&quot;</span><span class="hl def">,</span><span class="hl kwc">sep</span><span class="hl def">=</span><span class="hl sng">&quot;\t&quot;</span><span class="hl def">,</span><span class="hl kwc">quote</span><span class="hl def">=F)</span>


<span class="hl kwd">rownames</span><span class="hl def">(subset_data_KEGG)</span> <span class="hl kwb">&lt;-</span> <span class="hl def">subset_data_KEGG</span><span class="hl opt">$</span><span class="hl def">Pathway_name</span>

<span class="hl def">subset_data_KEGG</span> <span class="hl kwb">&lt;-</span> <span class="hl def">subset_data_KEGG[,</span> <span class="hl opt">-</span><span class="hl num">1</span><span class="hl def">]</span>

<span class="hl com"># Replace NA values with 0 (or any other specific value)</span>
<span class="hl def">subset_data_KEGG[</span><span class="hl kwd">is.na</span><span class="hl def">(subset_data_KEGG)]</span> <span class="hl kwb">&lt;-</span> <span class="hl num">0</span>


<span class="hl kwd">head</span><span class="hl def">(subset_data_KEGG,</span> <span class="hl num">3</span><span class="hl def">)</span>
</pre></div>
<div class="output"><pre class="knitr r">##                                           UP_ratio_AD_female UP_ratio_AD_male
## Pentose phosphate pathway                          0.0390625         0.106383
## Pentose and glucuronate interconversions           0.0625000         0.106383
## Galactose metabolism                               0.1093750         0.106383
##                                           DOWN_ratio_AD_female
## Pentose phosphate pathway                            0.0000000
## Pentose and glucuronate interconversions             0.0234375
## Galactose metabolism                                 0.0234375
##                                           DOWN_ratio_AD_male
## Pentose phosphate pathway                         0.04255319
## Pentose and glucuronate interconversions          0.02127660
## Galactose metabolism                              0.02127660
##                                           UP_ratio_AsymAD_female
## Pentose phosphate pathway                             0.01886793
## Pentose and glucuronate interconversions              0.02515723
## Galactose metabolism                                  0.05031447
##                                           UP_ratio_AsymAD_male
## Pentose phosphate pathway                           0.06557377
## Pentose and glucuronate interconversions            0.04918033
## Galactose metabolism                                0.04918033
##                                           DOWN_ratio_AsymAD_female
## Pentose phosphate pathway                               0.01886793
## Pentose and glucuronate interconversions                0.01886793
## Galactose metabolism                                    0.02515723
##                                           DOWN_ratio_AsymAD_male
## Pentose phosphate pathway                             0.04918033
## Pentose and glucuronate interconversions              0.03278689
## Galactose metabolism                                  0.00000000
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl com"># Create the heatmap with filtered row names</span>
<span class="hl kwd">heatmap.2</span><span class="hl def">(</span>
     <span class="hl kwd">as.matrix</span><span class="hl def">(subset_data_KEGG),</span>
     <span class="hl kwc">Rowv</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>
     <span class="hl kwc">Colv</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">,</span>
     <span class="hl kwc">col</span> <span class="hl def">= color_scale,</span>
     <span class="hl kwc">scale</span> <span class="hl def">=</span> <span class="hl sng">&quot;none&quot;</span><span class="hl def">,</span>
     <span class="hl kwc">trace</span> <span class="hl def">=</span> <span class="hl sng">&quot;none&quot;</span><span class="hl def">,</span>
     <span class="hl kwc">margins</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl num">20</span><span class="hl def">,</span> <span class="hl num">40</span><span class="hl def">),</span>
     <span class="hl kwc">key</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>
     <span class="hl kwc">keysize</span> <span class="hl def">=</span> <span class="hl num">.5</span><span class="hl def">,</span>
     <span class="hl kwc">cexCol</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">,</span>
     <span class="hl kwc">cexRow</span> <span class="hl def">=</span> <span class="hl num">1.5</span><span class="hl def">,</span>
    <span class="hl kwc">cellnote</span><span class="hl def">=</span><span class="hl kwd">round</span><span class="hl def">(subset_data_KEGG,</span> <span class="hl num">2</span><span class="hl def">),</span>
    <span class="hl kwc">notecol</span><span class="hl def">=</span><span class="hl sng">&quot;black&quot;</span><span class="hl def">,</span>
    <span class="hl kwc">dendrogram</span> <span class="hl def">=</span> <span class="hl sng">&quot;row&quot;</span><span class="hl def">,</span>
    <span class="hl kwc">main</span> <span class="hl def">=</span> <span class="hl sng">&quot;Pathways Present in atleast 10% Individuals&quot;</span>
<span class="hl def">)</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-1-2.png" alt="plot of chunk unnamed-chunk-1" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">########################################################################################################################################################################################################</span>

<span class="hl com"># Filter data_KEGG_new where 'Pathway_name' contains the phrase 'metabolism'</span>
<span class="hl com">#data_KEGG_new_metabolism &lt;- data_KEGG_new %&gt;%</span>
  <span class="hl com">#filter(grepl(&quot;metabolism&quot;, Pathway_name))</span>

<span class="hl com">#dim(data_KEGG_new_metabolism)</span>

<span class="hl com">#head(data_KEGG_new_metabolism)</span>

<span class="hl com">#write.table(data_KEGG_new_metabolism,&quot;data_KEGG_new_metabolism.txt&quot;,sep=&quot;\t&quot;,quote=F)</span>

<span class="hl com">#data_matrix_KEGG_new_metabolism &lt;- data_KEGG_new_metabolism[, -1]</span>

<span class="hl com"># Replace NA values with 0 (or any other specific value)</span>
<span class="hl com">#data_matrix_KEGG_new_metabolism[is.na(data_matrix_KEGG_new_metabolism)] &lt;- 0</span>


<span class="hl com">#heatmap.2(</span>
     <span class="hl com">#as.matrix(data_matrix_KEGG_new_metabolism),</span>
     <span class="hl com">#Rowv = TRUE,</span>
     <span class="hl com">#Colv = FALSE,</span>
     <span class="hl com">#col = color_scale,</span>
     <span class="hl com">#scale = &quot;none&quot;,</span>
     <span class="hl com">#trace = &quot;none&quot;,</span>
     <span class="hl com">#margins = c(20, 40),</span>
     <span class="hl com">#key = TRUE,</span>
     <span class="hl com">#keysize = .5,</span>
     <span class="hl com">#cexCol = 1.5,</span>
     <span class="hl com">#cexRow = 2,</span>
    <span class="hl com">#cellnote=round(data_matrix_KEGG_new_metabolism, 2),</span>
    <span class="hl com">#notecol=&quot;black&quot;,</span>
    <span class="hl com">#dendrogram = &quot;row&quot;,</span>
    <span class="hl com">#labRow = data_KEGG_new_metabolism$Pathway_name,</span>
    <span class="hl com">#main = &quot;Distributions of Metablic Pathways in Each Group&quot;</span>
<span class="hl com">#)</span>


<span class="hl com"># Filter data_KEGG_new where 'Pathway_name' contains the phrase 'metabolism'</span>
<span class="hl com">#data_KEGG_metabolism &lt;- data_KEGG %&gt;%</span>
  <span class="hl com">#filter(grepl(&quot;metabolism&quot;, Pathway_name))</span>

<span class="hl com"># Create the bar plot for Gene_ratio</span>
<span class="hl com">#ggplot(data_KEGG_metabolism, aes(x = Gene_ratio, y = reorder(Pathway_name, Pathway_name))) +</span>
  <span class="hl com">#geom_bar(stat = &quot;identity&quot;, fill = &quot;steelblue&quot;) +</span>
  <span class="hl com">#labs(title = &quot;Bar Plot of Gene_ratio&quot;,</span>
       <span class="hl com">#x = &quot;Gene ratio&quot;,</span>
       <span class="hl com">#y = &quot;Pathway Name&quot;) +</span>
  <span class="hl com">#theme(axis.text.x = element_text(hjust = 1, size =10),  # Rotate x-axis labels</span>
        <span class="hl com">#text = element_text(size = 20))  # Adjust text size</span>

<span class="hl com">########################################################################################################################################################################################################</span>

<span class="hl com"># Calculate the correlation matrix</span>

<span class="hl kwd">library</span><span class="hl def">(corrplot)</span>

<span class="hl com"># 1. Desired column order (AsymAD first, then AD)</span>
<span class="hl def">column_order</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">c</span><span class="hl def">(</span>
    <span class="hl sng">&quot;UP_ratio_AsymAD_female&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;UP_ratio_AsymAD_male&quot;</span><span class="hl def">,</span>
    <span class="hl sng">&quot;DOWN_ratio_AsymAD_female&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;DOWN_ratio_AsymAD_male&quot;</span><span class="hl def">,</span>
    <span class="hl sng">&quot;UP_ratio_AD_female&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;UP_ratio_AD_male&quot;</span><span class="hl def">,</span>
    <span class="hl sng">&quot;DOWN_ratio_AD_female&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;DOWN_ratio_AD_male&quot;</span>
<span class="hl def">)</span>

<span class="hl com"># 2. Reorder the matrix</span>
<span class="hl def">data_matrix_KEGG</span> <span class="hl kwb">&lt;-</span> <span class="hl def">data_matrix_KEGG[, column_order]</span>

<span class="hl com"># 3. Create custom labels</span>
<span class="hl def">col_labels</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">c</span><span class="hl def">(</span>
    <span class="hl sng">&quot;AsymAD_Female (UP)&quot;</span><span class="hl def">,</span>
    <span class="hl sng">&quot;AsymAD_Male (UP)&quot;</span><span class="hl def">,</span>
    <span class="hl sng">&quot;AsymAD_Female (DOWN)&quot;</span><span class="hl def">,</span>
    <span class="hl sng">&quot;AsymAD_Male (DOWN)&quot;</span><span class="hl def">,</span>
    <span class="hl sng">&quot;AD_Female (UP)&quot;</span><span class="hl def">,</span>
    <span class="hl sng">&quot;AD_Male (UP)&quot;</span><span class="hl def">,</span>
    <span class="hl sng">&quot;AD_Female (DOWN)&quot;</span><span class="hl def">,</span>
    <span class="hl sng">&quot;AD_Male (DOWN)&quot;</span>
<span class="hl def">)</span>

<span class="hl com"># 4. Compute correlation matrix</span>
<span class="hl def">correlation_matrix</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">cor</span><span class="hl def">(data_matrix_KEGG,</span> <span class="hl kwc">use</span> <span class="hl def">=</span> <span class="hl sng">&quot;complete.obs&quot;</span><span class="hl def">)</span>

<span class="hl com"># 5. Rename both row and column names for corrplot</span>
<span class="hl kwd">colnames</span><span class="hl def">(correlation_matrix)</span> <span class="hl kwb">&lt;-</span> <span class="hl def">col_labels</span>
<span class="hl kwd">rownames</span><span class="hl def">(correlation_matrix)</span> <span class="hl kwb">&lt;-</span> <span class="hl def">col_labels</span>

<span class="hl com"># 6. Draw the correlation plot</span>
<span class="hl kwd">corrplot</span><span class="hl def">(</span>
    <span class="hl def">correlation_matrix,</span>
    <span class="hl kwc">method</span> <span class="hl def">=</span> <span class="hl sng">&quot;circle&quot;</span><span class="hl def">,</span>
    <span class="hl kwc">type</span> <span class="hl def">=</span> <span class="hl sng">&quot;upper&quot;</span><span class="hl def">,</span>
    <span class="hl kwc">tl.col</span> <span class="hl def">=</span> <span class="hl sng">&quot;black&quot;</span><span class="hl def">,</span>
    <span class="hl kwc">tl.cex</span> <span class="hl def">=</span> <span class="hl num">1.4</span><span class="hl def">,</span>          <span class="hl com"># Axis label font size</span>
    <span class="hl kwc">number.cex</span> <span class="hl def">=</span> <span class="hl num">1.2</span><span class="hl def">,</span>      <span class="hl com"># Correlation number font size</span>
    <span class="hl kwc">addCoef.col</span> <span class="hl def">=</span> <span class="hl sng">&quot;black&quot;</span><span class="hl def">,</span>
    <span class="hl kwc">diag</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>
    <span class="hl kwc">tl.srt</span> <span class="hl def">=</span> <span class="hl num">45</span><span class="hl def">,</span>           <span class="hl com"># Rotate labels</span>
    <span class="hl kwc">cl.cex</span> <span class="hl def">=</span> <span class="hl num">1.5</span>           <span class="hl com"># ⬅️ Increase color bar font size</span>
<span class="hl def">)</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-1-3.png" alt="plot of chunk unnamed-chunk-1" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">########################################################################################################################################################################################################</span>
<span class="hl com"># UP and DOWN Pathway Separate ploting </span>

<span class="hl com"># Removing the 1st and 2nd columns</span>
<span class="hl com">#data_KEGG_new_UP &lt;- data_KEGG[, -c(1, 2, 4, 5, 6, 8, 10, 12, 14)]</span>

<span class="hl def">data_KEGG_new_UP</span> <span class="hl kwb">&lt;-</span> <span class="hl def">data_KEGG[,</span> <span class="hl opt">-</span><span class="hl kwd">c</span><span class="hl def">(</span><span class="hl num">1</span><span class="hl def">,</span> <span class="hl num">2</span><span class="hl def">,</span> <span class="hl num">6</span><span class="hl def">,</span> <span class="hl num">7</span><span class="hl def">,</span> <span class="hl num">10</span><span class="hl def">,</span> <span class="hl num">11</span><span class="hl def">)]</span>

<span class="hl kwd">head</span><span class="hl def">(data_KEGG_new_UP)</span>
</pre></div>
<div class="output"><pre class="knitr r">##                                Pathway_name UP_ratio_AD_female UP_ratio_AD_male
## 1             Glycolysis / Gluconeogenesis           0.0937500       0.04255319
## 2                Pentose phosphate pathway           0.0390625       0.10638298
## 3 Pentose and glucuronate interconversions           0.0625000       0.10638298
## 4          Fructose and mannose metabolism           0.0859375       0.08510638
## 5                     Galactose metabolism           0.1093750       0.10638298
## 6        Ascorbate and aldarate metabolism           0.0781250       0.04255319
##   UP_ratio_AsymAD_female UP_ratio_AsymAD_male
## 1             0.06289308           0.06557377
## 2             0.01886793           0.06557377
## 3             0.02515723           0.04918033
## 4             0.03144654           0.06557377
## 5             0.05031447           0.04918033
## 6             0.01257862           0.01639344
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl com">#data_KEGG_new_DOWN &lt;- data_KEGG[, -c(1, 2, 4, 5, 6, 7, 9, 11, 13)]</span>

<span class="hl def">data_KEGG_new_DOWN</span> <span class="hl kwb">&lt;-</span> <span class="hl def">data_KEGG[,</span> <span class="hl opt">-</span><span class="hl kwd">c</span><span class="hl def">(</span><span class="hl num">1</span><span class="hl def">,</span> <span class="hl num">2</span><span class="hl def">,</span> <span class="hl num">4</span><span class="hl def">,</span> <span class="hl num">5</span><span class="hl def">,</span> <span class="hl num">8</span><span class="hl def">,</span> <span class="hl num">9</span><span class="hl def">)]</span>

<span class="hl kwd">head</span><span class="hl def">(data_KEGG_new_DOWN)</span>
</pre></div>
<div class="output"><pre class="knitr r">##                                Pathway_name DOWN_ratio_AD_female
## 1             Glycolysis / Gluconeogenesis             0.0390625
## 2                Pentose phosphate pathway             0.0000000
## 3 Pentose and glucuronate interconversions             0.0234375
## 4          Fructose and mannose metabolism             0.0234375
## 5                     Galactose metabolism             0.0234375
## 6        Ascorbate and aldarate metabolism             0.0156250
##   DOWN_ratio_AD_male DOWN_ratio_AsymAD_female DOWN_ratio_AsymAD_male
## 1         0.04255319               0.02515723             0.06557377
## 2         0.04255319               0.01886793             0.04918033
## 3         0.02127660               0.01886793             0.03278689
## 4         0.04255319               0.01886793             0.03278689
## 5         0.02127660               0.02515723             0.00000000
## 6         0.04255319               0.02515723             0.03278689
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl def">data_matrix_KEGG_UP</span> <span class="hl kwb">&lt;-</span> <span class="hl def">data_KEGG_new_UP[,</span> <span class="hl opt">-</span><span class="hl num">1</span><span class="hl def">]</span>
<span class="hl def">data_matrix_KEGG_DOWN</span> <span class="hl kwb">&lt;-</span> <span class="hl def">data_KEGG_new_DOWN[,</span> <span class="hl opt">-</span><span class="hl num">1</span><span class="hl def">]</span>

<span class="hl com"># Reorder the columns</span>
<span class="hl def">data_matrix_KEGG_UP</span> <span class="hl kwb">&lt;-</span> <span class="hl def">data_matrix_KEGG_UP[,</span> <span class="hl kwd">c</span><span class="hl def">(</span>
  <span class="hl sng">&quot;UP_ratio_AsymAD_female&quot;</span><span class="hl def">,</span>
  <span class="hl sng">&quot;UP_ratio_AD_female&quot;</span><span class="hl def">,</span>
  <span class="hl sng">&quot;UP_ratio_AsymAD_male&quot;</span><span class="hl def">,</span>
  <span class="hl sng">&quot;UP_ratio_AD_male&quot;</span>
<span class="hl def">)]</span>

<span class="hl def">data_matrix_KEGG_DOWN</span> <span class="hl kwb">&lt;-</span> <span class="hl def">data_matrix_KEGG_DOWN[,</span> <span class="hl kwd">c</span><span class="hl def">(</span>
  <span class="hl sng">&quot;DOWN_ratio_AsymAD_female&quot;</span><span class="hl def">,</span>
  <span class="hl sng">&quot;DOWN_ratio_AD_female&quot;</span><span class="hl def">,</span>
  <span class="hl sng">&quot;DOWN_ratio_AsymAD_male&quot;</span><span class="hl def">,</span>
  <span class="hl sng">&quot;DOWN_ratio_AD_male&quot;</span>

<span class="hl def">)]</span>


<span class="hl com"># Assign custom column names</span>
<span class="hl kwd">colnames</span><span class="hl def">(data_matrix_KEGG_UP)</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;AD_Female&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;AsymAD_Female&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;AD_Male&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;AsymAD_Male&quot;</span><span class="hl def">)</span>
<span class="hl kwd">colnames</span><span class="hl def">(data_matrix_KEGG_DOWN)</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;AD_Female&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;AsymAD_Female&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;AD_Male&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;AsymAD_Male&quot;</span><span class="hl def">)</span>


<span class="hl com"># Replace NA values with 0 (or any other specific value)</span>
<span class="hl def">data_matrix_KEGG_UP[</span><span class="hl kwd">is.na</span><span class="hl def">(data_matrix_KEGG_UP)]</span> <span class="hl kwb">&lt;-</span> <span class="hl num">0</span>
<span class="hl def">data_matrix_KEGG_DOWN[</span><span class="hl kwd">is.na</span><span class="hl def">(data_matrix_KEGG_DOWN)]</span> <span class="hl kwb">&lt;-</span> <span class="hl num">0</span>

<span class="hl com"># Set up the color scale using RColorBrewer</span>
<span class="hl def">color_scale</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">colorRampPalette</span><span class="hl def">(</span><span class="hl kwd">brewer.pal</span><span class="hl def">(</span><span class="hl num">9</span><span class="hl def">,</span> <span class="hl sng">&quot;RdYlGn&quot;</span><span class="hl def">))</span>


<span class="hl kwd">heatmap.2</span><span class="hl def">(</span>
<span class="hl kwd">as.matrix</span><span class="hl def">(data_matrix_KEGG_UP),</span>
<span class="hl kwc">Rowv</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>
<span class="hl kwc">Colv</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">,</span>
<span class="hl kwc">col</span> <span class="hl def">= color_scale3,</span>
<span class="hl kwc">scale</span> <span class="hl def">=</span> <span class="hl sng">&quot;none&quot;</span><span class="hl def">,</span>  <span class="hl com"># Scale rows (genes) to Z-scores</span>
<span class="hl kwc">trace</span> <span class="hl def">=</span> <span class="hl sng">&quot;none&quot;</span><span class="hl def">,</span>  <span class="hl com"># Turn off row and column annotations</span>
<span class="hl kwc">margins</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl num">20</span><span class="hl def">,</span> <span class="hl num">40</span><span class="hl def">),</span>  <span class="hl com"># Set margins for row and column labels</span>
<span class="hl kwc">key</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>  <span class="hl com"># Include a color key</span>
<span class="hl kwc">keysize</span> <span class="hl def">=</span> <span class="hl num">.5</span><span class="hl def">,</span>  <span class="hl com"># Size of the color key</span>
<span class="hl kwc">cexCol</span> <span class="hl def">=</span> <span class="hl num">1.5</span><span class="hl def">,</span>  <span class="hl com"># Set column label size</span>
<span class="hl kwc">cexRow</span> <span class="hl def">=</span> <span class="hl num">1.5</span><span class="hl def">,</span>  <span class="hl com"># Set row label size</span>
<span class="hl kwc">cellnote</span><span class="hl def">=</span><span class="hl kwd">round</span><span class="hl def">(data_matrix_KEGG_UP,</span> <span class="hl num">2</span><span class="hl def">),</span>
<span class="hl kwc">notecol</span><span class="hl def">=</span><span class="hl sng">&quot;black&quot;</span><span class="hl def">,</span>
<span class="hl kwc">labRow</span> <span class="hl def">= data_KEGG</span><span class="hl opt">$</span><span class="hl def">Pathway_name,</span>  <span class="hl com"># Row labels (Pathway name)</span>
<span class="hl kwc">dendrogram</span> <span class="hl def">=</span> <span class="hl sng">&quot;row&quot;</span><span class="hl def">,</span>  <span class="hl com"># Show both row and column dendrograms</span>
<span class="hl kwc">main</span><span class="hl def">=</span><span class="hl sng">&quot;Individuals (ratio) in each group exhibit a significant pathway (KEGG)&quot;</span>
<span class="hl def">)</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-1-4.png" alt="plot of chunk unnamed-chunk-1" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl kwd">heatmap.2</span><span class="hl def">(</span>
<span class="hl kwd">as.matrix</span><span class="hl def">(data_matrix_KEGG_DOWN),</span>
<span class="hl kwc">Rowv</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>
<span class="hl kwc">Colv</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">,</span>
<span class="hl kwc">col</span> <span class="hl def">= color_scale6,</span>
<span class="hl kwc">scale</span> <span class="hl def">=</span> <span class="hl sng">&quot;none&quot;</span><span class="hl def">,</span>  <span class="hl com"># Scale rows (genes) to Z-scores</span>
<span class="hl kwc">trace</span> <span class="hl def">=</span> <span class="hl sng">&quot;none&quot;</span><span class="hl def">,</span>  <span class="hl com"># Turn off row and column annotations</span>
<span class="hl kwc">margins</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl num">20</span><span class="hl def">,</span> <span class="hl num">40</span><span class="hl def">),</span>  <span class="hl com"># Set margins for row and column labels</span>
<span class="hl kwc">key</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>  <span class="hl com"># Include a color key</span>
<span class="hl kwc">keysize</span> <span class="hl def">=</span> <span class="hl num">.5</span><span class="hl def">,</span>  <span class="hl com"># Size of the color key</span>
<span class="hl kwc">cexCol</span> <span class="hl def">=</span> <span class="hl num">1.5</span><span class="hl def">,</span>  <span class="hl com"># Set column label size</span>
<span class="hl kwc">cexRow</span> <span class="hl def">=</span> <span class="hl num">1.5</span><span class="hl def">,</span>  <span class="hl com"># Set row label size</span>
<span class="hl kwc">cellnote</span><span class="hl def">=</span><span class="hl kwd">round</span><span class="hl def">(data_matrix_KEGG_DOWN,</span> <span class="hl num">2</span><span class="hl def">),</span>
<span class="hl kwc">notecol</span><span class="hl def">=</span><span class="hl sng">&quot;black&quot;</span><span class="hl def">,</span>
<span class="hl kwc">labRow</span> <span class="hl def">= data_KEGG</span><span class="hl opt">$</span><span class="hl def">Pathway_name,</span>  <span class="hl com"># Row labels (Pathway name)</span>
<span class="hl kwc">dendrogram</span> <span class="hl def">=</span> <span class="hl sng">&quot;row&quot;</span><span class="hl def">,</span>  <span class="hl com"># Show both row and column dendrograms</span>
<span class="hl kwc">main</span><span class="hl def">=</span><span class="hl sng">&quot;Individuals (ratio) in each group exhibit a significant pathway (KEGG)&quot;</span>
<span class="hl def">)</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-1-5.png" alt="plot of chunk unnamed-chunk-1" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">##################################################################################################################################################</span>


<span class="hl def">subset_data_KEGG_UP</span> <span class="hl kwb">&lt;-</span> <span class="hl def">data_KEGG_new_UP</span> <span class="hl opt">%&gt;%</span> <span class="hl kwd">filter</span><span class="hl def">(UP_ratio_AD_female</span> <span class="hl opt">&gt;=</span> <span class="hl num">0.10</span> <span class="hl opt">|</span> <span class="hl def">UP_ratio_AD_male</span> <span class="hl opt">&gt;=</span> <span class="hl num">0.10</span> <span class="hl opt">|</span> <span class="hl def">UP_ratio_AsymAD_female</span> <span class="hl opt">&gt;=</span> <span class="hl num">0.10</span> <span class="hl opt">|</span> <span class="hl def">UP_ratio_AsymAD_male</span> <span class="hl opt">&gt;=</span> <span class="hl num">0.10</span><span class="hl def">)</span>

<span class="hl def">subset_data_KEGG_DOWN</span> <span class="hl kwb">&lt;-</span> <span class="hl def">data_KEGG_new_DOWN</span> <span class="hl opt">%&gt;%</span> <span class="hl kwd">filter</span><span class="hl def">(DOWN_ratio_AD_female</span> <span class="hl opt">&gt;=</span> <span class="hl num">0.10</span> <span class="hl opt">|</span> <span class="hl def">DOWN_ratio_AD_male</span> <span class="hl opt">&gt;=</span> <span class="hl num">0.10</span> <span class="hl opt">|</span> <span class="hl def">DOWN_ratio_AsymAD_female</span> <span class="hl opt">&gt;=</span> <span class="hl num">0.10</span> <span class="hl opt">|</span> <span class="hl def">DOWN_ratio_AsymAD_male</span> <span class="hl opt">&gt;=</span> <span class="hl num">0.10</span><span class="hl def">)</span>


<span class="hl kwd">rownames</span><span class="hl def">(subset_data_KEGG_UP)</span> <span class="hl kwb">&lt;-</span> <span class="hl def">subset_data_KEGG_UP</span><span class="hl opt">$</span><span class="hl def">Pathway_name</span>
<span class="hl kwd">rownames</span><span class="hl def">(subset_data_KEGG_DOWN)</span> <span class="hl kwb">&lt;-</span> <span class="hl def">subset_data_KEGG_DOWN</span><span class="hl opt">$</span><span class="hl def">Pathway_name</span>


<span class="hl def">subset_data_KEGG_UP</span> <span class="hl kwb">&lt;-</span> <span class="hl def">subset_data_KEGG_UP[,</span> <span class="hl opt">-</span><span class="hl num">1</span><span class="hl def">]</span>
<span class="hl def">subset_data_KEGG_DOWN</span> <span class="hl kwb">&lt;-</span> <span class="hl def">subset_data_KEGG_DOWN[,</span> <span class="hl opt">-</span><span class="hl num">1</span><span class="hl def">]</span>


<span class="hl com"># Replace NA values with 0 (or any other specific value)</span>
<span class="hl def">subset_data_KEGG_UP[</span><span class="hl kwd">is.na</span><span class="hl def">(subset_data_KEGG_UP)]</span> <span class="hl kwb">&lt;-</span> <span class="hl num">0</span>
<span class="hl def">subset_data_KEGG_DOWN[</span><span class="hl kwd">is.na</span><span class="hl def">(subset_data_KEGG_DOWN)]</span> <span class="hl kwb">&lt;-</span> <span class="hl num">0</span>


<span class="hl kwd">head</span><span class="hl def">(subset_data_KEGG_UP,</span> <span class="hl num">3</span><span class="hl def">)</span>
</pre></div>
<div class="output"><pre class="knitr r">##                                           UP_ratio_AD_female UP_ratio_AD_male
## Pentose phosphate pathway                          0.0390625         0.106383
## Pentose and glucuronate interconversions           0.0625000         0.106383
## Galactose metabolism                               0.1093750         0.106383
##                                           UP_ratio_AsymAD_female
## Pentose phosphate pathway                             0.01886793
## Pentose and glucuronate interconversions              0.02515723
## Galactose metabolism                                  0.05031447
##                                           UP_ratio_AsymAD_male
## Pentose phosphate pathway                           0.06557377
## Pentose and glucuronate interconversions            0.04918033
## Galactose metabolism                                0.04918033
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl kwd">head</span><span class="hl def">(subset_data_KEGG_DOWN,</span> <span class="hl num">3</span><span class="hl def">)</span>
</pre></div>
<div class="output"><pre class="knitr r">##                                             DOWN_ratio_AD_female
## Valine, leucine and isoleucine degradation             0.0234375
## beta-Alanine metabolism                                0.0625000
## Glutathione metabolism                                 0.0078125
##                                             DOWN_ratio_AD_male
## Valine, leucine and isoleucine degradation           0.1063830
## beta-Alanine metabolism                              0.1489362
## Glutathione metabolism                               0.1702128
##                                             DOWN_ratio_AsymAD_female
## Valine, leucine and isoleucine degradation               0.006289308
## beta-Alanine metabolism                                  0.018867925
## Glutathione metabolism                                   0.012578616
##                                             DOWN_ratio_AsymAD_male
## Valine, leucine and isoleucine degradation              0.01639344
## beta-Alanine metabolism                                 0.06557377
## Glutathione metabolism                                  0.14754098
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl com"># Reorder the columns</span>
<span class="hl def">subset_data_KEGG_UP</span> <span class="hl kwb">&lt;-</span> <span class="hl def">subset_data_KEGG_UP[,</span> <span class="hl kwd">c</span><span class="hl def">(</span>
  <span class="hl sng">&quot;UP_ratio_AsymAD_female&quot;</span><span class="hl def">,</span>
  <span class="hl sng">&quot;UP_ratio_AD_female&quot;</span><span class="hl def">,</span>
  <span class="hl sng">&quot;UP_ratio_AsymAD_male&quot;</span><span class="hl def">,</span>
  <span class="hl sng">&quot;UP_ratio_AD_male&quot;</span>
<span class="hl def">)]</span>

<span class="hl def">subset_data_KEGG_DOWN</span> <span class="hl kwb">&lt;-</span> <span class="hl def">subset_data_KEGG_DOWN[,</span> <span class="hl kwd">c</span><span class="hl def">(</span>
  <span class="hl sng">&quot;DOWN_ratio_AsymAD_female&quot;</span><span class="hl def">,</span>
  <span class="hl sng">&quot;DOWN_ratio_AD_female&quot;</span><span class="hl def">,</span>
  <span class="hl sng">&quot;DOWN_ratio_AsymAD_male&quot;</span><span class="hl def">,</span>
  <span class="hl sng">&quot;DOWN_ratio_AD_male&quot;</span>
<span class="hl def">)]</span>


<span class="hl com"># Assign custom column names</span>
<span class="hl kwd">colnames</span><span class="hl def">(subset_data_KEGG_UP)</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;AsymAD_Female&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;AD_Female&quot;</span><span class="hl def">,</span>  <span class="hl sng">&quot;AsymAD_Male&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;AD_Male&quot;</span><span class="hl def">)</span>
<span class="hl kwd">colnames</span><span class="hl def">(subset_data_KEGG_DOWN)</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;AsymAD_Female&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;AD_Female&quot;</span><span class="hl def">,</span>  <span class="hl sng">&quot;AsymAD_Male&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;AD_Male&quot;</span><span class="hl def">)</span>


<span class="hl com"># Create the heatmap with filtered row names</span>
<span class="hl def">UP</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">heatmap.2</span><span class="hl def">(</span>
     <span class="hl kwd">as.matrix</span><span class="hl def">(subset_data_KEGG_UP),</span>
     <span class="hl kwc">Rowv</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>
     <span class="hl kwc">Colv</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">,</span>
     <span class="hl kwc">col</span> <span class="hl def">= color_scale3,</span>
     <span class="hl kwc">scale</span> <span class="hl def">=</span> <span class="hl sng">&quot;none&quot;</span><span class="hl def">,</span>
     <span class="hl kwc">trace</span> <span class="hl def">=</span> <span class="hl sng">&quot;none&quot;</span><span class="hl def">,</span>
     <span class="hl kwc">margins</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl num">20</span><span class="hl def">,</span> <span class="hl num">50</span><span class="hl def">),</span>
     <span class="hl kwc">key</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>
     <span class="hl kwc">keysize</span> <span class="hl def">=</span> <span class="hl num">.50</span><span class="hl def">,</span>
     <span class="hl kwc">cexCol</span> <span class="hl def">=</span> <span class="hl num">2.5</span><span class="hl def">,</span>
     <span class="hl kwc">srtCol</span> <span class="hl def">=</span> <span class="hl num">45</span><span class="hl def">,</span>
     <span class="hl kwc">cexRow</span> <span class="hl def">=</span> <span class="hl num">2</span><span class="hl def">,</span>
    <span class="hl kwc">cellnote</span><span class="hl def">=</span><span class="hl kwd">round</span><span class="hl def">(subset_data_KEGG_UP,</span> <span class="hl num">2</span><span class="hl def">),</span>
    <span class="hl kwc">notecex</span><span class="hl def">=</span><span class="hl num">2</span><span class="hl def">,</span>
    <span class="hl kwc">notecol</span><span class="hl def">=</span><span class="hl sng">&quot;black&quot;</span><span class="hl def">,</span>
    <span class="hl kwc">dendrogram</span> <span class="hl def">=</span> <span class="hl sng">&quot;row&quot;</span><span class="hl def">,</span>
    <span class="hl kwc">main</span> <span class="hl def">=</span> <span class="hl sng">&quot;Pathways Present in atleast 10% Individuals&quot;</span>
<span class="hl def">)</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-1-6.png" alt="plot of chunk unnamed-chunk-1" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com"># Create the heatmap with filtered row names</span>
<span class="hl def">DOWN</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">heatmap.2</span><span class="hl def">(</span>
     <span class="hl kwd">as.matrix</span><span class="hl def">(subset_data_KEGG_DOWN),</span>
     <span class="hl kwc">Rowv</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>
     <span class="hl kwc">Colv</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">,</span>
     <span class="hl kwc">col</span> <span class="hl def">= color_scale6,</span>
     <span class="hl kwc">scale</span> <span class="hl def">=</span> <span class="hl sng">&quot;none&quot;</span><span class="hl def">,</span>
     <span class="hl kwc">trace</span> <span class="hl def">=</span> <span class="hl sng">&quot;none&quot;</span><span class="hl def">,</span>
     <span class="hl kwc">margins</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl num">20</span><span class="hl def">,</span> <span class="hl num">50</span><span class="hl def">),</span>
     <span class="hl kwc">key</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>
     <span class="hl kwc">keysize</span> <span class="hl def">=</span> <span class="hl num">.50</span><span class="hl def">,</span>
     <span class="hl kwc">cexCol</span> <span class="hl def">=</span> <span class="hl num">2.5</span><span class="hl def">,</span>
     <span class="hl kwc">srtCol</span> <span class="hl def">=</span> <span class="hl num">45</span><span class="hl def">,</span>
     <span class="hl kwc">cexRow</span> <span class="hl def">=</span> <span class="hl num">2</span><span class="hl def">,</span>
    <span class="hl kwc">cellnote</span><span class="hl def">=</span><span class="hl kwd">round</span><span class="hl def">(subset_data_KEGG_DOWN,</span> <span class="hl num">2</span><span class="hl def">),</span>
    <span class="hl kwc">notecex</span><span class="hl def">=</span><span class="hl num">2</span><span class="hl def">,</span>
    <span class="hl kwc">notecol</span><span class="hl def">=</span><span class="hl sng">&quot;black&quot;</span><span class="hl def">,</span>
    <span class="hl kwc">dendrogram</span> <span class="hl def">=</span> <span class="hl sng">&quot;row&quot;</span><span class="hl def">,</span>
    <span class="hl kwc">main</span> <span class="hl def">=</span> <span class="hl sng">&quot;Pathways Present in atleast 10% Individuals&quot;</span>
<span class="hl def">)</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-1-7.png" alt="plot of chunk unnamed-chunk-1" class="plot" /></div></div>

<p>Comparative analysis between/within group for associated pathways (KEGG)</p>

#<!--begin.rcode fig.width=25, fig.height=15

########################################################################################################################################################################################################

# Define a custom color palette with distinct colors for each biodomain (different type of Biodomain - color define by Greg Carry)
custom_colors <- c("Apoptosis" = "#673399", "APP Metabolism" = "#fe6500", "Autophagy" = "#9931fd", "Cell Cycle" = "#18857f", "DNA Repair" = "#f451ad", "Endolysosome" = "#3466cc", "Epigenetic" = "#cb3233", "Immune Response" = "#9ccdcc", "Lipid Metabolism" = "#989898", "Metal Binding and Homeostasis" = "#4b0d20", "Mitochondrial Metabolism" = "#97cb98", "Myelination" = "#996735", "Oxidative Stress" = "#ffcd66", "Proteostasis" = "#c8b269", "RNA Spliceosome" = "#0c9aff", "Structural Stabilization" = "#ff9a9a", "Synapse" = "#329a33", "Tau Homeostasis" = "#cb97cb", "Vasculature" = "#cecd02", "none" = "#7f7f7f")

## AD

# Total number of individual for pathway enrichment in AD (Male vs Female)
#ggplot(data_KEGG, aes(x= Total_count_AD_female, y= Total_count_AD_male, color = Pathway_gene_ratio)) +geom_point(alpha=0.3, shape=21, color="black") + theme_bw() + geom_text_repel(aes(label = Pathway_name), size = 7.5) + geom_smooth(method = 'lm', colour="red", size=0.5) +  scale_color_gradient(low="red", high="black") + theme(text = element_text(size=20))

# Sex specific UP pathway enrichment in AD (Male vs Female)
ggplot(data_KEGG, aes(x= UP_ratio_AD_female, y= UP_ratio_AD_male, color = Biodomain, size = Jaccard_Index)) + geom_point(aes(color=Biodomain, size = Jaccard_Index), alpha=.3, stroke = 0)  + theme_bw() + geom_text_repel(aes(label = Pathway_name), size = 5) + geom_smooth(method = 'lm', colour="red", size=0.5) + theme(text = element_text(size=20)) + scale_color_manual(values = custom_colors) + labs(color = "Biodomain")

# Sex specific DOWN pathway enrichment in AD (Male vs Female)
ggplot(data_KEGG, aes(x= DOWN_ratio_AD_female, y= DOWN_ratio_AD_male, color = Biodomain, size = Jaccard_Index)) + geom_point(aes(color=Biodomain, size = Jaccard_Index), alpha=.3, stroke = 0)  + theme_bw() + geom_text_repel(aes(label = Pathway_name), size = 5) + geom_smooth(method = 'lm', colour="red", size=0.5) + theme(text = element_text(size=20)) + scale_color_manual(values = custom_colors)  + labs(color = "Biodomain")


# UP and DOWN pathways comparison in AD (Female)
ggplot(data_KEGG, aes(x= DOWN_ratio_AD_female, y= UP_ratio_AD_female, color = Biodomain, size = Jaccard_Index)) + geom_point(aes(color=Biodomain, size = Jaccard_Index), alpha=.3, stroke = 0)  + theme_bw() + geom_text_repel(aes(label = Pathway_name), size = 5) + geom_smooth(method = 'lm', colour="red", size=0.5) + theme(text = element_text(size=20)) + scale_color_manual(values = custom_colors)  + labs(color = "Biodomain")

# UP and DOWN pathways comparison in AD (Male)
ggplot(data_KEGG, aes(x= DOWN_ratio_AD_male, y= UP_ratio_AD_male, color = Biodomain, size = Jaccard_Index)) + geom_point(aes(color=Biodomain, size = Jaccard_Index), alpha=.3, stroke = 0)  + theme_bw() + geom_text_repel(aes(label = Pathway_name), size = 5) + geom_smooth(method = 'lm', colour="red", size=0.5) + theme(text = element_text(size=20)) + scale_color_manual(values = custom_colors)  + labs(color = "Biodomain")

########################################################################################################################################################################################################
## AsymAD

# Sex specific UP pathway enrichment in AsymAD (Male vs Female)
ggplot(data_KEGG, aes(x= UP_ratio_AsymAD_female, y= UP_ratio_AsymAD_male, color = Biodomain, size = Jaccard_Index)) + geom_point(aes(color=Biodomain, size = Jaccard_Index), alpha=.3, stroke = 0)  + theme_bw() + geom_text_repel(aes(label = Pathway_name), size = 5) + geom_smooth(method = 'lm', colour="red", size=0.5) + theme(text = element_text(size=20)) + scale_color_manual(values = custom_colors)  + labs(color = "Biodomain")

# Sex specific DOWN pathway enrichment in AsymAD (Male vs Female)
ggplot(data_KEGG, aes(x= DOWN_ratio_AsymAD_female, y= DOWN_ratio_AsymAD_male, color = Biodomain, size = Jaccard_Index)) + geom_point(aes(color=Biodomain, size = Jaccard_Index), alpha=.3, stroke = 0)  + theme_bw() + geom_text_repel(aes(label = Pathway_name), size = 5) + geom_smooth(method = 'lm', colour="red", size=0.5) + theme(text = element_text(size=20)) + scale_color_manual(values = custom_colors)  + labs(color = "Biodomain")


# UP and DOWN pathways comparison in AsymAD (Female)
ggplot(data_KEGG, aes(x= DOWN_ratio_AsymAD_female, y= UP_ratio_AsymAD_female, color = Biodomain, size = Jaccard_Index)) + geom_point(aes(color=Biodomain, size = Jaccard_Index), alpha=.3, stroke = 0)  + theme_bw() + geom_text_repel(aes(label = Pathway_name), size = 5) + geom_smooth(method = 'lm', colour="red", size=0.5) + theme(text = element_text(size=20)) + scale_color_manual(values = custom_colors)  + labs(color = "Biodomain")

# UP and DOWN pathways comparison in AsymAD (Male)
ggplot(data_KEGG, aes(x= DOWN_ratio_AsymAD_male, y= UP_ratio_AsymAD_male, color = Biodomain, size = Jaccard_Index)) + geom_point(aes(color=Biodomain, size = Jaccard_Index), alpha=.3, stroke = 0)  + theme_bw() + geom_text_repel(aes(label = Pathway_name), size = 5) + geom_smooth(method = 'lm', colour="red", size=0.5) + theme(text = element_text(size=20)) + scale_color_manual(values = custom_colors)  + labs(color = "Biodomain")

########################################################################################################################################################################################################
## AD vs AsymAD

# UP pathway enrichment in AsymAD vs AD (Female)
ggplot(data_KEGG, aes(x= UP_ratio_AsymAD_female, y= UP_ratio_AD_female, color = Biodomain, size = Jaccard_Index)) + geom_point(aes(color=Biodomain, size = Jaccard_Index), alpha=.3, stroke = 0)  + theme_bw() + geom_text_repel(aes(label = Pathway_name), size = 5) + geom_smooth(method = 'lm', colour="red", size=0.5) + theme(text = element_text(size=20)) + scale_color_manual(values = custom_colors)  + labs(color = "Biodomain")


# DOWN pathway enrichment in AsymAD vs AD (Female)
ggplot(data_KEGG, aes(x= DOWN_ratio_AsymAD_female, y= DOWN_ratio_AD_female, color = Biodomain, size = Jaccard_Index)) + geom_point(aes(color=Biodomain, size = Jaccard_Index), alpha=.3, stroke = 0)  + theme_bw() + geom_text_repel(aes(label = Pathway_name), size = 5) + geom_smooth(method = 'lm', colour="red", size=0.5) + theme(text = element_text(size=20)) + scale_color_manual(values = custom_colors)  + labs(color = "Biodomain")


# UP pathway enrichment in AsymAD vs AD (Male)
ggplot(data_KEGG, aes(x= UP_ratio_AsymAD_male, y= UP_ratio_AD_male, color = Biodomain, size = Jaccard_Index)) + geom_point(aes(color=Biodomain, size = Jaccard_Index), alpha=.3, stroke = 0)  + theme_bw() + geom_text_repel(aes(label = Pathway_name), size = 5) + geom_smooth(method = 'lm', colour="red", size=0.5) + theme(text = element_text(size=20)) + scale_color_manual(values = custom_colors)  + labs(color = "Biodomain")


# DOWN pathway enrichment in AsymAD vs AD (Male)
ggplot(data_KEGG, aes(x= DOWN_ratio_AsymAD_male, y= DOWN_ratio_AD_male, color = Biodomain, size = Jaccard_Index)) + geom_point(aes(color=Biodomain, size = Jaccard_Index), alpha=.3, stroke = 0)  + theme_bw() + geom_text_repel(aes(label = Pathway_name), size = 5) + geom_smooth(method = 'lm', colour="red", size=0.5) + theme(text = element_text(size=20)) + scale_color_manual(values = custom_colors)  + labs(color = "Biodomain")


########################################################################################################################################################################################################

# Convert to interactive plot using plotly

data_KEGG_long <- melt(data_KEGG_new, id.vars = "Pathway_name")

p <- ggplot(data_KEGG_long, aes(x = Pathway_name, y = value, color = variable, group = variable)) +
     geom_line(size = 1.2) +              # Line plot
     geom_point(size = 3) +               # Add points to the line
     labs(title = "Comparative Plot for AD and AsymAD Ratios",
          x = "Pathway ID", 
          y = "Ratio Value") +
     theme_minimal() +
     theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 5),  # Rotate x-axis labels for better readability
           text = element_text(size = 14))

interactive_plot <- ggplotly(p)

interactive_plot

# Save the interactive plot as an HTML file
htmlwidgets::saveWidget(interactive_plot, "interactive_plot.html")

########################################################################################################################################################################################################

end.rcode-->

<p>Comparative analysis between/within group for associated pathways (KEGG)</p>

<div class="chunk" id="unnamed-chunk-2"><div class="rcode"><div class="source"><pre class="knitr r"><span class="hl com">########################################################################################################################################################################################################</span>

<span class="hl com"># Define a custom color palette with distinct colors for each biodomain (different type of Biodomain - color define by Greg Carry)</span>
<span class="hl com">#custom_colors &lt;- c(&quot;Apoptosis&quot; = &quot;#673399&quot;, &quot;APP Metabolism&quot; = &quot;#fe6500&quot;, &quot;Autophagy&quot; = &quot;#9931fd&quot;, &quot;Cell Cycle&quot; = &quot;#18857f&quot;, &quot;DNA Repair&quot; = &quot;#f451ad&quot;, &quot;Endolysosome&quot; = &quot;#3466cc&quot;, &quot;Epigenetic&quot; = &quot;#cb3233&quot;, #&quot;Immune Response&quot; = &quot;#9ccdcc&quot;, &quot;Lipid Metabolism&quot; = &quot;#989898&quot;, &quot;Metal Binding and Homeostasis&quot; = &quot;#4b0d20&quot;, &quot;Mitochondrial Metabolism&quot; = &quot;#97cb98&quot;, &quot;Myelination&quot; = &quot;#996735&quot;, &quot;Oxidative Stress&quot; = &quot;#ffcd66&quot;, #&quot;Proteostasis&quot; = &quot;#c8b269&quot;, &quot;RNA Spliceosome&quot; = &quot;#0c9aff&quot;, &quot;Structural Stabilization&quot; = &quot;#ff9a9a&quot;, &quot;Synapse&quot; = &quot;#329a33&quot;, &quot;Tau Homeostasis&quot; = &quot;#cb97cb&quot;, &quot;Vasculature&quot; = &quot;#cecd02&quot;, &quot;none&quot; = &quot;#7f7f7f&quot;)</span>


<span class="hl com">#  Male vs Female AD</span>
<span class="hl com"># Open the selected pathways where UP (+) and DOWN (-) pathways are merged together in a single column and the data is separated as a comparison for Male vs Female pathways</span>
<span class="hl com">#data_KEGG_MvF_AD &lt;- read.csv(&quot;Male_Female_comarative_pathway_results_final_111324.csv&quot;, sep =&quot;,&quot;, header = TRUE, stringsAsFactors = FALSE)</span>

<span class="hl def">data_KEGG_MvF_AD</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">read.csv</span><span class="hl def">(</span><span class="hl sng">&quot;Male_Female_comarative_pathway_results_final_050525.csv&quot;</span><span class="hl def">,</span> <span class="hl kwc">sep</span> <span class="hl def">=</span><span class="hl sng">&quot;,&quot;</span><span class="hl def">,</span> <span class="hl kwc">header</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span> <span class="hl kwc">stringsAsFactors</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">)</span>

<span class="hl kwd">head</span><span class="hl def">(data_KEGG_MvF_AD,</span> <span class="hl num">5</span><span class="hl def">)</span>
</pre></div>
<div class="output"><pre class="knitr r">##   Sno Pathway_ID                      Pathway_name_initial
## 1   1   hsa00010             Glycolysis / Gluconeogenesis 
## 2   3   hsa00030                Pentose phosphate pathway 
## 3   4   hsa00040 Pentose and glucuronate interconversions 
## 4   5   hsa00051          Fructose and mannose metabolism 
## 5   6   hsa00052                     Galactose metabolism 
##                                  Pathway_name    Female       Male Direction
## 1             Glycolysis / Gluconeogenesis  + 0.0937500 0.04255319        UP
## 2                Pentose phosphate pathway  + 0.0390625 0.10638298        UP
## 3 Pentose and glucuronate interconversions  + 0.0625000 0.10638298        UP
## 4          Fructose and mannose metabolism  + 0.0859375 0.08510638        UP
## 5                     Galactose metabolism  + 0.1093750 0.10638298        UP
##    Difference
## 1 0.051196809
## 2 0.067320479
## 3 0.043882979
## 4 0.000831117
## 5 0.002992021
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl com"># Plot the data as a comparison for Male vs Female pathways</span>

<span class="hl kwd">ggplot</span><span class="hl def">(data_KEGG_MvF_AD,</span> <span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">x</span> <span class="hl def">= Female,</span> <span class="hl kwc">y</span> <span class="hl def">= Male,</span> <span class="hl kwc">size</span> <span class="hl def">= Difference))</span> <span class="hl opt">+</span>
  <span class="hl kwd">geom_point</span><span class="hl def">(</span><span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">size</span> <span class="hl def">= Gene_ratio),</span> <span class="hl kwc">alpha</span><span class="hl def">=</span><span class="hl num">0.6</span><span class="hl def">,</span> <span class="hl kwc">stroke</span> <span class="hl def">=</span> <span class="hl num">0</span><span class="hl def">)</span> <span class="hl opt">+</span>
    <span class="hl kwd">theme_bw</span><span class="hl def">()</span> <span class="hl opt">+</span>
    <span class="hl com"># Conditionally label points where Female or Male values are greater than 0.05</span>
    <span class="hl kwd">geom_text_repel</span><span class="hl def">(</span>
        <span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">label</span> <span class="hl def">=</span> <span class="hl kwd">ifelse</span><span class="hl def">((Male</span> <span class="hl opt">&gt;</span> <span class="hl num">0.10</span> <span class="hl opt">&amp;</span> <span class="hl def">Female</span> <span class="hl opt">&lt;</span> <span class="hl num">0.10</span><span class="hl def">)</span> <span class="hl opt">|</span> <span class="hl def">(Female</span> <span class="hl opt">&gt;</span> <span class="hl num">0.10</span> <span class="hl opt">&amp;</span> <span class="hl def">Male</span> <span class="hl opt">&lt;</span> <span class="hl num">0.05</span><span class="hl def">)</span> <span class="hl opt">|</span> <span class="hl def">(Female</span> <span class="hl opt">&gt;</span> <span class="hl num">0.05</span> <span class="hl opt">&amp;</span> <span class="hl def">Male</span> <span class="hl opt">&gt;</span> <span class="hl num">0.05</span><span class="hl def">), Pathway_name,</span> <span class="hl sng">&quot;&quot;</span><span class="hl def">)),</span>
        <span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">6</span><span class="hl def">,</span> <span class="hl kwc">max.overlaps</span> <span class="hl def">=</span> <span class="hl num">Inf</span>
    <span class="hl def">)</span> <span class="hl opt">+</span>
    <span class="hl com">#scale_color_manual(values = custom_colors) +</span>
    <span class="hl com">#labs(color = &quot;Biodomain&quot;) +</span>
    <span class="hl com"># Add x = y line</span>
    <span class="hl kwd">geom_abline</span><span class="hl def">(</span><span class="hl kwc">slope</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">,</span> <span class="hl kwc">intercept</span> <span class="hl def">=</span> <span class="hl num">0</span><span class="hl def">,</span> <span class="hl kwc">linetype</span> <span class="hl def">=</span> <span class="hl sng">&quot;dashed&quot;</span><span class="hl def">,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;gray&quot;</span><span class="hl def">,</span> <span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">0.5</span><span class="hl def">)</span> <span class="hl opt">+</span>
    <span class="hl com"># Add horizontal and vertical dashed lines</span>
    <span class="hl kwd">geom_hline</span><span class="hl def">(</span><span class="hl kwc">yintercept</span> <span class="hl def">=</span> <span class="hl num">0.05</span><span class="hl def">,</span> <span class="hl kwc">linetype</span> <span class="hl def">=</span> <span class="hl sng">&quot;dashed&quot;</span><span class="hl def">,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;gray&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>
    <span class="hl kwd">geom_hline</span><span class="hl def">(</span><span class="hl kwc">yintercept</span> <span class="hl def">=</span> <span class="hl num">0.10</span><span class="hl def">,</span> <span class="hl kwc">linetype</span> <span class="hl def">=</span> <span class="hl sng">&quot;dashed&quot;</span><span class="hl def">,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;red&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>
    <span class="hl kwd">geom_vline</span><span class="hl def">(</span><span class="hl kwc">xintercept</span> <span class="hl def">=</span> <span class="hl num">0.05</span><span class="hl def">,</span> <span class="hl kwc">linetype</span> <span class="hl def">=</span> <span class="hl sng">&quot;dashed&quot;</span><span class="hl def">,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;gray&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>
    <span class="hl kwd">geom_vline</span><span class="hl def">(</span><span class="hl kwc">xintercept</span> <span class="hl def">=</span> <span class="hl num">0.10</span><span class="hl def">,</span> <span class="hl kwc">linetype</span> <span class="hl def">=</span> <span class="hl sng">&quot;dashed&quot;</span><span class="hl def">,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;red&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>
    <span class="hl kwd">labs</span><span class="hl def">(</span><span class="hl kwc">title</span> <span class="hl def">=</span> <span class="hl sng">&quot;Pathways Comparisons for Male_AD vs Female_AD &quot;</span><span class="hl def">,</span> <span class="hl kwc">x</span> <span class="hl def">=</span> <span class="hl sng">&quot;Female (% of subjects)&quot;</span><span class="hl def">,</span> <span class="hl kwc">y</span> <span class="hl def">=</span> <span class="hl sng">&quot;Male (% of subjects)&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>
    <span class="hl kwd">theme</span><span class="hl def">(</span>
    <span class="hl kwc">text</span> <span class="hl def">=</span> <span class="hl kwd">element_text</span><span class="hl def">(</span><span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">30</span><span class="hl def">),</span>
    <span class="hl kwc">legend.position</span> <span class="hl def">=</span> <span class="hl sng">&quot;right&quot;</span><span class="hl def">,</span>
    <span class="hl kwc">axis.text</span> <span class="hl def">=</span> <span class="hl kwd">element_text</span><span class="hl def">(</span><span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">30</span><span class="hl def">),</span>
    <span class="hl kwc">axis.title</span> <span class="hl def">=</span> <span class="hl kwd">element_text</span><span class="hl def">(</span><span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">30</span><span class="hl def">)</span>
  <span class="hl def">)</span>
</pre></div>
<div class="warning"><pre class="knitr r">## Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
## ℹ Please use `linewidth` instead.
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
## generated.
</pre></div>
<div class="error"><pre class="knitr r">## Error in `geom_point()`:
## ! Problem while computing aesthetics.
## ℹ Error occurred in the 1st layer.
## Caused by error:
## ! object 'Gene_ratio' not found
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl com">########################################################################################################################################################################################################</span>

<span class="hl kwd">ggplot</span><span class="hl def">(data_KEGG_MvF_AD,</span> <span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">x</span> <span class="hl def">= Female,</span> <span class="hl kwc">y</span> <span class="hl def">= Male,</span> <span class="hl kwc">size</span> <span class="hl def">= Difference))</span> <span class="hl opt">+</span>
    <span class="hl kwd">geom_point</span><span class="hl def">(</span><span class="hl kwc">alpha</span> <span class="hl def">=</span> <span class="hl num">0.6</span><span class="hl def">,</span> <span class="hl kwc">stroke</span> <span class="hl def">=</span> <span class="hl num">0</span><span class="hl def">)</span> <span class="hl opt">+</span>  <span class="hl com"># Increased alpha for better visibility</span>
    <span class="hl kwd">theme_bw</span><span class="hl def">()</span> <span class="hl opt">+</span>

    <span class="hl com"># Conditionally label points where |Male - Female| &gt;= 0.05</span>
    <span class="hl kwd">geom_text_repel</span><span class="hl def">(</span>
        <span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">label</span> <span class="hl def">=</span> <span class="hl kwd">ifelse</span><span class="hl def">((</span><span class="hl kwd">abs</span><span class="hl def">(Female</span> <span class="hl opt">-</span> <span class="hl def">Male)</span> <span class="hl opt">&gt;=</span> <span class="hl num">0.10</span><span class="hl def">)</span> <span class="hl opt">|</span> <span class="hl def">(Female</span> <span class="hl opt">&gt;</span> <span class="hl num">0.10</span><span class="hl def">)</span> <span class="hl opt">|</span> <span class="hl def">(Female</span> <span class="hl opt">&gt;</span> <span class="hl num">0.07</span> <span class="hl opt">&amp;</span> <span class="hl def">Male</span> <span class="hl opt">&gt;</span> <span class="hl num">0.07</span><span class="hl def">), Pathway_name,</span> <span class="hl sng">&quot;&quot;</span><span class="hl def">)),</span>
        <span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">6</span><span class="hl def">,</span>
        <span class="hl kwc">max.overlaps</span> <span class="hl def">=</span> <span class="hl num">Inf</span>  <span class="hl com"># Ensure all significant labels are shown</span>
    <span class="hl def">)</span> <span class="hl opt">+</span>

   <span class="hl com"># scale_color_manual(values = custom_colors) +</span>

    <span class="hl com">#labs(color = &quot;Biodomain&quot;) +</span>

    <span class="hl com"># Add x = y reference line</span>
    <span class="hl kwd">geom_abline</span><span class="hl def">(</span><span class="hl kwc">slope</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">,</span> <span class="hl kwc">intercept</span> <span class="hl def">=</span> <span class="hl num">0</span><span class="hl def">,</span> <span class="hl kwc">linetype</span> <span class="hl def">=</span> <span class="hl sng">&quot;dashed&quot;</span><span class="hl def">,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;gray&quot;</span><span class="hl def">,</span> <span class="hl kwc">linewidth</span> <span class="hl def">=</span> <span class="hl num">0.5</span><span class="hl def">)</span> <span class="hl opt">+</span>

    <span class="hl com"># Add horizontal and vertical threshold lines</span>
    <span class="hl kwd">geom_hline</span><span class="hl def">(</span><span class="hl kwc">yintercept</span> <span class="hl def">=</span> <span class="hl num">0.05</span><span class="hl def">,</span> <span class="hl kwc">linetype</span> <span class="hl def">=</span> <span class="hl sng">&quot;dashed&quot;</span><span class="hl def">,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;gray&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>
    <span class="hl kwd">geom_hline</span><span class="hl def">(</span><span class="hl kwc">yintercept</span> <span class="hl def">=</span> <span class="hl num">0.10</span><span class="hl def">,</span> <span class="hl kwc">linetype</span> <span class="hl def">=</span> <span class="hl sng">&quot;dashed&quot;</span><span class="hl def">,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;red&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>
    <span class="hl kwd">geom_vline</span><span class="hl def">(</span><span class="hl kwc">xintercept</span> <span class="hl def">=</span> <span class="hl num">0.05</span><span class="hl def">,</span> <span class="hl kwc">linetype</span> <span class="hl def">=</span> <span class="hl sng">&quot;dashed&quot;</span><span class="hl def">,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;gray&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>
    <span class="hl kwd">geom_vline</span><span class="hl def">(</span><span class="hl kwc">xintercept</span> <span class="hl def">=</span> <span class="hl num">0.10</span><span class="hl def">,</span> <span class="hl kwc">linetype</span> <span class="hl def">=</span> <span class="hl sng">&quot;dashed&quot;</span><span class="hl def">,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;red&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>

    <span class="hl kwd">labs</span><span class="hl def">(</span><span class="hl kwc">title</span> <span class="hl def">=</span> <span class="hl sng">&quot;Pathway Comparisons for Female_AD vs Male_AD&quot;</span><span class="hl def">,</span>
         <span class="hl kwc">x</span> <span class="hl def">=</span> <span class="hl sng">&quot;Female (% of subjects)&quot;</span><span class="hl def">,</span>
         <span class="hl kwc">y</span> <span class="hl def">=</span> <span class="hl sng">&quot;Male (% of subjects)&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>

    <span class="hl kwd">theme</span><span class="hl def">(</span>
        <span class="hl kwc">text</span> <span class="hl def">=</span> <span class="hl kwd">element_text</span><span class="hl def">(</span><span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">30</span><span class="hl def">),</span>
        <span class="hl kwc">legend.position</span> <span class="hl def">=</span> <span class="hl sng">&quot;right&quot;</span><span class="hl def">,</span>
        <span class="hl kwc">axis.text</span> <span class="hl def">=</span> <span class="hl kwd">element_text</span><span class="hl def">(</span><span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">30</span><span class="hl def">),</span>
        <span class="hl kwc">axis.title</span> <span class="hl def">=</span> <span class="hl kwd">element_text</span><span class="hl def">(</span><span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">30</span><span class="hl def">)</span>
    <span class="hl def">)</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-2-1.png" alt="plot of chunk unnamed-chunk-2" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">################################################################################################################################################################################################################################################################################################################################################################################################################</span>


<span class="hl com">#  AD vs AsymAD </span>
<span class="hl com"># Open the selected pathways where the Male and Female percentages are combined together to get AD and AsymAD specific pathways</span>
<span class="hl com"># Further, UP (+) and DOWN (-) pathways are merged together in a single column and the data is separated as a comparison for AD vs AsymAD pathways</span>
<span class="hl com">#data_KEGG_ADvAsymAD &lt;- read.csv(&quot;AD_AsymAD_comarative_pathway_results_final_111324.csv&quot;, sep =&quot;,&quot;, header = TRUE, stringsAsFactors = FALSE)</span>

<span class="hl def">data_KEGG_ADvAsymAD</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">read.csv</span><span class="hl def">(</span><span class="hl sng">&quot;AD_AsymAD_comarative_pathway_results_final_050525.csv&quot;</span><span class="hl def">,</span> <span class="hl kwc">sep</span> <span class="hl def">=</span><span class="hl sng">&quot;,&quot;</span><span class="hl def">,</span> <span class="hl kwc">header</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span> <span class="hl kwc">stringsAsFactors</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">)</span>

<span class="hl kwd">head</span><span class="hl def">(data_KEGG_ADvAsymAD,</span> <span class="hl num">5</span><span class="hl def">)</span>
</pre></div>
<div class="output"><pre class="knitr r">##   Sno Pathway_ID                      Pathway_name_initial
## 1   1   hsa00010             Glycolysis / Gluconeogenesis 
## 2   3   hsa00030                Pentose phosphate pathway 
## 3   4   hsa00040 Pentose and glucuronate interconversions 
## 4   5   hsa00051          Fructose and mannose metabolism 
## 5   6   hsa00052                     Galactose metabolism 
##                                  Pathway_name        AD     AsymAD Direction
## 1             Glycolysis / Gluconeogenesis  + 0.1363032 0.12846685        UP
## 2                Pentose phosphate pathway  + 0.1454455 0.08444169        UP
## 3 Pentose and glucuronate interconversions  + 0.1688830 0.07433756        UP
## 4          Fructose and mannose metabolism  + 0.1710439 0.09702031        UP
## 5                     Galactose metabolism  + 0.2157580 0.09949479        UP
##    Difference
## 1 0.007836339
## 2 0.061003784
## 3 0.094545418
## 4 0.074023572
## 5 0.116263185
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl com"># Plot the data as a comparison for AD vs AsymAD pathways</span>

<span class="hl kwd">ggplot</span><span class="hl def">(data_KEGG_ADvAsymAD,</span> <span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">x</span> <span class="hl def">= AsymAD,</span> <span class="hl kwc">y</span> <span class="hl def">= AD,</span> <span class="hl kwc">size</span> <span class="hl def">= Difference))</span> <span class="hl opt">+</span>
  <span class="hl kwd">geom_point</span><span class="hl def">(</span><span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">size</span> <span class="hl def">= Gene_ratio),</span> <span class="hl kwc">alpha</span><span class="hl def">=</span><span class="hl num">0.6</span><span class="hl def">,</span> <span class="hl kwc">stroke</span> <span class="hl def">=</span> <span class="hl num">0</span><span class="hl def">)</span> <span class="hl opt">+</span>
    <span class="hl kwd">theme_bw</span><span class="hl def">()</span> <span class="hl opt">+</span>
    <span class="hl com"># Conditionally label points where AD or AsymAD values are greater than 0.05</span>
    <span class="hl kwd">geom_text_repel</span><span class="hl def">(</span>
        <span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">label</span> <span class="hl def">=</span> <span class="hl kwd">ifelse</span><span class="hl def">((AD</span> <span class="hl opt">&gt;</span> <span class="hl num">0.10</span> <span class="hl opt">&amp;</span> <span class="hl def">AsymAD</span> <span class="hl opt">&lt;</span> <span class="hl num">0.05</span><span class="hl def">)</span> <span class="hl opt">|</span> <span class="hl def">(AsymAD</span> <span class="hl opt">&gt;</span> <span class="hl num">0.10</span> <span class="hl opt">&amp;</span> <span class="hl def">AD</span> <span class="hl opt">&lt;</span> <span class="hl num">0.05</span><span class="hl def">)</span> <span class="hl opt">|</span> <span class="hl def">(AD</span> <span class="hl opt">&gt;</span> <span class="hl num">0.10</span> <span class="hl opt">&amp;</span> <span class="hl def">AsymAD</span> <span class="hl opt">&gt;</span> <span class="hl num">0.10</span><span class="hl def">), Pathway_name,</span> <span class="hl sng">&quot;&quot;</span><span class="hl def">)),</span>
        <span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">6</span><span class="hl def">,</span> <span class="hl kwc">max.overlaps</span> <span class="hl def">=</span> <span class="hl num">Inf</span>
    <span class="hl def">)</span> <span class="hl opt">+</span>
    <span class="hl com">#scale_color_manual(values = custom_colors) +</span>
    <span class="hl com">#labs(color = &quot;Biodomain&quot;) +</span>
    <span class="hl com"># Add x = y line</span>
    <span class="hl kwd">geom_abline</span><span class="hl def">(</span><span class="hl kwc">slope</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">,</span> <span class="hl kwc">intercept</span> <span class="hl def">=</span> <span class="hl num">0</span><span class="hl def">,</span> <span class="hl kwc">linetype</span> <span class="hl def">=</span> <span class="hl sng">&quot;dashed&quot;</span><span class="hl def">,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;gray&quot;</span><span class="hl def">,</span> <span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">0.5</span><span class="hl def">)</span> <span class="hl opt">+</span>
    <span class="hl com"># Add horizontal and vertical dashed lines</span>
    <span class="hl kwd">geom_hline</span><span class="hl def">(</span><span class="hl kwc">yintercept</span> <span class="hl def">=</span> <span class="hl num">0.05</span><span class="hl def">,</span> <span class="hl kwc">linetype</span> <span class="hl def">=</span> <span class="hl sng">&quot;dashed&quot;</span><span class="hl def">,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;gray&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>
    <span class="hl kwd">geom_hline</span><span class="hl def">(</span><span class="hl kwc">yintercept</span> <span class="hl def">=</span> <span class="hl num">0.10</span><span class="hl def">,</span> <span class="hl kwc">linetype</span> <span class="hl def">=</span> <span class="hl sng">&quot;dashed&quot;</span><span class="hl def">,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;red&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>
    <span class="hl kwd">geom_vline</span><span class="hl def">(</span><span class="hl kwc">xintercept</span> <span class="hl def">=</span> <span class="hl num">0.05</span><span class="hl def">,</span> <span class="hl kwc">linetype</span> <span class="hl def">=</span> <span class="hl sng">&quot;dashed&quot;</span><span class="hl def">,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;gray&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>
    <span class="hl kwd">geom_vline</span><span class="hl def">(</span><span class="hl kwc">xintercept</span> <span class="hl def">=</span> <span class="hl num">0.10</span><span class="hl def">,</span> <span class="hl kwc">linetype</span> <span class="hl def">=</span> <span class="hl sng">&quot;dashed&quot;</span><span class="hl def">,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;red&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>
    <span class="hl kwd">labs</span><span class="hl def">(</span><span class="hl kwc">title</span> <span class="hl def">=</span> <span class="hl sng">&quot;Pathways Comparisons for AD vs AsymAD&quot;</span><span class="hl def">,</span> <span class="hl kwc">x</span> <span class="hl def">=</span> <span class="hl sng">&quot;AsymAD (% of subjects)&quot;</span><span class="hl def">,</span> <span class="hl kwc">y</span> <span class="hl def">=</span> <span class="hl sng">&quot;AD (% of subjects)&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>
    <span class="hl kwd">theme</span><span class="hl def">(</span>
    <span class="hl kwc">text</span> <span class="hl def">=</span> <span class="hl kwd">element_text</span><span class="hl def">(</span><span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">30</span><span class="hl def">),</span>
    <span class="hl kwc">legend.position</span> <span class="hl def">=</span> <span class="hl sng">&quot;right&quot;</span><span class="hl def">,</span>
    <span class="hl kwc">axis.text</span> <span class="hl def">=</span> <span class="hl kwd">element_text</span><span class="hl def">(</span><span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">30</span><span class="hl def">),</span>
    <span class="hl kwc">axis.title</span> <span class="hl def">=</span> <span class="hl kwd">element_text</span><span class="hl def">(</span><span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">30</span><span class="hl def">)</span>
  <span class="hl def">)</span>
</pre></div>
<div class="error"><pre class="knitr r">## Error in `geom_point()`:
## ! Problem while computing aesthetics.
## ℹ Error occurred in the 1st layer.
## Caused by error:
## ! object 'Gene_ratio' not found
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl com">########################################################################################################################################################################################################</span>

<span class="hl kwd">ggplot</span><span class="hl def">(data_KEGG_ADvAsymAD,</span> <span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">x</span> <span class="hl def">= AsymAD,</span> <span class="hl kwc">y</span> <span class="hl def">= AD,</span> <span class="hl kwc">size</span> <span class="hl def">= Difference))</span> <span class="hl opt">+</span>
    <span class="hl kwd">geom_point</span><span class="hl def">(</span><span class="hl kwc">alpha</span> <span class="hl def">=</span> <span class="hl num">0.6</span><span class="hl def">,</span> <span class="hl kwc">stroke</span> <span class="hl def">=</span> <span class="hl num">0</span><span class="hl def">)</span> <span class="hl opt">+</span>  <span class="hl com"># Points with transparency</span>
    <span class="hl kwd">theme_bw</span><span class="hl def">()</span> <span class="hl opt">+</span>

    <span class="hl com"># Conditionally label points where |AD - AsymAD| &gt;= 0.05</span>
    <span class="hl kwd">geom_text_repel</span><span class="hl def">(</span>
        <span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">label</span> <span class="hl def">=</span> <span class="hl kwd">ifelse</span><span class="hl def">((</span><span class="hl kwd">abs</span><span class="hl def">(AD</span> <span class="hl opt">-</span> <span class="hl def">AsymAD)</span> <span class="hl opt">&gt;=</span> <span class="hl num">0.10</span><span class="hl def">)</span> <span class="hl opt">|</span> <span class="hl def">(AsymAD</span> <span class="hl opt">&gt;</span> <span class="hl num">0.20</span><span class="hl def">)</span> <span class="hl opt">|</span> <span class="hl def">(AD</span> <span class="hl opt">&gt;</span> <span class="hl num">0.15</span> <span class="hl opt">&amp;</span> <span class="hl def">AsymAD</span> <span class="hl opt">&gt;</span> <span class="hl num">0.15</span><span class="hl def">), Pathway_name,</span> <span class="hl sng">&quot;&quot;</span><span class="hl def">)),</span>
        <span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">6</span>
    <span class="hl def">)</span> <span class="hl opt">+</span>

    <span class="hl com">#scale_color_manual(values = custom_colors) +</span>
    <span class="hl com">#labs(color = &quot;Biodomain&quot;) +</span>

    <span class="hl com"># Add x = y reference line</span>
    <span class="hl kwd">geom_abline</span><span class="hl def">(</span><span class="hl kwc">slope</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">,</span> <span class="hl kwc">intercept</span> <span class="hl def">=</span> <span class="hl num">0</span><span class="hl def">,</span> <span class="hl kwc">linetype</span> <span class="hl def">=</span> <span class="hl sng">&quot;dashed&quot;</span><span class="hl def">,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;gray&quot;</span><span class="hl def">,</span> <span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">0.5</span><span class="hl def">)</span> <span class="hl opt">+</span>

    <span class="hl com"># Add horizontal and vertical threshold lines</span>
    <span class="hl kwd">geom_hline</span><span class="hl def">(</span><span class="hl kwc">yintercept</span> <span class="hl def">=</span> <span class="hl num">0.05</span><span class="hl def">,</span> <span class="hl kwc">linetype</span> <span class="hl def">=</span> <span class="hl sng">&quot;dashed&quot;</span><span class="hl def">,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;gray&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>
    <span class="hl kwd">geom_hline</span><span class="hl def">(</span><span class="hl kwc">yintercept</span> <span class="hl def">=</span> <span class="hl num">0.10</span><span class="hl def">,</span> <span class="hl kwc">linetype</span> <span class="hl def">=</span> <span class="hl sng">&quot;dashed&quot;</span><span class="hl def">,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;red&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>
    <span class="hl kwd">geom_vline</span><span class="hl def">(</span><span class="hl kwc">xintercept</span> <span class="hl def">=</span> <span class="hl num">0.05</span><span class="hl def">,</span> <span class="hl kwc">linetype</span> <span class="hl def">=</span> <span class="hl sng">&quot;dashed&quot;</span><span class="hl def">,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;gray&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>
    <span class="hl kwd">geom_vline</span><span class="hl def">(</span><span class="hl kwc">xintercept</span> <span class="hl def">=</span> <span class="hl num">0.10</span><span class="hl def">,</span> <span class="hl kwc">linetype</span> <span class="hl def">=</span> <span class="hl sng">&quot;dashed&quot;</span><span class="hl def">,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;red&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>

    <span class="hl com"># Title and axis labels</span>
    <span class="hl kwd">labs</span><span class="hl def">(</span><span class="hl kwc">title</span> <span class="hl def">=</span> <span class="hl sng">&quot;Pathway Comparisons for AD vs AsymAD&quot;</span><span class="hl def">,</span>
         <span class="hl kwc">x</span> <span class="hl def">=</span> <span class="hl sng">&quot;AsymAD (% of subjects)&quot;</span><span class="hl def">,</span>
         <span class="hl kwc">y</span> <span class="hl def">=</span> <span class="hl sng">&quot;AD (% of subjects)&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>

    <span class="hl com"># Theme customization</span>
    <span class="hl kwd">theme</span><span class="hl def">(</span>
        <span class="hl kwc">text</span> <span class="hl def">=</span> <span class="hl kwd">element_text</span><span class="hl def">(</span><span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">30</span><span class="hl def">),</span>
        <span class="hl kwc">legend.position</span> <span class="hl def">=</span> <span class="hl sng">&quot;right&quot;</span><span class="hl def">,</span>
        <span class="hl kwc">axis.text</span> <span class="hl def">=</span> <span class="hl kwd">element_text</span><span class="hl def">(</span><span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">30</span><span class="hl def">),</span>
        <span class="hl kwc">axis.title</span> <span class="hl def">=</span> <span class="hl kwd">element_text</span><span class="hl def">(</span><span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">30</span><span class="hl def">)</span>
    <span class="hl def">)</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-2-2.png" alt="plot of chunk unnamed-chunk-2" class="plot" /></div></div>

<p>Graph-based Clustering using Pathway Specific Z score Distributions Results (AD male)</p>

#<!--begin.rcode fig.width=25, fig.height=15
library(reshape2)
library(dplyr)
library(igraph)
library(tidyr)
library(DT)
library(pheatmap)
library(ggraph)
library(ggplot2)
library(ggrepel)
library(tidygraph)

## AD_male

# Open the pathway-specifc Z score distribution file
sig_pathway_AD_male <- read.delim("/Users/poddea/Desktop/ROSMAP_data_100623/RNASeq_Harmonization_Study/Pathway_specific_analysis/sig_pathway_AD_male.txt", header = TRUE, sep = "\t");

# Ensure column names to` match `IDs`
sig_pathway_AD_male <- sig_pathway_AD_male %>% rename_with(~ gsub("X", "", .), starts_with("X"))

# Open the KEGG pathway annotation file
KEGG_ID <- read.delim("KEGG_IDs.txt", header = TRUE, sep = "\t");

head(KEGG_ID)

# Merge the KEGG annotation file to the Z score distribution file to get Pathway_name column
sig_pathway_AD_male<- cbind(KEGG_ID, sig_pathway_AD_male)

# Remove the other columns from the Z score distribution file and keep Pathwa_name and Sample_IDs columns only
sig_pathway_AD_male <- sig_pathway_AD_male[, !(colnames(sig_pathway_AD_male) %in% c("Pathway_ID", "Pathway_ID", "UP_Count", "DOWN_Count"))]

dim(sig_pathway_AD_male)

# Print the table using DT
datatable(sig_pathway_AD_male, options = list(pageLength = 5, autoWidth = TRUE))

# Create a data frame with only positive values
sig_pathway_AD_male_pos <- sig_pathway_AD_male
sig_pathway_AD_male_pos[sig_pathway_AD_male_pos <= 0] <- NA  # Replace all non-positive values with NA

#head(sig_pathway_AD_male_pos)

# Create a data frame with only negative values
sig_pathway_AD_male_neg <- sig_pathway_AD_male

# Replace all non-negative values with NA in the numeric columns only
sig_pathway_AD_male_neg[, -1] <- lapply(sig_pathway_AD_male_neg[, -1], function(x) {
  x[x >= 0] <- NA
  return(x)
})

#head(sig_pathway_AD_male_neg)

#########################################################################################################################################################

# tranfer the data matrix into two column interaction data
sig_pathway_AD_male_pos_edge <- melt(sig_pathway_AD_male_pos)

dim(sig_pathway_AD_male_pos_edge)

# Remove the edges where value (Z score) is "NA"
sig_pathway_AD_male_pos_edge <- filter(sig_pathway_AD_male_pos_edge, !is.na(value))

dim(sig_pathway_AD_male_pos_edge)

# Write the data frame to a file
write.table(sig_pathway_AD_male_pos_edge, file = "sig_pathway_AD_male_pos_edge.txt", row.names = FALSE, col.names = TRUE, sep = "\t")



# create igraph S3 object
net_pos_AD_male <- graph.data.frame(sig_pathway_AD_male_pos_edge, directed = FALSE)

# Get unique vertices from Pathway_name and variable
term_name_vertices <- unique(sig_pathway_AD_male_pos_edge$Pathway_name)
variable_vertices <- unique(sig_pathway_AD_male_pos_edge$variable)

# Assign shape "square" to term_name vertices and "circle" to variable vertices
vertex_shapes <- rep("circle", vcount(net_pos_AD_male))  # Initialize all vertices as circles
vertex_shapes[V(net_pos_AD_male)$name %in% term_name_vertices] <- "square"  # Assign square shape to Var1 vertices

# store original margins
orig_mar <- par()$mar

# set new margins to limit whitespace in plot
par(mar=rep(.1, 4))

# plot the igraph S3 object
# We can use various layout algorithms in igraph to generate the layout, such as Fruchterman-Reingold layout (layout_with_fr), Kamada-Kawai layout (layout_with_kk), or circular layout (layout_in_circle)
plot(net_pos_AD_male,  layout = layout_with_fr(net_pos_AD_male), edge.width = E(net_pos_AD_male)$value, vertex.label.cex = .5, vertex.size = 3, vertex.shape = vertex_shapes)

# Export your igraph object as an edge list
write.graph(net_pos_AD_male, file = "net_pos_AD_male.graphml", format = "graphml")



# community detection based on edge betweenness (Newman-Girvan)
#ceb_pos_AD_male <- cluster_edge_betweenness(net_pos_AD_male, weights = E(net_pos_AD_male)$value)

# community detection based on Infomap Algorithm
#ceb_pos_AD_male <- cluster_infomap(net_pos_AD_male, e.weights = E(net_pos_AD_male)$value)

# community detection based on Louvain Clustering (Modularity Optimization)
ceb_pos_AD_male <- cluster_louvain(net_pos_AD_male, weights = E(net_pos_AD_male)$value)

class(ceb_pos_AD_male)

# plot the igraph S3 objec twith sub-cluster
plot(ceb_pos_AD_male, net_pos_AD_male, layout = layout_with_fr(net_pos_AD_male), edge.width = E(net_pos_AD_male)$value, vertex.label.cex = .5, vertex.size = 3, vertex.shape = vertex_shapes)


# community membership for each node
par(mar=orig_mar)

dendPlot(ceb_pos_AD_male, mode="hclust", cex=.5)

# Get the community membership data
community_membership_pos_AD_male <- membership(ceb_pos_AD_male)

# Convert it to a data frame with two columns: Node and Community
community_df_pos_AD_male <- data.frame(Node = names(community_membership_pos_AD_male), Community = as.integer(community_membership_pos_AD_male))

# Count the number of community detected
unique(community_df_pos_AD_male$Community)

# Write the data frame to a file
write.table(community_df_pos_AD_male, file = "community_membership_KEGG_pos_AD_male.txt", row.names = FALSE, col.names = TRUE, sep = "\t")

##################################################################################################################################################
# Make individual community as example

# Subset the graph for Community 5
community_5_nodes <- V(net_pos_AD_male)[community_df_pos_AD_male$Community == 5]  # Select nodes from Community 5

# Induce a subgraph with only Community 5 nodes
subgraph_community_5 <- induced_subgraph(net_pos_AD_male, community_5_nodes)

# Assign "circle" shape to all vertices
V(subgraph_community_5)$shape <- "circle"

# Set "square" shape for vertices that match the term_name_vertices
V(subgraph_community_5)$shape[V(subgraph_community_5)$name %in% term_name_vertices] <- "square"

# Assign "blue" color to all vertices (default for circles)
V(subgraph_community_5)$color <- "lightblue"

# Set "red" color for vertices that are squares
V(subgraph_community_5)$color[V(subgraph_community_5)$shape == "square"] <- "khaki1"

# set new margins to limit whitespace in plot
#par(mar=rep(.1, 4))

# Plot the subgraph for Community 5
plot(subgraph_community_5, layout = layout_with_fr(subgraph_community_5), edge.width = E(net_pos_AD_male)$value, vertex.size = 10, vertex.label.cex = 1, vertex.label.color = "black", main = "Community 5 Subgraph")

# Convert igraph object to tbl_graph (a tidygraph object)
#graph_tbl <- as_tbl_graph(subgraph_community_5)

# Create the ggraph plot
#ggraph(graph_tbl, layout = 'fr') +  # Fruchterman-Reingold layout
  #geom_edge_link(aes(width = E(subgraph_community_5)$value), alpha = 0.5) +  # Edges
  #geom_node_point(aes(color = V(subgraph_community_5)$color, shape = V(subgraph_community_5)$shape), size = 10) +  # Vertices
  #geom_node_text(aes(label = V(subgraph_community_5)$name), repel = TRUE, size = 10) +  # Non-overlapping labels
  #theme_void() +  # Minimalist theme
  #scale_shape_manual(values = c("circle" = 16, "square" = 15)) +  # Map shapes
  #scale_color_manual(values = c("lightblue", "khaki1"))  # Map colors


##################################################################################################################################################
# Make heatmap for the comunitties

# Select the Pathway_name and variable entities from the sig_pathway_AD_male_pos_edge file
term_name <- sig_pathway_AD_male_pos_edge$Pathway_name
variable <- sig_pathway_AD_male_pos_edge$variable

# Separate term_name and variable entities based on their community membership and sort the Community column in ascending order
term_name_nodes <- community_df_pos_AD_male[community_df_pos_AD_male$Node %in% term_name, ]
term_name_nodes <- term_name_nodes[order(term_name_nodes$Community), ]


variable_nodes <- community_df_pos_AD_male[community_df_pos_AD_male$Node %in% variable, ]
variable_nodes <- variable_nodes[order(variable_nodes$Community), ]


head(term_name_nodes)

head(variable_nodes)

# Prepare the row and col entries for a heatmap based on term_name and variable entities
row_names <- unique(c(term_name_nodes$Node))

col_names <- unique(c(variable_nodes$Node))

# Create a matrix filled with zeros
matrix_data_AD_male_pos <- matrix(0, nrow = length(row_names), ncol = length(col_names), dimnames = list(row_names, col_names))

# Populate the matrix based on the information in term_name_nodes and variable_nodes
for (i in 1:nrow(term_name_nodes)) {
          matrix_data_AD_male_pos[term_name_nodes[i, "Node"], variable_nodes[variable_nodes$Community == term_name_nodes[i, "Community"], "Node"]] <- variable_nodes[variable_nodes$Community == term_name_nodes[i, "Community"], "Community"]
}


write.table(matrix_data_AD_male_pos,"KEGG_matrix_data_pos_AD_male.txt",sep="\t",quote=F)

# Create heatmap with matrix
matrix_data_AD_male_pos <- as.matrix(matrix_data_AD_male_pos)

# Adjust the margins to make room for the X-axis labels
par(mar = c(5, 5, 4, 2) + 0.1)

heatmap(matrix_data_AD_male_pos, Rowv = NA, Colv = NA, cex.axis = 0.5, main = "Communities identified by Graph Based Clustering for KEGG Comparative Pathway Enrichment Analysis (Male UP Pathways)")


#########################################################################################################################################################

# tranfer the data matrix into two column interaction data
sig_pathway_AD_male_neg_edge <- melt(sig_pathway_AD_male_neg)

dim(sig_pathway_AD_male_neg_edge)

# Remove the edges where value (Z score) is "NA"
sig_pathway_AD_male_neg_edge <- filter(sig_pathway_AD_male_neg_edge, !is.na(value))

dim(sig_pathway_AD_male_neg_edge)

# Convert the negative values in the 'value' column to absolute value for network construction
sig_pathway_AD_male_neg_edge$value <- abs(sig_pathway_AD_male_neg_edge$value)

# create igraph S3 object
net_neg_AD_male <- graph.data.frame(sig_pathway_AD_male_neg_edge, directed = FALSE)

# Get unique vertices from Pathway_name and variable
term_name_vertices <- unique(sig_pathway_AD_male_neg_edge$Pathway_name)
variable_vertices <- unique(sig_pathway_AD_male_neg_edge$variable)

# Assign shape "square" to term_name vertices and "circle" to variable vertices
vertex_shapes <- rep("circle", vcount(net_neg_AD_male))  # Initialize all vertices as circles
vertex_shapes[V(net_neg_AD_male)$name %in% term_name_vertices] <- "square"  # Assign square shape to Var1 vertices

# store original margins
orig_mar <- par()$mar

# set new margins to limit whitespace in plot
par(mar=rep(.1, 4))

# plot the igraph S3 object
# We can use various layout algorithms in igraph to generate the layout, such as Fruchterman-Reingold layout (layout_with_fr), Kamada-Kawai layout (layout_with_kk), or circular layout (layout_in_circle)
plot(net_neg_AD_male,  layout = layout_with_fr(net_neg_AD_male), edge.width = E(net_neg_AD_male)$value, vertex.label.cex = .5, vertex.size = 3, vertex.shape = vertex_shapes)

# community detection based on Weighted Newman-Girvan (Edge Betweenness)
#ceb_neg_AD_male <- cluster_edge_betweenness(net_neg_AD_male, weights = E(net_neg_AD_male)$value)

# community detection based on Louvain Clustering (Modularity Optimization)
ceb_neg_AD_male <- cluster_louvain(net_neg_AD_male, weights = E(net_neg_AD_male)$value)

# community detection based on Infomap Algorithm
#ceb_neg_AD_male <- cluster_infomap(net_neg_AD_male, e.weights = E(net_neg_AD_male)$value)

# community detection based on Walktrap Algorithm
#ceb_neg_AD_male <- cluster_walktrap(net_neg_AD_male, weights = E(net_neg_AD_male)$value)


class(ceb_neg_AD_male)


# plot the igraph S3 objec twith sub-cluster
plot(ceb_neg_AD_male, net_neg_AD_male, layout = layout_with_fr(net_neg_AD_male), edge.width = E(net_neg_AD_male)$value, vertex.label.cex = .5, vertex.size = 3, vertex.shape = vertex_shapes)

# community membership for each node
par(mar=orig_mar)

dendPlot(ceb_neg_AD_male, mode="hclust", cex=.5)

# Get the community membership data
community_membership_neg_AD_male <- membership(ceb_neg_AD_male)

# Convert it to a data frame with two columns: Node and Community
community_df_neg_AD_male <- data.frame(Node = names(community_membership_neg_AD_male), Community = as.integer(community_membership_neg_AD_male))

# Count the number of community detected
unique(community_df_neg_AD_male$Community)

# Write the data frame to a file
write.table(community_df_neg_AD_male, file = "community_membership_KEGG_neg_AD_male.txt", row.names = FALSE, col.names = TRUE, sep = "\t")

# Select the Pathway_name and variable entities from the sig_pathway_AD_male_neg_edge file
term_name <- sig_pathway_AD_male_neg_edge$Pathway_name
variable <- sig_pathway_AD_male_neg_edge$variable

# Separate term_name and variable entities based on their community membership and sort the Community column in ascending order
term_name_nodes <- community_df_neg_AD_male[community_df_neg_AD_male$Node %in% term_name, ]
term_name_nodes <- term_name_nodes[order(term_name_nodes$Community), ]


variable_nodes <- community_df_neg_AD_male[community_df_neg_AD_male$Node %in% variable, ]
variable_nodes <- variable_nodes[order(variable_nodes$Community), ]


head(term_name_nodes)

head(variable_nodes)

# Prepare the row and col entries for a heatmap based on term_name and variable entities
row_names <- unique(c(term_name_nodes$Node))

col_names <- unique(c(variable_nodes$Node))

# Create a matrix filled with zeros
matrix_data_AD_male_neg <- matrix(0, nrow = length(row_names), ncol = length(col_names), dimnames = list(row_names, col_names))

# Populate the matrix based on the information in term_name_nodes and variable_nodes
for (i in 1:nrow(term_name_nodes)) {
          matrix_data_AD_male_neg[term_name_nodes[i, "Node"], variable_nodes[variable_nodes$Community == term_name_nodes[i, "Community"], "Node"]] <- variable_nodes[variable_nodes$Community == term_name_nodes[i, "Community"], "Community"]
}


write.table(matrix_data_AD_male_neg,"KEGG_matrix_data_neg_AD_male.txt",sep="\t",quote=F)

# Create heatmap with matrix
matrix_data_AD_male_neg <- as.matrix(matrix_data_AD_male_neg)

# Adjust the margins to make room for the X-axis labels
par(mar = c(5, 5, 4, 2) + 0.1)

heatmap(matrix_data_AD_male_neg, Rowv = NA, Colv = NA, cex.axis = 0.5, main = "Communities identified by Graph Based Clustering for KEGG Comparative Pathway Enrichment Analysis (Male DOWN Pathways)")



end.rcode-->

<p>Graph-based Clustering using Pathway Specific Z score Distributions Results (AD female)</p>

#<!--begin.rcode fig.width=25, fig.height=15
library(reshape2)
library(dplyr)
library(igraph)
library(tidyr)
library(DT)

## AD_male

# Open the pathway-specifc Z score distribution file
sig_pathway_AD_female <- read.delim("/Users/poddea/Desktop/ROSMAP_data_100623/RNASeq_Harmonization_Study/Pathway_specific_analysis/sig_pathway_AD_female.txt", header = TRUE, sep = "\t");

# Ensure column names to` match `IDs`
sig_pathway_AD_female <- sig_pathway_AD_female %>% rename_with(~ gsub("X", "", .), starts_with("X"))

# Open the KEGG pathway annotation file
KEGG_ID <- read.delim("KEGG_IDs.txt", header = TRUE, sep = "\t");

head(KEGG_ID)

# Merge the KEGG annotation file to the Z score distribution file to get Pathway_name column
sig_pathway_AD_female<- cbind(KEGG_ID, sig_pathway_AD_female)

# Remove the other columns from the Z score distribution file and keep Pathwa_name and Sample_IDs columns only
sig_pathway_AD_female <- sig_pathway_AD_female[, !(colnames(sig_pathway_AD_female) %in% c("Pathway_ID", "Pathway_ID", "UP_Count", "DOWN_Count"))]

dim(sig_pathway_AD_female)

# Print the table using DT
datatable(sig_pathway_AD_female, options = list(pageLength = 5, autoWidth = TRUE))

# Create a data frame with only positive values
sig_pathway_AD_female_pos <- sig_pathway_AD_female
sig_pathway_AD_female_pos[sig_pathway_AD_female_pos <= 0] <- NA  # Replace all non-positive values with NA

#head(sig_pathway_AD_female_pos)

# Create a data frame with only negative values
sig_pathway_AD_female_neg <- sig_pathway_AD_female

# Replace all non-negative values with NA in the numeric columns only
sig_pathway_AD_female_neg[, -1] <- lapply(sig_pathway_AD_female_neg[, -1], function(x) {
  x[x >= 0] <- NA
  return(x)
})

#head(sig_pathway_AD_female_neg)

#########################################################################################################################################################

# tranfer the data matrix into two column interaction data
sig_pathway_AD_female_pos_edge <- melt(sig_pathway_AD_female_pos)

dim(sig_pathway_AD_female_pos_edge)

# select the edges where value (Z score) is "NA"
sig_pathway_AD_female_pos_edge <- filter(sig_pathway_AD_female_pos_edge, !is.na(value))

dim(sig_pathway_AD_female_pos_edge)

write.table(sig_pathway_AD_female_pos_edge, file = "sig_pathway_AD_female_pos_edge.txt", row.names = FALSE, col.names = TRUE, sep = "\t")

# create igraph S3 object
net_pos_AD_female <- graph.data.frame(sig_pathway_AD_female_pos_edge, directed = FALSE)

# Get unique vertices from Pathway_name and variable
term_name_vertices <- unique(sig_pathway_AD_female_pos_edge$Pathway_name)
variable_vertices <- unique(sig_pathway_AD_female_pos_edge$variable)

# Assign shape "square" to term_name vertices and "circle" to variable vertices
vertex_shapes <- rep("circle", vcount(net_pos_AD_female))  # Initialize all vertices as circles
vertex_shapes[V(net_pos_AD_female)$name %in% term_name_vertices] <- "square"  # Assign square shape to Var1 vertices

# store original margins
orig_mar <- par()$mar

# set new margins to limit whitespace in plot
par(mar=rep(.1, 4))

# plot the igraph S3 object
# We can use various layout algorithms in igraph to generate the layout, such as Fruchterman-Reingold layout (layout_with_fr), Kamada-Kawai layout (layout_with_kk), or circular layout (layout_in_circle)
plot(net_pos_AD_female,  layout = layout_with_fr(net_pos_AD_female), edge.width = E(net_pos_AD_female)$value, vertex.label.cex = .5, vertex.size = 3, vertex.shape = vertex_shapes)

# community detection based on edge betweenness (Newman-Girvan)
#ceb_pos_AD_female <- cluster_edge_betweenness(net_pos_AD_female)

# community detection based on Infomap Algorithm
#ceb_pos_AD_female <- cluster_infomap(net_pos_AD_female, e.weights = E(net_pos_AD_female)$value)

# community detection based on Louvain Clustering (Modularity Optimization)
ceb_pos_AD_female <- cluster_louvain(net_pos_AD_female, weights = E(net_pos_AD_female)$value)

class(ceb_pos_AD_female)

# plot the igraph S3 objec twith sub-cluster
plot(ceb_pos_AD_female, net_pos_AD_female, layout = layout_with_fr(net_pos_AD_female), edge.width = E(net_pos_AD_female)$value, vertex.label.cex = .5, vertex.size = 3, vertex.shape = vertex_shapes)

# community membership for each node
par(mar=orig_mar)

dendPlot(ceb_pos_AD_female, mode="hclust", cex=.5)

# Get the community membership data
community_membership_pos_AD_female <- membership(ceb_pos_AD_female)

# Convert it to a data frame with two columns: Node and Community
community_df_pos_AD_female <- data.frame(Node = names(community_membership_pos_AD_female), Community = as.integer(community_membership_pos_AD_female))

# Count the number of community detected
unique(community_df_pos_AD_female$Community)

# Write the data frame to a file
write.table(community_df_pos_AD_female, file = "community_membership_KEGG_pos_AD_female.txt", row.names = FALSE, col.names = TRUE, sep = "\t")

# Select the Pathway_name and variable entities from the sig_pathway_AD_female_pos_edge file
term_name <- sig_pathway_AD_female_pos_edge$Pathway_name
variable <- sig_pathway_AD_female_pos_edge$variable

# Separate term_name and variable entities based on their community membership and sort the Community column in ascending order
term_name_nodes <- community_df_pos_AD_female[community_df_pos_AD_female$Node %in% term_name, ]
term_name_nodes <- term_name_nodes[order(term_name_nodes$Community), ]


variable_nodes <- community_df_pos_AD_female[community_df_pos_AD_female$Node %in% variable, ]
variable_nodes <- variable_nodes[order(variable_nodes$Community), ]


head(term_name_nodes)

head(variable_nodes)

# Prepare the row and col entries for a heatmap based on term_name and variable entities
row_names <- unique(c(term_name_nodes$Node))

col_names <- unique(c(variable_nodes$Node))

# Create a matrix filled with zeros
matrix_data_AD_female_pos <- matrix(0, nrow = length(row_names), ncol = length(col_names), dimnames = list(row_names, col_names))

# Populate the matrix based on the information in term_name_nodes and variable_nodes
for (i in 1:nrow(term_name_nodes)) {
          matrix_data_AD_female_pos[term_name_nodes[i, "Node"], variable_nodes[variable_nodes$Community == term_name_nodes[i, "Community"], "Node"]] <- variable_nodes[variable_nodes$Community == term_name_nodes[i, "Community"], "Community"]
}


write.table(matrix_data_AD_female_pos,"KEGG_matrix_data_pos_AD_female.txt",sep="\t",quote=F)

# Create heatmap with matrix
matrix_data_AD_female_pos <- as.matrix(matrix_data_AD_female_pos)

# Adjust the margins to make room for the X-axis labels
par(mar = c(5, 5, 4, 2) + 0.1)

heatmap(matrix_data_AD_female_pos, Rowv = NA, Colv = NA, cex.axis = 0.5, main = "Communities identified by Graph Based Clustering for KEGG Comparative Pathway Enrichment Analysis (Female UP Pathways)")



#########################################################################################################################################################

# tranfer the data matrix into two column interaction data
sig_pathway_AD_female_neg_edge <- melt(sig_pathway_AD_female_neg)

dim(sig_pathway_AD_female_neg_edge)

# select the edges where value (Z score) is "NA"
sig_pathway_AD_female_neg_edge <- filter(sig_pathway_AD_female_neg_edge, !is.na(value))

dim(sig_pathway_AD_female_neg_edge)

# Convert the negative values in the 'value' column to absolute value for network construction
sig_pathway_AD_female_neg_edge$value <- abs(sig_pathway_AD_female_neg_edge$value)

# create igraph S3 object
net_neg_AD_female <- graph.data.frame(sig_pathway_AD_female_neg_edge, directed = FALSE)

# Get unique vertices from Pathway_name and variable
term_name_vertices <- unique(sig_pathway_AD_female_neg_edge$Pathway_name)
variable_vertices <- unique(sig_pathway_AD_female_neg_edge$variable)

# Assign shape "square" to term_name vertices and "circle" to variable vertices
vertex_shapes <- rep("circle", vcount(net_neg_AD_female))  # Initialize all vertices as circles
vertex_shapes[V(net_neg_AD_female)$name %in% term_name_vertices] <- "square"  # Assign square shape to Var1 vertices

# store original margins
orig_mar <- par()$mar

# set new margins to limit whitespace in plot
par(mar=rep(.1, 4))

# plot the igraph S3 object
# We can use various layout algorithms in igraph to generate the layout, such as Fruchterman-Reingold layout (layout_with_fr), Kamada-Kawai layout (layout_with_kk), or circular layout (layout_in_circle)
plot(net_neg_AD_female,  layout = layout_with_fr(net_neg_AD_female), edge.width = E(net_neg_AD_female)$value, vertex.label.cex = .5, vertex.size = 3, vertex.shape = vertex_shapes)

# community detection based on edge betweenness (Newman-Girvan)
#ceb_neg_AD_female <- cluster_edge_betweenness(net_neg_AD_female)

# community detection based on Infomap Algorithm
#ceb_neg_AD_female <- cluster_infomap(net_neg_AD_female, e.weights = E(net_neg_AD_female)$value)

# community detection based on Louvain Clustering (Modularity Optimization)
ceb_neg_AD_female <- cluster_louvain(net_neg_AD_female, weights = E(net_neg_AD_female)$value)

class(ceb_neg_AD_female)

# plot the igraph S3 objec twith sub-cluster
plot(ceb_neg_AD_female, net_neg_AD_female, layout = layout_with_fr(net_neg_AD_female), edge.width = E(net_neg_AD_female)$value, vertex.label.cex = .5, vertex.size = 3, vertex.shape = vertex_shapes)

# community membership for each node
par(mar=orig_mar)

dendPlot(ceb_neg_AD_female, mode="hclust", cex=.5)

# Get the community membership data
community_membership_neg_AD_female <- membership(ceb_neg_AD_female)

# Convert it to a data frame with two columns: Node and Community
community_df_neg_AD_female <- data.frame(Node = names(community_membership_neg_AD_female), Community = as.integer(community_membership_neg_AD_female))

# Count the number of community detected
unique(community_df_neg_AD_female$Community)

# Write the data frame to a file
write.table(community_df_neg_AD_female, file = "community_membership_KEGG_neg_AD_female.txt", row.names = FALSE, col.names = TRUE, sep = "\t")

# Select the Pathway_name and variable entities from the sig_pathway_AD_female_neg_edge file
term_name <- sig_pathway_AD_female_neg_edge$Pathway_name
variable <- sig_pathway_AD_female_neg_edge$variable

# Separate term_name and variable entities based on their community membership and sort the Community column in ascending order
term_name_nodes <- community_df_neg_AD_female[community_df_neg_AD_female$Node %in% term_name, ]
term_name_nodes <- term_name_nodes[order(term_name_nodes$Community), ]


variable_nodes <- community_df_neg_AD_female[community_df_neg_AD_female$Node %in% variable, ]
variable_nodes <- variable_nodes[order(variable_nodes$Community), ]


head(term_name_nodes)

head(variable_nodes)

# Prepare the row and col entries for a heatmap based on term_name and variable entities
row_names <- unique(c(term_name_nodes$Node))

col_names <- unique(c(variable_nodes$Node))

# Create a matrix filled with zeros
matrix_data_AD_female_neg <- matrix(0, nrow = length(row_names), ncol = length(col_names), dimnames = list(row_names, col_names))

# Populate the matrix based on the information in term_name_nodes and variable_nodes
for (i in 1:nrow(term_name_nodes)) {
          matrix_data_AD_female_neg[term_name_nodes[i, "Node"], variable_nodes[variable_nodes$Community == term_name_nodes[i, "Community"], "Node"]] <- variable_nodes[variable_nodes$Community == term_name_nodes[i, "Community"], "Community"]
}


write.table(matrix_data_AD_female_neg,"KEGG_matrix_data_neg_AD_female.txt",sep="\t",quote=F)

# Create heatmap with matrix
matrix_data_AD_female_neg <- as.matrix(matrix_data_AD_female_neg)

# Adjust the margins to make room for the X-axis labels
par(mar = c(5, 5, 4, 2) + 0.1)

heatmap(matrix_data_AD_female_neg, Rowv = NA, Colv = NA, cex.axis = 0.5, main = "Communities identified by Graph Based Clustering for KEGG Comparative Pathway Enrichment Analysis (Female DOWN Pathways)")


#tiff("matrix_data.tiff", width = 20, height = 15, units = "in", res = 300)  # Increase width, height, and resolution
end.rcode-->

<p>Save individual matrix file as .tiff format</p>

#<!--begin.rcode fig.width=25, fig.height=15

# List of matrix data
matrix_list <- list(matrix_data_AD_male_pos, matrix_data_AD_male_neg, matrix_data_AD_female_pos, matrix_data_AD_female_neg)

# Corresponding file names for each heatmap
file_names <- c("matrix_data_AD_male_pos.tiff", "matrix_data_AD_male_neg.tiff", "matrix_data_AD_female_pos.tiff", "matrix_data_AD_female_neg.tiff")

# Loop through each matrix and save heatmap as a TIFF file
for (i in 1:length(matrix_list)) {
  # Open TIFF device with specific resolution and size
  tiff(file = file_names[i], width = 20, height = 10, units = "in", res = 300)
  
  # Plot the heatmap (you can customize the heatmap function as needed)
  heatmap(matrix_list[[i]], Rowv = NA, Colv = NA, cexRow = 0.5, cexCol = 0.5, main = paste("Heatmap", i))
  
  # Close the device to save the file
  dev.off()
}

end.rcode-->



</body>
</html>
