<html>

<head>
<style type="text/css">
.inline {
  background-color: #f7f7f7;
  border:solid 1px #B0B0B0;
}
.error {
	font-weight: bold;
	color: #FF0000;
}
.warning {
	font-weight: bold;
}
.message {
	font-style: italic;
}
.source, .output, .warning, .error, .message {
	padding: 0 1em;
  border:solid 1px #F7F7F7;
}
.source {
  background-color: #f5f5f5;
}
.left {
  text-align: left;
}
.right {
  text-align: right;
}
.center {
  text-align: center;
}
.hl.num {
  color: #AF0F91;
}
.hl.sng {
  color: #317ECC;
}
.hl.com {
  color: #AD95AF;
  font-style: italic;
}
.hl.opt {
  color: #000000;
}
.hl.def {
  color: #585858;
}
.hl.kwa {
  color: #295F94;
  font-weight: bold;
}
.hl.kwb {
  color: #B05A65;
}
.hl.kwc {
  color: #55aa55;
}
.hl.kwd {
  color: #BC5A65;
  font-weight: bold;
}
</style>
<title>Avijit Podder Date: Aug 29, 2024</title>

<head color="blue">

<h1><font color="blue">Individual Pathway Specific Z score Calculation using TMT Data from the ROSMAP-DLPFC Cohort</font></h1>

</head>

<body>

<div class="chunk" id="unnamed-chunk-1"><div class="rcode"><div class="source"><pre class="knitr r"><span class="hl kwd">library</span><span class="hl def">(jpeg)</span>

<span class="hl com"># Load the .jpeg image</span>
<span class="hl def">img</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">readJPEG</span><span class="hl def">(</span><span class="hl sng">&quot;/Users/poddea/Desktop/ROSMAP_data_100623/Pathway_specific_Zscore.jpg&quot;</span><span class="hl def">)</span>

<span class="hl com"># Display the image</span>
<span class="hl kwd">plot</span><span class="hl def">(</span><span class="hl num">1</span><span class="hl opt">:</span><span class="hl num">2</span><span class="hl def">,</span> <span class="hl kwc">type</span> <span class="hl def">=</span> <span class="hl sng">&quot;n&quot;</span><span class="hl def">,</span> <span class="hl kwc">xlab</span> <span class="hl def">=</span> <span class="hl sng">&quot;&quot;</span><span class="hl def">,</span> <span class="hl kwc">ylab</span> <span class="hl def">=</span> <span class="hl sng">&quot;&quot;</span><span class="hl def">,</span> <span class="hl kwc">axes</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">)</span>
<span class="hl kwd">rasterImage</span><span class="hl def">(img,</span> <span class="hl num">1</span><span class="hl def">,</span> <span class="hl num">1</span><span class="hl def">,</span> <span class="hl num">2</span><span class="hl def">,</span> <span class="hl num">2</span><span class="hl def">)</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-1-1.png" alt="plot of chunk unnamed-chunk-1" class="plot" /></div></div>

<p>Retrieve human KEGG pathways and genes associated with each pathway:</p>

#<!--begin.rcode fig.width=7, fig.height=6
# Tenenbaum D, Maintainer B (2024). KEGGREST: Client-side REST access to the Kyoto Encyclopedia of Genes and Genomes (KEGG). R package version 1.44.1, https://bioconductor.org/packages/KEGGREST.
# Install and load the necessary packages
if (!requireNamespace("BiocManager", quietly = TRUE))
  install.packages("BiocManager")
BiocManager::install(c("KEGGREST", "org.Hs.eg.db", "AnnotationDbi"))

library(KEGGREST)
library(org.Hs.eg.db)
library(AnnotationDbi)
library(dplyr)
library(DT)

# Retrieve human KEGG pathways
human_pathways <- keggList("pathway", "hsa")

# Function to retrieve Entrez gene IDs for a given pathway ID
get_genes_in_pathway <- function(pathway_id) {
  pathway_info <- keggGet(pathway_id)[[1]]
  genes <- pathway_info$GENE
  if (is.null(genes)) {
    return(NULL)
  }
  genes <- genes[seq(1, length(genes), by = 2)]  # Extract Entrez gene IDs
  return(genes)
}

# Retrieve genes for each pathway
pathway_genes <- lapply(names(human_pathways), get_genes_in_pathway)

# Create a data frame
pathway_gene_df <- data.frame(
  pathway_id = rep(names(human_pathways), lengths(pathway_genes)),
  pathway_name = rep(unname(human_pathways), lengths(pathway_genes)),
  entrez_id = unlist(pathway_genes)
)

# Check for any NA or invalid values in entrez_id
pathway_gene_df <- pathway_gene_df %>% filter(!is.na(entrez_id) & entrez_id != "")

# Function to map Entrez IDs to gene symbols
map_entrez_to_symbols <- function(entrez_ids) {
  gene_symbols <- mapIds(org.Hs.eg.db,
                         keys = as.character(entrez_ids),
                         column = "SYMBOL",
                         keytype = "ENTREZID",
                         multiVals = "first")
  return(gene_symbols)
}

# Apply the mapping function
pathway_gene_df$gene_symbol <- map_entrez_to_symbols(pathway_gene_df$entrez_id)

# Check for any NA or invalid values in gene_symbol
pathway_gene_df <- pathway_gene_df %>% filter(!is.na(gene_symbol) & gene_symbol != "")

# Function to map Gene Symbols to ENSEMBL IDs
map_symbols_to_ensembl <- function(gene_symbols) {
  ensembl_ids <- mapIds(org.Hs.eg.db,
                        keys = gene_symbols,
                        column = "ENSEMBL",
                        keytype = "SYMBOL",
                        multiVals = "first")
  return(ensembl_ids)
}

# Apply the mapping function
pathway_gene_df$ensembl_id <- map_symbols_to_ensembl(pathway_gene_df$gene_symbol)

# Inspect the resulting data frame
#head(pathway_gene_df)

# Print the table using DT
datatable(pathway_gene_df, options = list(pageLength = 5, autoWidth = TRUE))

# Remove rows with NA values in ensembl_id
pathway_gene_df <- pathway_gene_df %>% filter(!is.na(ensembl_id))

# Save the data frame to a file if needed
write.table(pathway_gene_df, "KEGG_pathway_gene_info.txt", sep = "\t", quote = FALSE, row.names = FALSE, col.names = TRUE)

# Count the number of unique pathway IDs
num_unique_pathways <- length(unique(pathway_gene_df$pathway_id))

# Print the total number of pathways
print(num_unique_pathways)

# Count the number of unique gene IDs
num_unique_gene <- length(unique(pathway_gene_df$entrez_id))

# Print the total number of gene
print(num_unique_gene)

# Group by pathway_id and pathway_name and count the occurrences of each pathway_id to summarize the gene count for each pathway
pathway_gene_counts <- pathway_gene_df %>%
  group_by(pathway_id, pathway_name) %>%
  summarise(gene_count = n()) %>%
  ungroup()

# View the first few rows of the resulting data frame
head(pathway_gene_counts)

# Write the result to a txt file
 write.table(pathway_gene_counts, "KEGG_pathway_gene_counts_summary.txt",  sep = "\t", quote = FALSE, row.names = FALSE, col.names = TRUE)


end.rcode-->

<p>Create Individual Pathway Files:</p>

#<!--begin.rcode fig.width=7, fig.height=6
# create individual tab-delimited .txt files for each unique pathway_id and write them to a specified directory
# Load necessary package
library(dplyr)

# Define the directory to save the files
output_directory <- "/Users/poddea/Desktop/ROSMAP_data_100623/TMT_proteomics/Round2/Pathway_sprcific_analysis/kegg_hsa_all"  # Change this to your desired directory

# Ensure the output directory exists
if (!dir.exists(output_directory)) {
  dir.create(output_directory)
}

# Get the unique pathway IDs
unique_pathways <- unique(pathway_gene_df$pathway_id)

# Loop over each unique pathway ID
for (pathway in unique_pathways) {
  # Subset the data for the current pathway ID
  pathway_data <- subset(pathway_gene_df, pathway_id == pathway)
  
  # Create a file name for the current pathway ID
  file_name <- paste0(output_directory, "/", pathway, ".txt")
  
  # Write the data to a tab-delimited file
  write.table(pathway_data, file_name, sep = "\t", row.names = FALSE, col.names = TRUE, quote = FALSE)
}

end.rcode-->

<p>Open Protein-wise Zscore for CT, AD and AsymAD Individuals (Male and Female Separately):</p>

<div class="chunk" id="unnamed-chunk-2"><div class="rcode"><div class="source"><pre class="knitr r"><span class="hl kwd">library</span><span class="hl def">(dplyr)</span>
</pre></div>
<div class="message"><pre class="knitr r">## 
## Attaching package: 'dplyr'
</pre></div>
<div class="message"><pre class="knitr r">## The following objects are masked from 'package:stats':
## 
##     filter, lag
</pre></div>
<div class="message"><pre class="knitr r">## The following objects are masked from 'package:base':
## 
##     intersect, setdiff, setequal, union
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl kwd">library</span><span class="hl def">(tidyr)</span>

<span class="hl com">## Open individual Zscore distribution files for CT and AD individuals</span>

<span class="hl def">zscore_CT_male</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">read.delim</span><span class="hl def">(</span><span class="hl sng">&quot;/Users/poddea/Desktop/ROSMAP_data_100623/TMT_proteomics/Round2/z_stat_result_CT_male.txt&quot;</span><span class="hl def">,</span> <span class="hl kwc">header</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span> <span class="hl kwc">sep</span> <span class="hl def">=</span> <span class="hl sng">&quot;\t&quot;</span><span class="hl def">);</span>

<span class="hl com"># Ensure column names to` match `IDs` in `zscore_` files</span>
<span class="hl def">zscore_CT_male</span> <span class="hl kwb">&lt;-</span> <span class="hl def">zscore_CT_male</span> <span class="hl opt">%&gt;%</span> <span class="hl kwd">rename_with</span><span class="hl def">(</span><span class="hl opt">~</span> <span class="hl kwd">gsub</span><span class="hl def">(</span><span class="hl sng">&quot;X&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;&quot;</span><span class="hl def">, .),</span> <span class="hl kwd">starts_with</span><span class="hl def">(</span><span class="hl sng">&quot;X&quot;</span><span class="hl def">))</span>

<span class="hl kwd">dim</span><span class="hl def">(zscore_CT_male)</span>
</pre></div>
<div class="output"><pre class="knitr r">## [1] 7800   47
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl def">zscore_CT_female</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">read.delim</span><span class="hl def">(</span><span class="hl sng">&quot;/Users/poddea/Desktop/ROSMAP_data_100623/TMT_proteomics/Round2/z_stat_result_CT_female.txt&quot;</span><span class="hl def">,</span> <span class="hl kwc">header</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span> <span class="hl kwc">sep</span> <span class="hl def">=</span> <span class="hl sng">&quot;\t&quot;</span><span class="hl def">);</span>

<span class="hl com"># Ensure column names to` match `IDs` in `zscore_` files</span>
<span class="hl def">zscore_CT_female</span> <span class="hl kwb">&lt;-</span> <span class="hl def">zscore_CT_female</span> <span class="hl opt">%&gt;%</span> <span class="hl kwd">rename_with</span><span class="hl def">(</span><span class="hl opt">~</span> <span class="hl kwd">gsub</span><span class="hl def">(</span><span class="hl sng">&quot;X&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;&quot;</span><span class="hl def">, .),</span> <span class="hl kwd">starts_with</span><span class="hl def">(</span><span class="hl sng">&quot;X&quot;</span><span class="hl def">))</span>

<span class="hl kwd">dim</span><span class="hl def">(zscore_CT_female)</span>
</pre></div>
<div class="output"><pre class="knitr r">## [1] 7800   69
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl def">zscore_AD_male</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">read.delim</span><span class="hl def">(</span><span class="hl sng">&quot;/Users/poddea/Desktop/ROSMAP_data_100623/TMT_proteomics/Round2/z_stat_result_AD_male.txt&quot;</span><span class="hl def">,</span> <span class="hl kwc">header</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span> <span class="hl kwc">sep</span> <span class="hl def">=</span> <span class="hl sng">&quot;\t&quot;</span><span class="hl def">);</span>

<span class="hl com"># Ensure column names to` match `IDs` in `zscore_` files</span>
<span class="hl def">zscore_AD_male</span> <span class="hl kwb">&lt;-</span> <span class="hl def">zscore_AD_male</span> <span class="hl opt">%&gt;%</span> <span class="hl kwd">rename_with</span><span class="hl def">(</span><span class="hl opt">~</span> <span class="hl kwd">gsub</span><span class="hl def">(</span><span class="hl sng">&quot;X&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;&quot;</span><span class="hl def">, .),</span> <span class="hl kwd">starts_with</span><span class="hl def">(</span><span class="hl sng">&quot;X&quot;</span><span class="hl def">))</span>

<span class="hl kwd">dim</span><span class="hl def">(zscore_AD_male)</span>
</pre></div>
<div class="output"><pre class="knitr r">## [1] 7800   48
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl def">zscore_AD_female</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">read.delim</span><span class="hl def">(</span><span class="hl sng">&quot;/Users/poddea/Desktop/ROSMAP_data_100623/TMT_proteomics/Round2/z_stat_result_AD_female.txt&quot;</span><span class="hl def">,</span> <span class="hl kwc">header</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span> <span class="hl kwc">sep</span> <span class="hl def">=</span> <span class="hl sng">&quot;\t&quot;</span><span class="hl def">);</span>

<span class="hl com"># Ensure column names to` match `IDs` in `zscore_` files</span>
<span class="hl def">zscore_AD_female</span> <span class="hl kwb">&lt;-</span> <span class="hl def">zscore_AD_female</span> <span class="hl opt">%&gt;%</span> <span class="hl kwd">rename_with</span><span class="hl def">(</span><span class="hl opt">~</span> <span class="hl kwd">gsub</span><span class="hl def">(</span><span class="hl sng">&quot;X&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;&quot;</span><span class="hl def">, .),</span> <span class="hl kwd">starts_with</span><span class="hl def">(</span><span class="hl sng">&quot;X&quot;</span><span class="hl def">))</span>

<span class="hl kwd">dim</span><span class="hl def">(zscore_AD_female)</span>
</pre></div>
<div class="output"><pre class="knitr r">## [1] 7800  128
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl def">zscore_AsymAD_male</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">read.delim</span><span class="hl def">(</span><span class="hl sng">&quot;/Users/poddea/Desktop/ROSMAP_data_100623/TMT_proteomics/Round2/z_stat_result_AsymAD_male.txt&quot;</span><span class="hl def">,</span> <span class="hl kwc">header</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span> <span class="hl kwc">sep</span> <span class="hl def">=</span> <span class="hl sng">&quot;\t&quot;</span><span class="hl def">);</span>

<span class="hl com"># Ensure column names to` match `IDs` in `zscore_` files</span>
<span class="hl def">zscore_AsymAD_male</span> <span class="hl kwb">&lt;-</span> <span class="hl def">zscore_AsymAD_male</span> <span class="hl opt">%&gt;%</span> <span class="hl kwd">rename_with</span><span class="hl def">(</span><span class="hl opt">~</span> <span class="hl kwd">gsub</span><span class="hl def">(</span><span class="hl sng">&quot;X&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;&quot;</span><span class="hl def">, .),</span> <span class="hl kwd">starts_with</span><span class="hl def">(</span><span class="hl sng">&quot;X&quot;</span><span class="hl def">))</span>

<span class="hl kwd">dim</span><span class="hl def">(zscore_AsymAD_male)</span>
</pre></div>
<div class="output"><pre class="knitr r">## [1] 7800   62
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl def">zscore_AsymAD_female</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">read.delim</span><span class="hl def">(</span><span class="hl sng">&quot;/Users/poddea/Desktop/ROSMAP_data_100623/TMT_proteomics/Round2/z_stat_result_AsymAD_female.txt&quot;</span><span class="hl def">,</span> <span class="hl kwc">header</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span> <span class="hl kwc">sep</span> <span class="hl def">=</span> <span class="hl sng">&quot;\t&quot;</span><span class="hl def">);</span>

<span class="hl com"># Ensure column names to` match `IDs` in `zscore_` files</span>
<span class="hl def">zscore_AsymAD_female</span> <span class="hl kwb">&lt;-</span> <span class="hl def">zscore_AsymAD_female</span> <span class="hl opt">%&gt;%</span> <span class="hl kwd">rename_with</span><span class="hl def">(</span><span class="hl opt">~</span> <span class="hl kwd">gsub</span><span class="hl def">(</span><span class="hl sng">&quot;X&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;&quot;</span><span class="hl def">, .),</span> <span class="hl kwd">starts_with</span><span class="hl def">(</span><span class="hl sng">&quot;X&quot;</span><span class="hl def">))</span>

<span class="hl kwd">dim</span><span class="hl def">(zscore_AsymAD_female)</span>
</pre></div>
<div class="output"><pre class="knitr r">## [1] 7800  160
</pre></div>
</div></div>

<p>Compare Gene-wise Zscore for CT and AD Individuals across Pathways (Example for Glycerophospholipid metabolism   pathway):</p>

<div class="chunk" id="unnamed-chunk-3"><div class="rcode"><div class="source"><pre class="knitr r"><span class="hl kwd">library</span><span class="hl def">(dplyr)</span>
<span class="hl kwd">library</span><span class="hl def">(tidyr)</span>
<span class="hl kwd">library</span><span class="hl def">(DT)</span>

<span class="hl kwd">setwd</span><span class="hl def">(</span><span class="hl sng">&quot;/Users/poddea/Desktop/ROSMAP_data_100623/TMT_proteomics/Round2/Pathway_sprcific_analysis/kegg_hsa_all&quot;</span><span class="hl def">)</span>

<span class="hl com"># Open individual KEGG pathway file</span>
<span class="hl def">hsa00564</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">read.delim</span><span class="hl def">(</span><span class="hl sng">&quot;hsa00564.txt&quot;</span><span class="hl def">,</span> <span class="hl kwc">header</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span> <span class="hl kwc">sep</span> <span class="hl def">=</span> <span class="hl sng">&quot;\t&quot;</span><span class="hl def">);</span>
<span class="hl kwd">dim</span><span class="hl def">(hsa00564)</span>
</pre></div>
<div class="output"><pre class="knitr r">## [1] 103   5
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl com">#head(hsa00564, 5)</span>

<span class="hl com"># Print the table using DT</span>
<span class="hl kwd">datatable</span><span class="hl def">(hsa00564,</span> <span class="hl kwc">options</span> <span class="hl def">=</span> <span class="hl kwd">list</span><span class="hl def">(</span><span class="hl kwc">pageLength</span> <span class="hl def">=</span> <span class="hl num">5</span><span class="hl def">,</span> <span class="hl kwc">autoWidth</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">))</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-3-1.png" alt="plot of chunk unnamed-chunk-3" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com"># Merge the data frames</span>
<span class="hl def">hsa00564_zscore_AD_female</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">merge</span><span class="hl def">(hsa00564, zscore_AD_female,</span> <span class="hl kwc">by.x</span> <span class="hl def">=</span> <span class="hl sng">&quot;gene_symbol&quot;</span><span class="hl def">,</span> <span class="hl kwc">by.y</span> <span class="hl def">=</span> <span class="hl sng">&quot;Protein_name&quot;</span><span class="hl def">,</span> <span class="hl kwc">all.x</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>

<span class="hl com">#head(hsa00564_zscore_AD_female, 2)</span>


<span class="hl com"># Print the table using DT</span>
<span class="hl kwd">datatable</span><span class="hl def">(hsa00564_zscore_AD_female,</span> <span class="hl kwc">options</span> <span class="hl def">=</span> <span class="hl kwd">list</span><span class="hl def">(</span><span class="hl kwc">pageLength</span> <span class="hl def">=</span> <span class="hl num">5</span><span class="hl def">,</span> <span class="hl kwc">autoWidth</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">))</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-3-2.png" alt="plot of chunk unnamed-chunk-3" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">#write.table(hsa04710_zscore_AD_female,&quot;/Users/poddea/Desktop/hsa04710_zscore_AD_female.txt&quot;,sep=&quot;\t&quot;,quote=F, row.name= FALSE)</span>

<span class="hl com"># Identify columns ending with &quot;_Zscore&quot;</span>
<span class="hl def">zscore_columns</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">grep</span><span class="hl def">(</span><span class="hl sng">&quot;_Zscore$&quot;</span><span class="hl def">,</span> <span class="hl kwd">names</span><span class="hl def">(hsa00564_zscore_AD_female),</span> <span class="hl kwc">value</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>

<span class="hl com"># Calculate the average for each of these columns</span>
<span class="hl def">average_zscores</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">colMeans</span><span class="hl def">(hsa00564_zscore_AD_female[, zscore_columns],</span> <span class="hl kwc">na.rm</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>

<span class="hl com"># Remove &quot;_Zscore&quot; suffix from the Sample_ID names</span>
<span class="hl def">sample_ids</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">sub</span><span class="hl def">(</span><span class="hl sng">&quot;_Zscore$&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;&quot;</span><span class="hl def">, zscore_columns)</span>

<span class="hl com"># Create a new data frame with &quot;Sample_ID&quot; and the calculated averages</span>
<span class="hl def">Pathway_zscore_AD_female</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">data.frame</span><span class="hl def">(</span><span class="hl kwc">Sample_ID</span> <span class="hl def">= sample_ids,</span> <span class="hl kwc">hsa00564_zscore_AD_female</span> <span class="hl def">= average_zscores)</span>

<span class="hl com">#head(Pathway_zscore_AD_female)</span>

<span class="hl com"># Print the table using DT</span>
<span class="hl kwd">datatable</span><span class="hl def">(Pathway_zscore_AD_female,</span> <span class="hl kwc">options</span> <span class="hl def">=</span> <span class="hl kwd">list</span><span class="hl def">(</span><span class="hl kwc">pageLength</span> <span class="hl def">=</span> <span class="hl num">5</span><span class="hl def">,</span> <span class="hl kwc">autoWidth</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">))</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-3-3.png" alt="plot of chunk unnamed-chunk-3" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com"># Write the new data frame to a file</span>
<span class="hl com">#write.table(Pathway_zscore_AD_female, &quot;/Users/poddea/Desktop/Pathway_zscore_AD_female.txt&quot;, sep = &quot;\t&quot;, quote = FALSE, row.names = FALSE)</span>
</pre></div>
</div></div>

<p>Compare Gene-wise Zscore for CT, AD and AsymAD Individuals across Pathways:</p>

<div class="chunk" id="unnamed-chunk-4"><div class="rcode"><div class="source"><pre class="knitr r"><span class="hl kwd">library</span><span class="hl def">(ggplot2)</span>
</pre></div>
<div class="warning"><pre class="knitr r">## Warning: package 'ggplot2' was built under R version 4.5.2
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl kwd">library</span><span class="hl def">(gplots)</span>
</pre></div>
<div class="message"><pre class="knitr r">## 
## Attaching package: 'gplots'
</pre></div>
<div class="message"><pre class="knitr r">## The following object is masked from 'package:stats':
## 
##     lowess
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl kwd">library</span><span class="hl def">(RColorBrewer)</span>
<span class="hl kwd">library</span><span class="hl def">(dplyr)</span>
<span class="hl kwd">library</span><span class="hl def">(tidyr)</span>
<span class="hl kwd">library</span><span class="hl def">(DT)</span>

<span class="hl com">############################################################################################################################################################################</span>
<span class="hl com">## AD_female</span>

<span class="hl com"># Function to process each file</span>
<span class="hl def">process_kegg_file</span> <span class="hl kwb">&lt;-</span> <span class="hl kwa">function</span><span class="hl def">(</span><span class="hl kwc">file_path</span><span class="hl def">,</span> <span class="hl kwc">zscore_data</span><span class="hl def">) {</span>
  <span class="hl com"># Read the KEGG pathway file</span>
  <span class="hl def">kegg_data</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">read.delim</span><span class="hl def">(file_path,</span> <span class="hl kwc">header</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span> <span class="hl kwc">sep</span> <span class="hl def">=</span> <span class="hl sng">&quot;\t&quot;</span><span class="hl def">)</span>

  <span class="hl com"># Merge with zscore_data</span>
  <span class="hl def">merged_data</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">merge</span><span class="hl def">(kegg_data, zscore_data,</span> <span class="hl kwc">by.x</span> <span class="hl def">=</span> <span class="hl sng">&quot;gene_symbol&quot;</span><span class="hl def">,</span> <span class="hl kwc">by.y</span> <span class="hl def">=</span> <span class="hl sng">&quot;Protein_name&quot;</span><span class="hl def">,</span> <span class="hl kwc">all.x</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>

  <span class="hl com"># Identify columns ending with &quot;_Zscore&quot;</span>
  <span class="hl def">zscore_columns</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">grep</span><span class="hl def">(</span><span class="hl sng">&quot;_Zscore$&quot;</span><span class="hl def">,</span> <span class="hl kwd">names</span><span class="hl def">(merged_data),</span> <span class="hl kwc">value</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>

  <span class="hl com"># Calculate Stouffer's combined Z-scores for each column</span>
  <span class="hl def">stouffer_zscores</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">apply</span><span class="hl def">(merged_data[, zscore_columns],</span> <span class="hl num">2</span><span class="hl def">,</span> <span class="hl kwa">function</span><span class="hl def">(</span><span class="hl kwc">z_scores</span><span class="hl def">) {</span>
    <span class="hl com"># Remove NA values</span>
    <span class="hl def">z_scores</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">na.omit</span><span class="hl def">(z_scores)</span>

    <span class="hl com"># Calculate the Stouffer's Z-score</span>
    <span class="hl def">stouffer_z</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">sum</span><span class="hl def">(z_scores)</span> <span class="hl opt">/</span> <span class="hl kwd">sqrt</span><span class="hl def">(</span><span class="hl kwd">length</span><span class="hl def">(z_scores))</span>

    <span class="hl kwd">return</span><span class="hl def">(stouffer_z)</span>
  <span class="hl def">})</span>

  <span class="hl com"># Remove &quot;_Zscore&quot; suffix from the Sample_ID names</span>
  <span class="hl def">sample_ids</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">sub</span><span class="hl def">(</span><span class="hl sng">&quot;_Zscore$&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;&quot;</span><span class="hl def">, zscore_columns)</span>

  <span class="hl com"># Create a new data frame with &quot;Sample_ID&quot; and the calculated Stouffer's Z-scores</span>
  <span class="hl def">pathway_id</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">unique</span><span class="hl def">(kegg_data</span><span class="hl opt">$</span><span class="hl def">pathway_id)[</span><span class="hl num">1</span><span class="hl def">]</span>
  <span class="hl def">colname</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">paste0</span><span class="hl def">(pathway_id,</span> <span class="hl sng">&quot;_Zscore&quot;</span><span class="hl def">)</span>
  <span class="hl def">results_df</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">data.frame</span><span class="hl def">(</span><span class="hl kwc">Sample_ID</span> <span class="hl def">= sample_ids)</span>
  <span class="hl def">results_df[[colname]]</span> <span class="hl kwb">&lt;-</span> <span class="hl def">stouffer_zscores</span>

  <span class="hl kwd">return</span><span class="hl def">(results_df)</span>
<span class="hl def">}</span>

<span class="hl com"># Path to the directory containing hsa*.txt files</span>
<span class="hl def">dir_path</span> <span class="hl kwb">&lt;-</span> <span class="hl sng">&quot;/Users/poddea/Desktop/ROSMAP_data_100623/TMT_proteomics/Round2/Pathway_sprcific_analysis/kegg_hsa_all&quot;</span>

<span class="hl com"># Read the zscore_AD_female data</span>
<span class="hl com">#zscore_AD_female &lt;- read.delim(&quot;zscore_AD_female.txt&quot;, header = TRUE, sep = &quot;\t&quot;)</span>

<span class="hl com"># List all hsa*.txt files in the directory</span>
<span class="hl def">kegg_files</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">list.files</span><span class="hl def">(</span><span class="hl kwc">path</span> <span class="hl def">= dir_path,</span> <span class="hl kwc">pattern</span> <span class="hl def">=</span> <span class="hl sng">&quot;^hsa.*\\.txt$&quot;</span><span class="hl def">,</span> <span class="hl kwc">full.names</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>

<span class="hl com"># Initialize an empty list to store the results</span>
<span class="hl def">all_results</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">list</span><span class="hl def">()</span>

<span class="hl com"># Process each KEGG file</span>
<span class="hl kwa">for</span> <span class="hl def">(file</span> <span class="hl kwa">in</span> <span class="hl def">kegg_files) {</span>
  <span class="hl def">result</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">process_kegg_file</span><span class="hl def">(file, zscore_AD_female)</span>
  <span class="hl def">all_results[[file]]</span> <span class="hl kwb">&lt;-</span> <span class="hl def">result</span>
<span class="hl def">}</span>

<span class="hl com"># Combine all results into one data frame by merging them on Sample_ID</span>
<span class="hl def">final_results_AD_female</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">Reduce</span><span class="hl def">(</span><span class="hl kwa">function</span><span class="hl def">(</span><span class="hl kwc">x</span><span class="hl def">,</span> <span class="hl kwc">y</span><span class="hl def">)</span> <span class="hl kwd">merge</span><span class="hl def">(x, y,</span> <span class="hl kwc">by</span> <span class="hl def">=</span> <span class="hl sng">&quot;Sample_ID&quot;</span><span class="hl def">,</span> <span class="hl kwc">all</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">), all_results)</span>

<span class="hl com"># Write the final results to a file</span>
<span class="hl kwd">write.table</span><span class="hl def">(final_results_AD_female,</span> <span class="hl sng">&quot;Pathway_zscore_AD_female_summary.txt&quot;</span><span class="hl def">,</span> <span class="hl kwc">sep</span> <span class="hl def">=</span> <span class="hl sng">&quot;\t&quot;</span><span class="hl def">,</span> <span class="hl kwc">quote</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">,</span> <span class="hl kwc">row.names</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">)</span>

<span class="hl com">###########################################################################################################################################################</span>
<span class="hl com"># Transpose the file to count 1st and 99th percentile of the Z score distributions in individual pathway</span>
<span class="hl def">final_results_AD_female_t</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">as.data.frame</span><span class="hl def">(</span><span class="hl kwd">t</span><span class="hl def">(final_results_AD_female[,</span><span class="hl opt">-</span><span class="hl num">1</span><span class="hl def">]))</span>

<span class="hl kwd">colnames</span><span class="hl def">(final_results_AD_female_t)</span> <span class="hl kwb">&lt;-</span> <span class="hl def">final_results_AD_female</span><span class="hl opt">$</span><span class="hl def">Sample_ID</span>

<span class="hl com"># Convert the row names to a new column named &quot;Pathway_ID&quot;</span>
<span class="hl def">final_results_AD_female_t</span><span class="hl opt">$</span><span class="hl def">Pathway_ID</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">rownames</span><span class="hl def">(final_results_AD_female_t)</span>

<span class="hl com"># Reorder the columns so that &quot;Pathway_ID&quot; comes first</span>
<span class="hl def">final_results_AD_female_t</span> <span class="hl kwb">&lt;-</span> <span class="hl def">final_results_AD_female_t[,</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;Pathway_ID&quot;</span><span class="hl def">,</span> <span class="hl kwd">names</span><span class="hl def">(final_results_AD_female_t)[</span><span class="hl opt">-</span><span class="hl kwd">ncol</span><span class="hl def">(final_results_AD_female_t)])]</span>

<span class="hl com"># Reset the row names as they are no longer needed</span>
<span class="hl kwd">rownames</span><span class="hl def">(final_results_AD_female_t)</span> <span class="hl kwb">&lt;-</span> <span class="hl kwa">NULL</span>

<span class="hl com"># Print the table using DT</span>
<span class="hl kwd">datatable</span><span class="hl def">(final_results_AD_female_t,</span> <span class="hl kwc">options</span> <span class="hl def">=</span> <span class="hl kwd">list</span><span class="hl def">(</span><span class="hl kwc">pageLength</span> <span class="hl def">=</span> <span class="hl num">5</span><span class="hl def">,</span> <span class="hl kwc">autoWidth</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">))</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-4-1.png" alt="plot of chunk unnamed-chunk-4" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">###########################################################################################################################################################</span>
<span class="hl com"># Plot tha data</span>
<span class="hl com"># Set rownames to Sample_ID and remove Sample_ID column</span>
<span class="hl kwd">rownames</span><span class="hl def">(final_results_AD_female)</span> <span class="hl kwb">&lt;-</span> <span class="hl def">final_results_AD_female</span><span class="hl opt">$</span><span class="hl def">Sample_ID</span>

<span class="hl def">final_results_AD_female</span> <span class="hl kwb">&lt;-</span> <span class="hl def">final_results_AD_female[,</span> <span class="hl opt">-</span><span class="hl num">1</span><span class="hl def">]</span>

<span class="hl com"># Remove columns with NA values</span>
<span class="hl def">final_results_AD_female</span> <span class="hl kwb">&lt;-</span> <span class="hl def">final_results_AD_female[,</span> <span class="hl kwd">colSums</span><span class="hl def">(</span><span class="hl kwd">is.na</span><span class="hl def">(final_results_AD_female))</span> <span class="hl opt">==</span> <span class="hl num">0</span><span class="hl def">]</span>


<span class="hl com"># Set up the color scale using RColorBrewer</span>
<span class="hl def">color_scale</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">colorRampPalette</span><span class="hl def">(</span><span class="hl kwd">brewer.pal</span><span class="hl def">(</span><span class="hl num">9</span><span class="hl def">,</span> <span class="hl sng">&quot;RdYlGn&quot;</span><span class="hl def">))</span>

<span class="hl com"># Plot the heatmap using heatmap.2</span>
<span class="hl kwd">heatmap.2</span><span class="hl def">(</span>
<span class="hl kwd">as.matrix</span><span class="hl def">(final_results_AD_female),</span>
<span class="hl kwc">Rowv</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>
<span class="hl kwc">Colv</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>
<span class="hl kwc">col</span> <span class="hl def">= color_scale,</span>
<span class="hl kwc">scale</span> <span class="hl def">=</span> <span class="hl sng">&quot;column&quot;</span><span class="hl def">,</span>  <span class="hl com"># Scale rows (genes) to Z-scores</span>
<span class="hl kwc">trace</span> <span class="hl def">=</span> <span class="hl sng">&quot;none&quot;</span><span class="hl def">,</span>  <span class="hl com"># Turn off row and column annotations</span>
<span class="hl kwc">margins</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl num">20</span><span class="hl def">,</span> <span class="hl num">20</span><span class="hl def">),</span>  <span class="hl com"># Set margins for row and column labels</span>
<span class="hl com">#col = cm.colors(256),  # Set the color map (256 colors)</span>
<span class="hl kwc">key</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>  <span class="hl com"># Include a color key</span>
<span class="hl kwc">keysize</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">,</span>  <span class="hl com"># Size of the color key</span>
<span class="hl kwc">cexCol</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">,</span>  <span class="hl com"># Set column label size</span>
<span class="hl kwc">cexRow</span> <span class="hl def">=</span> <span class="hl num">0.8</span><span class="hl def">,</span>  <span class="hl com"># Set row label size</span>
<span class="hl kwc">dendrogram</span> <span class="hl def">=</span> <span class="hl sng">&quot;both&quot;</span><span class="hl def">,</span>  <span class="hl com"># Show both row and column dendrograms</span>
<span class="hl kwc">main</span><span class="hl def">=</span><span class="hl sng">&quot;Pathway-specific Z score for AD Individuals (female)&quot;</span>
<span class="hl def">)</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-4-2.png" alt="plot of chunk unnamed-chunk-4" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">############################################################################################################################################################################</span>
<span class="hl com">## Count the number of gene identified in each pathway</span>

<span class="hl com"># Function to process each file</span>
<span class="hl def">process_kegg_file</span> <span class="hl kwb">&lt;-</span> <span class="hl kwa">function</span><span class="hl def">(</span><span class="hl kwc">file_path</span><span class="hl def">,</span> <span class="hl kwc">zscore_data</span><span class="hl def">) {</span>
  <span class="hl com"># Read the KEGG pathway file</span>
  <span class="hl def">kegg_data</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">read.delim</span><span class="hl def">(file_path,</span> <span class="hl kwc">header</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span> <span class="hl kwc">sep</span> <span class="hl def">=</span> <span class="hl sng">&quot;\t&quot;</span><span class="hl def">)</span>

  <span class="hl com"># Merge with zscore_data</span>
   <span class="hl def">merged_data</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">merge</span><span class="hl def">(kegg_data, zscore_data,</span> <span class="hl kwc">by.x</span> <span class="hl def">=</span> <span class="hl sng">&quot;gene_symbol&quot;</span><span class="hl def">,</span> <span class="hl kwc">by.y</span> <span class="hl def">=</span> <span class="hl sng">&quot;Protein_name&quot;</span><span class="hl def">,</span> <span class="hl kwc">all.x</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>

  <span class="hl com"># Identify columns ending with &quot;_Zscore&quot;</span>
  <span class="hl def">zscore_columns</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">grep</span><span class="hl def">(</span><span class="hl sng">&quot;_Zscore$&quot;</span><span class="hl def">,</span> <span class="hl kwd">names</span><span class="hl def">(merged_data),</span> <span class="hl kwc">value</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>

  <span class="hl com"># Calculate the number of non-NA values (mapped genes) for each sample</span>
  <span class="hl def">gene_counts</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">apply</span><span class="hl def">(merged_data[, zscore_columns],</span> <span class="hl num">2</span><span class="hl def">,</span> <span class="hl kwa">function</span><span class="hl def">(</span><span class="hl kwc">z_scores</span><span class="hl def">) {</span>
    <span class="hl kwd">return</span><span class="hl def">(</span><span class="hl kwd">sum</span><span class="hl def">(</span><span class="hl opt">!</span><span class="hl kwd">is.na</span><span class="hl def">(z_scores)))</span>
  <span class="hl def">})</span>

  <span class="hl com"># Remove &quot;_Zscore&quot; suffix from the Sample_ID names</span>
  <span class="hl def">sample_ids</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">sub</span><span class="hl def">(</span><span class="hl sng">&quot;_Zscore$&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;&quot;</span><span class="hl def">, zscore_columns)</span>

  <span class="hl com"># Create a new data frame with &quot;Sample_ID&quot; and the number of mapped genes</span>
  <span class="hl def">pathway_id</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">unique</span><span class="hl def">(kegg_data</span><span class="hl opt">$</span><span class="hl def">pathway_id)[</span><span class="hl num">1</span><span class="hl def">]</span>
  <span class="hl def">colname</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">paste0</span><span class="hl def">(pathway_id,</span> <span class="hl sng">&quot;_gene_count&quot;</span><span class="hl def">)</span>
  <span class="hl def">gene_count_df</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">data.frame</span><span class="hl def">(</span><span class="hl kwc">Sample_ID</span> <span class="hl def">= sample_ids)</span>
  <span class="hl def">gene_count_df[[colname]]</span> <span class="hl kwb">&lt;-</span> <span class="hl def">gene_counts</span>

  <span class="hl kwd">return</span><span class="hl def">(gene_count_df)</span>
<span class="hl def">}</span>

<span class="hl com"># Path to the directory containing hsa*.txt files</span>
<span class="hl def">dir_path</span> <span class="hl kwb">&lt;-</span> <span class="hl sng">&quot;/Users/poddea/Desktop/ROSMAP_data_100623/TMT_proteomics/Round2/Pathway_sprcific_analysis/kegg_hsa_all&quot;</span>

<span class="hl com"># Read the zscore_AD_female data</span>
<span class="hl com"># zscore_AD_female &lt;- read.delim(&quot;zscore_AD_female.txt&quot;, header = TRUE, sep = &quot;\t&quot;)</span>

<span class="hl com"># List all hsa*.txt files in the directory</span>
<span class="hl def">kegg_files</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">list.files</span><span class="hl def">(</span><span class="hl kwc">path</span> <span class="hl def">= dir_path,</span> <span class="hl kwc">pattern</span> <span class="hl def">=</span> <span class="hl sng">&quot;^hsa.*\\.txt$&quot;</span><span class="hl def">,</span> <span class="hl kwc">full.names</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>

<span class="hl com"># Initialize an empty list to store the results</span>
<span class="hl def">all_gene_counts</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">list</span><span class="hl def">()</span>

<span class="hl com"># Process each KEGG file to count the number of mapped genes</span>
<span class="hl kwa">for</span> <span class="hl def">(file</span> <span class="hl kwa">in</span> <span class="hl def">kegg_files) {</span>
  <span class="hl def">gene_count_result</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">process_kegg_file</span><span class="hl def">(file, zscore_AD_female)</span>
  <span class="hl def">all_gene_counts[[file]]</span> <span class="hl kwb">&lt;-</span> <span class="hl def">gene_count_result</span>
<span class="hl def">}</span>

<span class="hl com"># Combine all gene count results into one data frame by merging them on Sample_ID</span>
<span class="hl def">final_gene_counts_AD_female</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">Reduce</span><span class="hl def">(</span><span class="hl kwa">function</span><span class="hl def">(</span><span class="hl kwc">x</span><span class="hl def">,</span> <span class="hl kwc">y</span><span class="hl def">)</span> <span class="hl kwd">merge</span><span class="hl def">(x, y,</span> <span class="hl kwc">by</span> <span class="hl def">=</span> <span class="hl sng">&quot;Sample_ID&quot;</span><span class="hl def">,</span> <span class="hl kwc">all</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">), all_gene_counts)</span>

<span class="hl com"># Write the final gene counts to a file</span>
<span class="hl kwd">write.table</span><span class="hl def">(final_gene_counts_AD_female,</span> <span class="hl sng">&quot;Pathway_gene_count_AD_female_summary.txt&quot;</span><span class="hl def">,</span> <span class="hl kwc">sep</span> <span class="hl def">=</span> <span class="hl sng">&quot;\t&quot;</span><span class="hl def">,</span> <span class="hl kwc">quote</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">,</span> <span class="hl kwc">row.names</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">)</span>


<span class="hl com">############################################################################################################################################################################</span>
<span class="hl com">############################################################################################################################################################################</span>
<span class="hl com">## AD_male</span>

<span class="hl com"># Function to process each file</span>
<span class="hl def">process_kegg_file</span> <span class="hl kwb">&lt;-</span> <span class="hl kwa">function</span><span class="hl def">(</span><span class="hl kwc">file_path</span><span class="hl def">,</span> <span class="hl kwc">zscore_data</span><span class="hl def">) {</span>
  <span class="hl com"># Read the KEGG pathway file</span>
  <span class="hl def">kegg_data</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">read.delim</span><span class="hl def">(file_path,</span> <span class="hl kwc">header</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span> <span class="hl kwc">sep</span> <span class="hl def">=</span> <span class="hl sng">&quot;\t&quot;</span><span class="hl def">)</span>

  <span class="hl com"># Merge with zscore_data</span>
   <span class="hl def">merged_data</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">merge</span><span class="hl def">(kegg_data, zscore_data,</span> <span class="hl kwc">by.x</span> <span class="hl def">=</span> <span class="hl sng">&quot;gene_symbol&quot;</span><span class="hl def">,</span> <span class="hl kwc">by.y</span> <span class="hl def">=</span> <span class="hl sng">&quot;Protein_name&quot;</span><span class="hl def">,</span> <span class="hl kwc">all.x</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>

  <span class="hl com"># Identify columns ending with &quot;_Zscore&quot;</span>
  <span class="hl def">zscore_columns</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">grep</span><span class="hl def">(</span><span class="hl sng">&quot;_Zscore$&quot;</span><span class="hl def">,</span> <span class="hl kwd">names</span><span class="hl def">(merged_data),</span> <span class="hl kwc">value</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>

  <span class="hl com"># Calculate Stouffer's combined Z-scores for each column</span>
  <span class="hl def">stouffer_zscores</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">apply</span><span class="hl def">(merged_data[, zscore_columns],</span> <span class="hl num">2</span><span class="hl def">,</span> <span class="hl kwa">function</span><span class="hl def">(</span><span class="hl kwc">z_scores</span><span class="hl def">) {</span>
    <span class="hl com"># Remove NA values</span>
    <span class="hl def">z_scores</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">na.omit</span><span class="hl def">(z_scores)</span>

    <span class="hl com"># Calculate the Stouffer's Z-score</span>
    <span class="hl def">stouffer_z</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">sum</span><span class="hl def">(z_scores)</span> <span class="hl opt">/</span> <span class="hl kwd">sqrt</span><span class="hl def">(</span><span class="hl kwd">length</span><span class="hl def">(z_scores))</span>

    <span class="hl kwd">return</span><span class="hl def">(stouffer_z)</span>
  <span class="hl def">})</span>

  <span class="hl com"># Remove &quot;_Zscore&quot; suffix from the Sample_ID names</span>
  <span class="hl def">sample_ids</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">sub</span><span class="hl def">(</span><span class="hl sng">&quot;_Zscore$&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;&quot;</span><span class="hl def">, zscore_columns)</span>

  <span class="hl com"># Create a new data frame with &quot;Sample_ID&quot; and the calculated Stouffer's Z-scores</span>
  <span class="hl def">pathway_id</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">unique</span><span class="hl def">(kegg_data</span><span class="hl opt">$</span><span class="hl def">pathway_id)[</span><span class="hl num">1</span><span class="hl def">]</span>
  <span class="hl def">colname</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">paste0</span><span class="hl def">(pathway_id,</span> <span class="hl sng">&quot;_Zscore&quot;</span><span class="hl def">)</span>
  <span class="hl def">results_df</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">data.frame</span><span class="hl def">(</span><span class="hl kwc">Sample_ID</span> <span class="hl def">= sample_ids)</span>
  <span class="hl def">results_df[[colname]]</span> <span class="hl kwb">&lt;-</span> <span class="hl def">stouffer_zscores</span>

  <span class="hl kwd">return</span><span class="hl def">(results_df)</span>
<span class="hl def">}</span>

<span class="hl com"># Path to the directory containing hsa*.txt files</span>
<span class="hl def">dir_path</span> <span class="hl kwb">&lt;-</span> <span class="hl sng">&quot;/Users/poddea/Desktop/ROSMAP_data_100623/TMT_proteomics/Round2/Pathway_sprcific_analysis/kegg_hsa_all&quot;</span>

<span class="hl com"># Read the zscore_AD_female data</span>
<span class="hl com">#zscore_AD_female &lt;- read.delim(&quot;zscore_AD_male.txt&quot;, header = TRUE, sep = &quot;\t&quot;)</span>

<span class="hl com"># List all hsa*.txt files in the directory</span>
<span class="hl def">kegg_files</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">list.files</span><span class="hl def">(</span><span class="hl kwc">path</span> <span class="hl def">= dir_path,</span> <span class="hl kwc">pattern</span> <span class="hl def">=</span> <span class="hl sng">&quot;^hsa.*\\.txt$&quot;</span><span class="hl def">,</span> <span class="hl kwc">full.names</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>

<span class="hl com"># Initialize an empty list to store the results</span>
<span class="hl def">all_results</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">list</span><span class="hl def">()</span>

<span class="hl com"># Process each KEGG file</span>
<span class="hl kwa">for</span> <span class="hl def">(file</span> <span class="hl kwa">in</span> <span class="hl def">kegg_files) {</span>
  <span class="hl def">result</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">process_kegg_file</span><span class="hl def">(file, zscore_AD_male)</span>
  <span class="hl def">all_results[[file]]</span> <span class="hl kwb">&lt;-</span> <span class="hl def">result</span>
<span class="hl def">}</span>

<span class="hl com"># Combine all results into one data frame by merging them on Sample_ID</span>
<span class="hl def">final_results_AD_male</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">Reduce</span><span class="hl def">(</span><span class="hl kwa">function</span><span class="hl def">(</span><span class="hl kwc">x</span><span class="hl def">,</span> <span class="hl kwc">y</span><span class="hl def">)</span> <span class="hl kwd">merge</span><span class="hl def">(x, y,</span> <span class="hl kwc">by</span> <span class="hl def">=</span> <span class="hl sng">&quot;Sample_ID&quot;</span><span class="hl def">,</span> <span class="hl kwc">all</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">), all_results)</span>

<span class="hl com"># Write the final results to a file</span>
<span class="hl kwd">write.table</span><span class="hl def">(final_results_AD_male,</span> <span class="hl sng">&quot;Pathway_zscore_AD_male_summary.txt&quot;</span><span class="hl def">,</span> <span class="hl kwc">sep</span> <span class="hl def">=</span> <span class="hl sng">&quot;\t&quot;</span><span class="hl def">,</span> <span class="hl kwc">quote</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">,</span> <span class="hl kwc">row.names</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">)</span>

<span class="hl com">###########################################################################################################################################################</span>
<span class="hl com"># Transpose the file to count 1st and 99th percentile of the Z score distributions in individual pathway</span>
<span class="hl def">final_results_AD_male_t</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">as.data.frame</span><span class="hl def">(</span><span class="hl kwd">t</span><span class="hl def">(final_results_AD_male[,</span><span class="hl opt">-</span><span class="hl num">1</span><span class="hl def">]))</span>

<span class="hl kwd">colnames</span><span class="hl def">(final_results_AD_male_t)</span> <span class="hl kwb">&lt;-</span> <span class="hl def">final_results_AD_male</span><span class="hl opt">$</span><span class="hl def">Sample_ID</span>

<span class="hl com"># Convert the row names to a new column named &quot;Pathway_ID&quot;</span>
<span class="hl def">final_results_AD_male_t</span><span class="hl opt">$</span><span class="hl def">Pathway_ID</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">rownames</span><span class="hl def">(final_results_AD_male_t)</span>

<span class="hl com"># Reorder the columns so that &quot;Pathway_ID&quot; comes first</span>
<span class="hl def">final_results_AD_male_t</span> <span class="hl kwb">&lt;-</span> <span class="hl def">final_results_AD_male_t[,</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;Pathway_ID&quot;</span><span class="hl def">,</span> <span class="hl kwd">names</span><span class="hl def">(final_results_AD_male_t)[</span><span class="hl opt">-</span><span class="hl kwd">ncol</span><span class="hl def">(final_results_AD_male_t)])]</span>

<span class="hl com"># Reset the row names as they are no longer needed</span>
<span class="hl kwd">rownames</span><span class="hl def">(final_results_AD_male_t)</span> <span class="hl kwb">&lt;-</span> <span class="hl kwa">NULL</span>

<span class="hl com"># Print the table using DT</span>
<span class="hl kwd">datatable</span><span class="hl def">(final_results_AD_male_t,</span> <span class="hl kwc">options</span> <span class="hl def">=</span> <span class="hl kwd">list</span><span class="hl def">(</span><span class="hl kwc">pageLength</span> <span class="hl def">=</span> <span class="hl num">5</span><span class="hl def">,</span> <span class="hl kwc">autoWidth</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">))</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-4-3.png" alt="plot of chunk unnamed-chunk-4" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">###########################################################################################################################################################</span>
<span class="hl com"># Plot tha data</span>
<span class="hl com"># Set rownames to Sample_ID and remove Sample_ID column</span>
<span class="hl kwd">rownames</span><span class="hl def">(final_results_AD_male)</span> <span class="hl kwb">&lt;-</span> <span class="hl def">final_results_AD_male</span><span class="hl opt">$</span><span class="hl def">Sample_ID</span>

<span class="hl def">final_results_AD_male</span> <span class="hl kwb">&lt;-</span> <span class="hl def">final_results_AD_male[,</span> <span class="hl opt">-</span><span class="hl num">1</span><span class="hl def">]</span>

<span class="hl com"># Remove columns with NA values</span>
<span class="hl def">final_results_AD_male</span> <span class="hl kwb">&lt;-</span> <span class="hl def">final_results_AD_male[,</span> <span class="hl kwd">colSums</span><span class="hl def">(</span><span class="hl kwd">is.na</span><span class="hl def">(final_results_AD_male))</span> <span class="hl opt">==</span> <span class="hl num">0</span><span class="hl def">]</span>


<span class="hl com"># Set up the color scale using RColorBrewer</span>
<span class="hl def">color_scale</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">colorRampPalette</span><span class="hl def">(</span><span class="hl kwd">brewer.pal</span><span class="hl def">(</span><span class="hl num">9</span><span class="hl def">,</span> <span class="hl sng">&quot;RdYlGn&quot;</span><span class="hl def">))</span>

<span class="hl com"># Plot the heatmap using heatmap.2</span>
<span class="hl kwd">heatmap.2</span><span class="hl def">(</span>
<span class="hl kwd">as.matrix</span><span class="hl def">(final_results_AD_male),</span>
<span class="hl kwc">Rowv</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>
<span class="hl kwc">Colv</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>
<span class="hl kwc">col</span> <span class="hl def">= color_scale,</span>
<span class="hl kwc">scale</span> <span class="hl def">=</span> <span class="hl sng">&quot;column&quot;</span><span class="hl def">,</span>  <span class="hl com"># Scale rows (genes) to Z-scores</span>
<span class="hl kwc">trace</span> <span class="hl def">=</span> <span class="hl sng">&quot;none&quot;</span><span class="hl def">,</span>  <span class="hl com"># Turn off row and column annotations</span>
<span class="hl kwc">margins</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl num">20</span><span class="hl def">,</span> <span class="hl num">20</span><span class="hl def">),</span>  <span class="hl com"># Set margins for row and column labels</span>
<span class="hl com">#col = cm.colors(256),  # Set the color map (256 colors)</span>
<span class="hl kwc">key</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>  <span class="hl com"># Include a color key</span>
<span class="hl kwc">keysize</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">,</span>  <span class="hl com"># Size of the color key</span>
<span class="hl kwc">cexCol</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">,</span>  <span class="hl com"># Set column label size</span>
<span class="hl kwc">cexRow</span> <span class="hl def">=</span> <span class="hl num">0.8</span><span class="hl def">,</span>  <span class="hl com"># Set row label size</span>
<span class="hl kwc">dendrogram</span> <span class="hl def">=</span> <span class="hl sng">&quot;both&quot;</span><span class="hl def">,</span>  <span class="hl com"># Show both row and column dendrograms</span>
<span class="hl kwc">main</span><span class="hl def">=</span><span class="hl sng">&quot;Pathway-specific Z score for AD Individuals (male)&quot;</span>
<span class="hl def">)</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-4-4.png" alt="plot of chunk unnamed-chunk-4" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">############################################################################################################################################################################</span>
<span class="hl com">############################################################################################################################################################################</span>
<span class="hl com">## AsymAD_female</span>

<span class="hl com"># Function to process each file</span>
<span class="hl def">process_kegg_file</span> <span class="hl kwb">&lt;-</span> <span class="hl kwa">function</span><span class="hl def">(</span><span class="hl kwc">file_path</span><span class="hl def">,</span> <span class="hl kwc">zscore_data</span><span class="hl def">) {</span>
  <span class="hl com"># Read the KEGG pathway file</span>
  <span class="hl def">kegg_data</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">read.delim</span><span class="hl def">(file_path,</span> <span class="hl kwc">header</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span> <span class="hl kwc">sep</span> <span class="hl def">=</span> <span class="hl sng">&quot;\t&quot;</span><span class="hl def">)</span>

  <span class="hl com"># Merge with zscore_data</span>
  <span class="hl def">merged_data</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">merge</span><span class="hl def">(kegg_data, zscore_data,</span> <span class="hl kwc">by.x</span> <span class="hl def">=</span> <span class="hl sng">&quot;gene_symbol&quot;</span><span class="hl def">,</span> <span class="hl kwc">by.y</span> <span class="hl def">=</span> <span class="hl sng">&quot;Protein_name&quot;</span><span class="hl def">,</span> <span class="hl kwc">all.x</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>

  <span class="hl com"># Identify columns ending with &quot;_Zscore&quot;</span>
  <span class="hl def">zscore_columns</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">grep</span><span class="hl def">(</span><span class="hl sng">&quot;_Zscore$&quot;</span><span class="hl def">,</span> <span class="hl kwd">names</span><span class="hl def">(merged_data),</span> <span class="hl kwc">value</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>

  <span class="hl com"># Calculate Stouffer's combined Z-scores for each column</span>
  <span class="hl def">stouffer_zscores</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">apply</span><span class="hl def">(merged_data[, zscore_columns],</span> <span class="hl num">2</span><span class="hl def">,</span> <span class="hl kwa">function</span><span class="hl def">(</span><span class="hl kwc">z_scores</span><span class="hl def">) {</span>
    <span class="hl com"># Remove NA values</span>
    <span class="hl def">z_scores</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">na.omit</span><span class="hl def">(z_scores)</span>

    <span class="hl com"># Calculate the Stouffer's Z-score</span>
    <span class="hl def">stouffer_z</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">sum</span><span class="hl def">(z_scores)</span> <span class="hl opt">/</span> <span class="hl kwd">sqrt</span><span class="hl def">(</span><span class="hl kwd">length</span><span class="hl def">(z_scores))</span>

    <span class="hl kwd">return</span><span class="hl def">(stouffer_z)</span>
  <span class="hl def">})</span>

  <span class="hl com"># Remove &quot;_Zscore&quot; suffix from the Sample_ID names</span>
  <span class="hl def">sample_ids</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">sub</span><span class="hl def">(</span><span class="hl sng">&quot;_Zscore$&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;&quot;</span><span class="hl def">, zscore_columns)</span>

  <span class="hl com"># Create a new data frame with &quot;Sample_ID&quot; and the calculated Stouffer's Z-scores</span>
  <span class="hl def">pathway_id</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">unique</span><span class="hl def">(kegg_data</span><span class="hl opt">$</span><span class="hl def">pathway_id)[</span><span class="hl num">1</span><span class="hl def">]</span>
  <span class="hl def">colname</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">paste0</span><span class="hl def">(pathway_id,</span> <span class="hl sng">&quot;_Zscore&quot;</span><span class="hl def">)</span>
  <span class="hl def">results_df</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">data.frame</span><span class="hl def">(</span><span class="hl kwc">Sample_ID</span> <span class="hl def">= sample_ids)</span>
  <span class="hl def">results_df[[colname]]</span> <span class="hl kwb">&lt;-</span> <span class="hl def">stouffer_zscores</span>

  <span class="hl kwd">return</span><span class="hl def">(results_df)</span>
<span class="hl def">}</span>

<span class="hl com"># Path to the directory containing hsa*.txt files</span>
<span class="hl def">dir_path</span> <span class="hl kwb">&lt;-</span> <span class="hl sng">&quot;/Users/poddea/Desktop/ROSMAP_data_100623/TMT_proteomics/Round2/Pathway_sprcific_analysis/kegg_hsa_all&quot;</span>

<span class="hl com"># Read the zscore_AsymAD_female data</span>
<span class="hl com">#zscore_AsymAD_female &lt;- read.delim(&quot;zscore_AsymAD_female.txt&quot;, header = TRUE, sep = &quot;\t&quot;)</span>

<span class="hl com"># List all hsa*.txt files in the directory</span>
<span class="hl def">kegg_files</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">list.files</span><span class="hl def">(</span><span class="hl kwc">path</span> <span class="hl def">= dir_path,</span> <span class="hl kwc">pattern</span> <span class="hl def">=</span> <span class="hl sng">&quot;^hsa.*\\.txt$&quot;</span><span class="hl def">,</span> <span class="hl kwc">full.names</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>

<span class="hl com"># Initialize an empty list to store the results</span>
<span class="hl def">all_results</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">list</span><span class="hl def">()</span>

<span class="hl com"># Process each KEGG file</span>
<span class="hl kwa">for</span> <span class="hl def">(file</span> <span class="hl kwa">in</span> <span class="hl def">kegg_files) {</span>
  <span class="hl def">result</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">process_kegg_file</span><span class="hl def">(file, zscore_AsymAD_female)</span>
  <span class="hl def">all_results[[file]]</span> <span class="hl kwb">&lt;-</span> <span class="hl def">result</span>
<span class="hl def">}</span>

<span class="hl com"># Combine all results into one data frame by merging them on Sample_ID</span>
<span class="hl def">final_results_AsymAD_female</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">Reduce</span><span class="hl def">(</span><span class="hl kwa">function</span><span class="hl def">(</span><span class="hl kwc">x</span><span class="hl def">,</span> <span class="hl kwc">y</span><span class="hl def">)</span> <span class="hl kwd">merge</span><span class="hl def">(x, y,</span> <span class="hl kwc">by</span> <span class="hl def">=</span> <span class="hl sng">&quot;Sample_ID&quot;</span><span class="hl def">,</span> <span class="hl kwc">all</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">), all_results)</span>

<span class="hl com"># Write the final results to a file</span>
<span class="hl kwd">write.table</span><span class="hl def">(final_results_AsymAD_female,</span> <span class="hl sng">&quot;Pathway_zscore_AsymAD_female_summary.txt&quot;</span><span class="hl def">,</span> <span class="hl kwc">sep</span> <span class="hl def">=</span> <span class="hl sng">&quot;\t&quot;</span><span class="hl def">,</span> <span class="hl kwc">quote</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">,</span> <span class="hl kwc">row.names</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">)</span>

<span class="hl com">###########################################################################################################################################################</span>
<span class="hl com"># Transpose the file to count 1st and 99th percentile of the Z score distributions in individual pathway</span>
<span class="hl def">final_results_AsymAD_female_t</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">as.data.frame</span><span class="hl def">(</span><span class="hl kwd">t</span><span class="hl def">(final_results_AsymAD_female[,</span><span class="hl opt">-</span><span class="hl num">1</span><span class="hl def">]))</span>

<span class="hl kwd">colnames</span><span class="hl def">(final_results_AsymAD_female_t)</span> <span class="hl kwb">&lt;-</span> <span class="hl def">final_results_AsymAD_female</span><span class="hl opt">$</span><span class="hl def">Sample_ID</span>

<span class="hl com"># Convert the row names to a new column named &quot;Pathway_ID&quot;</span>
<span class="hl def">final_results_AsymAD_female_t</span><span class="hl opt">$</span><span class="hl def">Pathway_ID</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">rownames</span><span class="hl def">(final_results_AsymAD_female_t)</span>

<span class="hl com"># Reorder the columns so that &quot;Pathway_ID&quot; comes first</span>
<span class="hl def">final_results_AsymAD_female_t</span> <span class="hl kwb">&lt;-</span> <span class="hl def">final_results_AsymAD_female_t[,</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;Pathway_ID&quot;</span><span class="hl def">,</span> <span class="hl kwd">names</span><span class="hl def">(final_results_AsymAD_female_t)[</span><span class="hl opt">-</span><span class="hl kwd">ncol</span><span class="hl def">(final_results_AsymAD_female_t)])]</span>

<span class="hl com"># Reset the row names as they are no longer needed</span>
<span class="hl kwd">rownames</span><span class="hl def">(final_results_AsymAD_female_t)</span> <span class="hl kwb">&lt;-</span> <span class="hl kwa">NULL</span>

<span class="hl com"># Print the table using DT</span>
<span class="hl kwd">datatable</span><span class="hl def">(final_results_AsymAD_female_t,</span> <span class="hl kwc">options</span> <span class="hl def">=</span> <span class="hl kwd">list</span><span class="hl def">(</span><span class="hl kwc">pageLength</span> <span class="hl def">=</span> <span class="hl num">5</span><span class="hl def">,</span> <span class="hl kwc">autoWidth</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">))</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-4-5.png" alt="plot of chunk unnamed-chunk-4" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">###########################################################################################################################################################</span>
<span class="hl com"># Plot tha data</span>
<span class="hl com"># Set rownames to Sample_ID and remove Sample_ID column</span>
<span class="hl kwd">rownames</span><span class="hl def">(final_results_AsymAD_female)</span> <span class="hl kwb">&lt;-</span> <span class="hl def">final_results_AsymAD_female</span><span class="hl opt">$</span><span class="hl def">Sample_ID</span>

<span class="hl def">final_results_AsymAD_female</span> <span class="hl kwb">&lt;-</span> <span class="hl def">final_results_AsymAD_female[,</span> <span class="hl opt">-</span><span class="hl num">1</span><span class="hl def">]</span>

<span class="hl com"># Remove columns with NA values</span>
<span class="hl def">final_results_AsymAD_female</span> <span class="hl kwb">&lt;-</span> <span class="hl def">final_results_AsymAD_female[,</span> <span class="hl kwd">colSums</span><span class="hl def">(</span><span class="hl kwd">is.na</span><span class="hl def">(final_results_AsymAD_female))</span> <span class="hl opt">==</span> <span class="hl num">0</span><span class="hl def">]</span>


<span class="hl com"># Set up the color scale using RColorBrewer</span>
<span class="hl def">color_scale</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">colorRampPalette</span><span class="hl def">(</span><span class="hl kwd">brewer.pal</span><span class="hl def">(</span><span class="hl num">9</span><span class="hl def">,</span> <span class="hl sng">&quot;RdYlGn&quot;</span><span class="hl def">))</span>

<span class="hl com"># Plot the heatmap using heatmap.2</span>
<span class="hl kwd">heatmap.2</span><span class="hl def">(</span>
<span class="hl kwd">as.matrix</span><span class="hl def">(final_results_AsymAD_female),</span>
<span class="hl kwc">Rowv</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>
<span class="hl kwc">Colv</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>
<span class="hl kwc">col</span> <span class="hl def">= color_scale,</span>
<span class="hl kwc">scale</span> <span class="hl def">=</span> <span class="hl sng">&quot;column&quot;</span><span class="hl def">,</span>  <span class="hl com"># Scale rows (genes) to Z-scores</span>
<span class="hl kwc">trace</span> <span class="hl def">=</span> <span class="hl sng">&quot;none&quot;</span><span class="hl def">,</span>  <span class="hl com"># Turn off row and column annotations</span>
<span class="hl kwc">margins</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl num">20</span><span class="hl def">,</span> <span class="hl num">20</span><span class="hl def">),</span>  <span class="hl com"># Set margins for row and column labels</span>
<span class="hl com">#col = cm.colors(256),  # Set the color map (256 colors)</span>
<span class="hl kwc">key</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>  <span class="hl com"># Include a color key</span>
<span class="hl kwc">keysize</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">,</span>  <span class="hl com"># Size of the color key</span>
<span class="hl kwc">cexCol</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">,</span>  <span class="hl com"># Set column label size</span>
<span class="hl kwc">cexRow</span> <span class="hl def">=</span> <span class="hl num">0.8</span><span class="hl def">,</span>  <span class="hl com"># Set row label size</span>
<span class="hl kwc">dendrogram</span> <span class="hl def">=</span> <span class="hl sng">&quot;both&quot;</span><span class="hl def">,</span>  <span class="hl com"># Show both row and column dendrograms</span>
<span class="hl kwc">main</span><span class="hl def">=</span><span class="hl sng">&quot;Pathway-specific Z score for AsymAD Individuals (female)&quot;</span>
<span class="hl def">)</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-4-6.png" alt="plot of chunk unnamed-chunk-4" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">############################################################################################################################################################################</span>
<span class="hl com">############################################################################################################################################################################</span>
<span class="hl com">## AsymAD_male</span>

<span class="hl com"># Function to process each file</span>
<span class="hl def">process_kegg_file</span> <span class="hl kwb">&lt;-</span> <span class="hl kwa">function</span><span class="hl def">(</span><span class="hl kwc">file_path</span><span class="hl def">,</span> <span class="hl kwc">zscore_data</span><span class="hl def">) {</span>
  <span class="hl com"># Read the KEGG pathway file</span>
  <span class="hl def">kegg_data</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">read.delim</span><span class="hl def">(file_path,</span> <span class="hl kwc">header</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span> <span class="hl kwc">sep</span> <span class="hl def">=</span> <span class="hl sng">&quot;\t&quot;</span><span class="hl def">)</span>

  <span class="hl com"># Merge with zscore_data</span>
  <span class="hl def">merged_data</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">merge</span><span class="hl def">(kegg_data, zscore_data,</span> <span class="hl kwc">by.x</span> <span class="hl def">=</span> <span class="hl sng">&quot;gene_symbol&quot;</span><span class="hl def">,</span> <span class="hl kwc">by.y</span> <span class="hl def">=</span> <span class="hl sng">&quot;Protein_name&quot;</span><span class="hl def">,</span> <span class="hl kwc">all.x</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>

  <span class="hl com"># Identify columns ending with &quot;_Zscore&quot;</span>
  <span class="hl def">zscore_columns</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">grep</span><span class="hl def">(</span><span class="hl sng">&quot;_Zscore$&quot;</span><span class="hl def">,</span> <span class="hl kwd">names</span><span class="hl def">(merged_data),</span> <span class="hl kwc">value</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>

  <span class="hl com"># Calculate Stouffer's combined Z-scores for each column</span>
  <span class="hl def">stouffer_zscores</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">apply</span><span class="hl def">(merged_data[, zscore_columns],</span> <span class="hl num">2</span><span class="hl def">,</span> <span class="hl kwa">function</span><span class="hl def">(</span><span class="hl kwc">z_scores</span><span class="hl def">) {</span>
    <span class="hl com"># Remove NA values</span>
    <span class="hl def">z_scores</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">na.omit</span><span class="hl def">(z_scores)</span>

    <span class="hl com"># Calculate the Stouffer's Z-score</span>
    <span class="hl def">stouffer_z</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">sum</span><span class="hl def">(z_scores)</span> <span class="hl opt">/</span> <span class="hl kwd">sqrt</span><span class="hl def">(</span><span class="hl kwd">length</span><span class="hl def">(z_scores))</span>

    <span class="hl kwd">return</span><span class="hl def">(stouffer_z)</span>
  <span class="hl def">})</span>

  <span class="hl com"># Remove &quot;_Zscore&quot; suffix from the Sample_ID names</span>
  <span class="hl def">sample_ids</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">sub</span><span class="hl def">(</span><span class="hl sng">&quot;_Zscore$&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;&quot;</span><span class="hl def">, zscore_columns)</span>

  <span class="hl com"># Create a new data frame with &quot;Sample_ID&quot; and the calculated Stouffer's Z-scores</span>
  <span class="hl def">pathway_id</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">unique</span><span class="hl def">(kegg_data</span><span class="hl opt">$</span><span class="hl def">pathway_id)[</span><span class="hl num">1</span><span class="hl def">]</span>
  <span class="hl def">colname</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">paste0</span><span class="hl def">(pathway_id,</span> <span class="hl sng">&quot;_Zscore&quot;</span><span class="hl def">)</span>
  <span class="hl def">results_df</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">data.frame</span><span class="hl def">(</span><span class="hl kwc">Sample_ID</span> <span class="hl def">= sample_ids)</span>
  <span class="hl def">results_df[[colname]]</span> <span class="hl kwb">&lt;-</span> <span class="hl def">stouffer_zscores</span>

  <span class="hl kwd">return</span><span class="hl def">(results_df)</span>
<span class="hl def">}</span>

<span class="hl com"># Path to the directory containing hsa*.txt files</span>
<span class="hl def">dir_path</span> <span class="hl kwb">&lt;-</span> <span class="hl sng">&quot;/Users/poddea/Desktop/ROSMAP_data_100623/TMT_proteomics/Round2/Pathway_sprcific_analysis/kegg_hsa_all&quot;</span>

<span class="hl com"># Read the zscore_AsymAD_male data</span>
<span class="hl com">#zscore_AsymAD_male &lt;- read.delim(&quot;zscore_AsymAD_male.txt&quot;, header = TRUE, sep = &quot;\t&quot;)</span>

<span class="hl com"># List all hsa*.txt files in the directory</span>
<span class="hl def">kegg_files</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">list.files</span><span class="hl def">(</span><span class="hl kwc">path</span> <span class="hl def">= dir_path,</span> <span class="hl kwc">pattern</span> <span class="hl def">=</span> <span class="hl sng">&quot;^hsa.*\\.txt$&quot;</span><span class="hl def">,</span> <span class="hl kwc">full.names</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>

<span class="hl com"># Initialize an empty list to store the results</span>
<span class="hl def">all_results</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">list</span><span class="hl def">()</span>

<span class="hl com"># Process each KEGG file</span>
<span class="hl kwa">for</span> <span class="hl def">(file</span> <span class="hl kwa">in</span> <span class="hl def">kegg_files) {</span>
  <span class="hl def">result</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">process_kegg_file</span><span class="hl def">(file, zscore_AsymAD_male)</span>
  <span class="hl def">all_results[[file]]</span> <span class="hl kwb">&lt;-</span> <span class="hl def">result</span>
<span class="hl def">}</span>

<span class="hl com"># Combine all results into one data frame by merging them on Sample_ID</span>
<span class="hl def">final_results_AsymAD_male</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">Reduce</span><span class="hl def">(</span><span class="hl kwa">function</span><span class="hl def">(</span><span class="hl kwc">x</span><span class="hl def">,</span> <span class="hl kwc">y</span><span class="hl def">)</span> <span class="hl kwd">merge</span><span class="hl def">(x, y,</span> <span class="hl kwc">by</span> <span class="hl def">=</span> <span class="hl sng">&quot;Sample_ID&quot;</span><span class="hl def">,</span> <span class="hl kwc">all</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">), all_results)</span>

<span class="hl com"># Write the final results to a file</span>
<span class="hl kwd">write.table</span><span class="hl def">(final_results_AsymAD_male,</span> <span class="hl sng">&quot;Pathway_zscore_AsymAD_male_summary.txt&quot;</span><span class="hl def">,</span> <span class="hl kwc">sep</span> <span class="hl def">=</span> <span class="hl sng">&quot;\t&quot;</span><span class="hl def">,</span> <span class="hl kwc">quote</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">,</span> <span class="hl kwc">row.names</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">)</span>

<span class="hl com">###########################################################################################################################################################</span>
<span class="hl com"># Transpose the file to count 1st and 99th percentile of the Z score distributions in individual pathway</span>
<span class="hl def">final_results_AsymAD_male_t</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">as.data.frame</span><span class="hl def">(</span><span class="hl kwd">t</span><span class="hl def">(final_results_AsymAD_male[,</span><span class="hl opt">-</span><span class="hl num">1</span><span class="hl def">]))</span>

<span class="hl kwd">colnames</span><span class="hl def">(final_results_AsymAD_male_t)</span> <span class="hl kwb">&lt;-</span> <span class="hl def">final_results_AsymAD_male</span><span class="hl opt">$</span><span class="hl def">Sample_ID</span>

<span class="hl com"># Convert the row names to a new column named &quot;Pathway_ID&quot;</span>
<span class="hl def">final_results_AsymAD_male_t</span><span class="hl opt">$</span><span class="hl def">Pathway_ID</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">rownames</span><span class="hl def">(final_results_AsymAD_male_t)</span>

<span class="hl com"># Reorder the columns so that &quot;Pathway_ID&quot; comes first</span>
<span class="hl def">final_results_AsymAD_male_t</span> <span class="hl kwb">&lt;-</span> <span class="hl def">final_results_AsymAD_male_t[,</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;Pathway_ID&quot;</span><span class="hl def">,</span> <span class="hl kwd">names</span><span class="hl def">(final_results_AsymAD_male_t)[</span><span class="hl opt">-</span><span class="hl kwd">ncol</span><span class="hl def">(final_results_AsymAD_male_t)])]</span>

<span class="hl com"># Reset the row names as they are no longer needed</span>
<span class="hl kwd">rownames</span><span class="hl def">(final_results_AsymAD_male_t)</span> <span class="hl kwb">&lt;-</span> <span class="hl kwa">NULL</span>

<span class="hl com"># Print the table using DT</span>
<span class="hl kwd">datatable</span><span class="hl def">(final_results_AsymAD_male_t,</span> <span class="hl kwc">options</span> <span class="hl def">=</span> <span class="hl kwd">list</span><span class="hl def">(</span><span class="hl kwc">pageLength</span> <span class="hl def">=</span> <span class="hl num">5</span><span class="hl def">,</span> <span class="hl kwc">autoWidth</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">))</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-4-7.png" alt="plot of chunk unnamed-chunk-4" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">###########################################################################################################################################################</span>
<span class="hl com"># Plot tha data</span>
<span class="hl com"># Set rownames to Sample_ID and remove Sample_ID column</span>
<span class="hl kwd">rownames</span><span class="hl def">(final_results_AsymAD_male)</span> <span class="hl kwb">&lt;-</span> <span class="hl def">final_results_AsymAD_male</span><span class="hl opt">$</span><span class="hl def">Sample_ID</span>

<span class="hl def">final_results_AsymAD_male</span> <span class="hl kwb">&lt;-</span> <span class="hl def">final_results_AsymAD_male[,</span> <span class="hl opt">-</span><span class="hl num">1</span><span class="hl def">]</span>

<span class="hl com"># Remove columns with NA values</span>
<span class="hl def">final_results_AsymAD_male</span> <span class="hl kwb">&lt;-</span> <span class="hl def">final_results_AsymAD_male[,</span> <span class="hl kwd">colSums</span><span class="hl def">(</span><span class="hl kwd">is.na</span><span class="hl def">(final_results_AsymAD_male))</span> <span class="hl opt">==</span> <span class="hl num">0</span><span class="hl def">]</span>


<span class="hl com"># Set up the color scale using RColorBrewer</span>
<span class="hl def">color_scale</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">colorRampPalette</span><span class="hl def">(</span><span class="hl kwd">brewer.pal</span><span class="hl def">(</span><span class="hl num">9</span><span class="hl def">,</span> <span class="hl sng">&quot;RdYlGn&quot;</span><span class="hl def">))</span>

<span class="hl com"># Plot the heatmap using heatmap.2</span>
<span class="hl kwd">heatmap.2</span><span class="hl def">(</span>
<span class="hl kwd">as.matrix</span><span class="hl def">(final_results_AsymAD_male),</span>
<span class="hl kwc">Rowv</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>
<span class="hl kwc">Colv</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>
<span class="hl kwc">col</span> <span class="hl def">= color_scale,</span>
<span class="hl kwc">scale</span> <span class="hl def">=</span> <span class="hl sng">&quot;column&quot;</span><span class="hl def">,</span>  <span class="hl com"># Scale rows (genes) to Z-scores</span>
<span class="hl kwc">trace</span> <span class="hl def">=</span> <span class="hl sng">&quot;none&quot;</span><span class="hl def">,</span>  <span class="hl com"># Turn off row and column annotations</span>
<span class="hl kwc">margins</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl num">20</span><span class="hl def">,</span> <span class="hl num">20</span><span class="hl def">),</span>  <span class="hl com"># Set margins for row and column labels</span>
<span class="hl com">#col = cm.colors(256),  # Set the color map (256 colors)</span>
<span class="hl kwc">key</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>  <span class="hl com"># Include a color key</span>
<span class="hl kwc">keysize</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">,</span>  <span class="hl com"># Size of the color key</span>
<span class="hl kwc">cexCol</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">,</span>  <span class="hl com"># Set column label size</span>
<span class="hl kwc">cexRow</span> <span class="hl def">=</span> <span class="hl num">0.8</span><span class="hl def">,</span>  <span class="hl com"># Set row label size</span>
<span class="hl kwc">dendrogram</span> <span class="hl def">=</span> <span class="hl sng">&quot;both&quot;</span><span class="hl def">,</span>  <span class="hl com"># Show both row and column dendrograms</span>
<span class="hl kwc">main</span><span class="hl def">=</span><span class="hl sng">&quot;Pathway-specific Z score for AsymAD Individuals (male)&quot;</span>
<span class="hl def">)</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-4-8.png" alt="plot of chunk unnamed-chunk-4" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">############################################################################################################################################################################</span>
<span class="hl com">############################################################################################################################################################################</span>
<span class="hl com">## CT_female</span>

<span class="hl com"># Function to process each file</span>
<span class="hl def">process_kegg_file</span> <span class="hl kwb">&lt;-</span> <span class="hl kwa">function</span><span class="hl def">(</span><span class="hl kwc">file_path</span><span class="hl def">,</span> <span class="hl kwc">zscore_data</span><span class="hl def">) {</span>
  <span class="hl com"># Read the KEGG pathway file</span>
  <span class="hl def">kegg_data</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">read.delim</span><span class="hl def">(file_path,</span> <span class="hl kwc">header</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span> <span class="hl kwc">sep</span> <span class="hl def">=</span> <span class="hl sng">&quot;\t&quot;</span><span class="hl def">)</span>

  <span class="hl com"># Merge with zscore_data</span>
  <span class="hl def">merged_data</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">merge</span><span class="hl def">(kegg_data, zscore_data,</span> <span class="hl kwc">by.x</span> <span class="hl def">=</span> <span class="hl sng">&quot;gene_symbol&quot;</span><span class="hl def">,</span> <span class="hl kwc">by.y</span> <span class="hl def">=</span> <span class="hl sng">&quot;Protein_name&quot;</span><span class="hl def">,</span> <span class="hl kwc">all.x</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>

  <span class="hl com"># Identify columns ending with &quot;_Zscore&quot;</span>
  <span class="hl def">zscore_columns</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">grep</span><span class="hl def">(</span><span class="hl sng">&quot;_Zscore$&quot;</span><span class="hl def">,</span> <span class="hl kwd">names</span><span class="hl def">(merged_data),</span> <span class="hl kwc">value</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>

  <span class="hl com"># Calculate Stouffer's combined Z-scores for each column</span>
  <span class="hl def">stouffer_zscores</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">apply</span><span class="hl def">(merged_data[, zscore_columns],</span> <span class="hl num">2</span><span class="hl def">,</span> <span class="hl kwa">function</span><span class="hl def">(</span><span class="hl kwc">z_scores</span><span class="hl def">) {</span>
    <span class="hl com"># Remove NA values</span>
    <span class="hl def">z_scores</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">na.omit</span><span class="hl def">(z_scores)</span>

    <span class="hl com"># Calculate the Stouffer's Z-score</span>
    <span class="hl def">stouffer_z</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">sum</span><span class="hl def">(z_scores)</span> <span class="hl opt">/</span> <span class="hl kwd">sqrt</span><span class="hl def">(</span><span class="hl kwd">length</span><span class="hl def">(z_scores))</span>

    <span class="hl kwd">return</span><span class="hl def">(stouffer_z)</span>
  <span class="hl def">})</span>

  <span class="hl com"># Remove &quot;_Zscore&quot; suffix from the Sample_ID names</span>
  <span class="hl def">sample_ids</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">sub</span><span class="hl def">(</span><span class="hl sng">&quot;_Zscore$&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;&quot;</span><span class="hl def">, zscore_columns)</span>

  <span class="hl com"># Create a new data frame with &quot;Sample_ID&quot; and the calculated Stouffer's Z-scores</span>
  <span class="hl def">pathway_id</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">unique</span><span class="hl def">(kegg_data</span><span class="hl opt">$</span><span class="hl def">pathway_id)[</span><span class="hl num">1</span><span class="hl def">]</span>
  <span class="hl def">colname</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">paste0</span><span class="hl def">(pathway_id,</span> <span class="hl sng">&quot;_Zscore&quot;</span><span class="hl def">)</span>
  <span class="hl def">results_df</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">data.frame</span><span class="hl def">(</span><span class="hl kwc">Sample_ID</span> <span class="hl def">= sample_ids)</span>
  <span class="hl def">results_df[[colname]]</span> <span class="hl kwb">&lt;-</span> <span class="hl def">stouffer_zscores</span>

  <span class="hl kwd">return</span><span class="hl def">(results_df)</span>
<span class="hl def">}</span>

<span class="hl com"># Path to the directory containing hsa*.txt files</span>
<span class="hl def">dir_path</span> <span class="hl kwb">&lt;-</span> <span class="hl sng">&quot;/Users/poddea/Desktop/ROSMAP_data_100623/TMT_proteomics/Round2/Pathway_sprcific_analysis/kegg_hsa_all&quot;</span>

<span class="hl com"># Read the zscore_AD_female data</span>
<span class="hl com">#zscore_AD_female &lt;- read.delim(&quot;zscore_AD_female.txt&quot;, header = TRUE, sep = &quot;\t&quot;)</span>

<span class="hl com"># List all hsa*.txt files in the directory</span>
<span class="hl def">kegg_files</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">list.files</span><span class="hl def">(</span><span class="hl kwc">path</span> <span class="hl def">= dir_path,</span> <span class="hl kwc">pattern</span> <span class="hl def">=</span> <span class="hl sng">&quot;^hsa.*\\.txt$&quot;</span><span class="hl def">,</span> <span class="hl kwc">full.names</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>

<span class="hl com"># Initialize an empty list to store the results</span>
<span class="hl def">all_results</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">list</span><span class="hl def">()</span>

<span class="hl com"># Process each KEGG file</span>
<span class="hl kwa">for</span> <span class="hl def">(file</span> <span class="hl kwa">in</span> <span class="hl def">kegg_files) {</span>
  <span class="hl def">result</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">process_kegg_file</span><span class="hl def">(file, zscore_CT_female)</span>
  <span class="hl def">all_results[[file]]</span> <span class="hl kwb">&lt;-</span> <span class="hl def">result</span>
<span class="hl def">}</span>

<span class="hl com"># Combine all results into one data frame by merging them on Sample_ID</span>
<span class="hl def">final_results_CT_female</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">Reduce</span><span class="hl def">(</span><span class="hl kwa">function</span><span class="hl def">(</span><span class="hl kwc">x</span><span class="hl def">,</span> <span class="hl kwc">y</span><span class="hl def">)</span> <span class="hl kwd">merge</span><span class="hl def">(x, y,</span> <span class="hl kwc">by</span> <span class="hl def">=</span> <span class="hl sng">&quot;Sample_ID&quot;</span><span class="hl def">,</span> <span class="hl kwc">all</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">), all_results)</span>

<span class="hl com"># Write the final results to a file</span>
<span class="hl kwd">write.table</span><span class="hl def">(final_results_CT_female,</span> <span class="hl sng">&quot;Pathway_zscore_CT_female_summary.txt&quot;</span><span class="hl def">,</span> <span class="hl kwc">sep</span> <span class="hl def">=</span> <span class="hl sng">&quot;\t&quot;</span><span class="hl def">,</span> <span class="hl kwc">quote</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">,</span> <span class="hl kwc">row.names</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">)</span>

<span class="hl com">###########################################################################################################################################################</span>
<span class="hl com"># Transpose the file to count 1st and 99th percentile of the Z score distributions in individual pathway</span>
<span class="hl def">final_results_CT_female_t</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">as.data.frame</span><span class="hl def">(</span><span class="hl kwd">t</span><span class="hl def">(final_results_CT_female[,</span><span class="hl opt">-</span><span class="hl num">1</span><span class="hl def">]))</span>

<span class="hl kwd">colnames</span><span class="hl def">(final_results_CT_female_t)</span> <span class="hl kwb">&lt;-</span> <span class="hl def">final_results_CT_female</span><span class="hl opt">$</span><span class="hl def">Sample_ID</span>

<span class="hl com"># Convert the row names to a new column named &quot;Pathway_ID&quot;</span>
<span class="hl def">final_results_CT_female_t</span><span class="hl opt">$</span><span class="hl def">Pathway_ID</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">rownames</span><span class="hl def">(final_results_CT_female_t)</span>

<span class="hl com"># Reorder the columns so that &quot;Pathway_ID&quot; comes first</span>
<span class="hl def">final_results_CT_female_t</span> <span class="hl kwb">&lt;-</span> <span class="hl def">final_results_CT_female_t[,</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;Pathway_ID&quot;</span><span class="hl def">,</span> <span class="hl kwd">names</span><span class="hl def">(final_results_CT_female_t)[</span><span class="hl opt">-</span><span class="hl kwd">ncol</span><span class="hl def">(final_results_CT_female_t)])]</span>

<span class="hl com"># Reset the row names as they are no longer needed</span>
<span class="hl kwd">rownames</span><span class="hl def">(final_results_CT_female_t)</span> <span class="hl kwb">&lt;-</span> <span class="hl kwa">NULL</span>

<span class="hl com"># Print the table using DT</span>
<span class="hl kwd">datatable</span><span class="hl def">(final_results_CT_female_t,</span> <span class="hl kwc">options</span> <span class="hl def">=</span> <span class="hl kwd">list</span><span class="hl def">(</span><span class="hl kwc">pageLength</span> <span class="hl def">=</span> <span class="hl num">5</span><span class="hl def">,</span> <span class="hl kwc">autoWidth</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">))</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-4-9.png" alt="plot of chunk unnamed-chunk-4" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com"># Function to calculate percentiles</span>
<span class="hl def">calc_percentiles</span> <span class="hl kwb">&lt;-</span> <span class="hl kwa">function</span><span class="hl def">(</span><span class="hl kwc">x</span><span class="hl def">) {</span>
  <span class="hl def">lower</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">quantile</span><span class="hl def">(x,</span> <span class="hl kwc">probs</span> <span class="hl def">=</span> <span class="hl num">0.01</span><span class="hl def">,</span> <span class="hl kwc">na.rm</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>
  <span class="hl def">upper</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">quantile</span><span class="hl def">(x,</span> <span class="hl kwc">probs</span> <span class="hl def">=</span> <span class="hl num">0.99</span><span class="hl def">,</span> <span class="hl kwc">na.rm</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>
  <span class="hl kwd">return</span><span class="hl def">(</span><span class="hl kwd">c</span><span class="hl def">(lower, upper))</span>
<span class="hl def">}</span>

<span class="hl com"># Apply the function to each row</span>
<span class="hl def">percentiles</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">t</span><span class="hl def">(</span><span class="hl kwd">apply</span><span class="hl def">(final_results_CT_female_t[,</span><span class="hl opt">-</span><span class="hl num">1</span><span class="hl def">],</span> <span class="hl num">1</span><span class="hl def">, calc_percentiles))</span>

<span class="hl com"># Add the percentiles to the data frame</span>
<span class="hl def">final_results_CT_female_t_percentile</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">cbind</span><span class="hl def">(final_results_CT_female_t,</span> <span class="hl kwc">lower_1st_percentile</span> <span class="hl def">= percentiles[,</span> <span class="hl num">1</span><span class="hl def">],</span> <span class="hl kwc">upper_99th_percentile</span> <span class="hl def">= percentiles[,</span> <span class="hl num">2</span><span class="hl def">])</span>

<span class="hl com"># Define a dataframe with thresold values for each gene within the control sample</span>
<span class="hl def">zscored_threshold_female</span> <span class="hl kwb">&lt;-</span> <span class="hl def">final_results_CT_female_t_percentile[,</span><span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;Pathway_ID&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;lower_1st_percentile&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;upper_99th_percentile&quot;</span><span class="hl def">)]</span>

<span class="hl com"># Write the z-scored and 5th and 95th Percentiles for Each Gene to a text file</span>
<span class="hl kwd">write.table</span><span class="hl def">(zscored_threshold_female,</span> <span class="hl sng">&quot;zscored_percentiles_pathway_data_DLPFC_contorl_female.txt&quot;</span><span class="hl def">,</span> <span class="hl kwc">sep</span><span class="hl def">=</span><span class="hl sng">&quot;\t&quot;</span><span class="hl def">,</span> <span class="hl kwc">quote</span><span class="hl def">=</span><span class="hl num">FALSE</span><span class="hl def">,</span> <span class="hl kwc">row.names</span><span class="hl def">=</span><span class="hl num">FALSE</span><span class="hl def">)</span>

<span class="hl com"># Print the table using DT</span>
<span class="hl kwd">datatable</span><span class="hl def">(zscored_threshold_female,</span> <span class="hl kwc">options</span> <span class="hl def">=</span> <span class="hl kwd">list</span><span class="hl def">(</span><span class="hl kwc">pageLength</span> <span class="hl def">=</span> <span class="hl num">5</span><span class="hl def">,</span> <span class="hl kwc">autoWidth</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">))</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-4-10.png" alt="plot of chunk unnamed-chunk-4" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">############################################################################################################################################################</span>
<span class="hl com"># Plot the data</span>

<span class="hl com"># Set rownames to Sample_ID and remove Sample_ID column</span>
<span class="hl kwd">rownames</span><span class="hl def">(final_results_CT_female)</span> <span class="hl kwb">&lt;-</span> <span class="hl def">final_results_CT_female</span><span class="hl opt">$</span><span class="hl def">Sample_ID</span>

<span class="hl def">final_results_CT_female</span> <span class="hl kwb">&lt;-</span> <span class="hl def">final_results_CT_female[,</span> <span class="hl opt">-</span><span class="hl num">1</span><span class="hl def">]</span>

<span class="hl com"># Remove columns with NA values</span>
<span class="hl def">final_results_CT_female</span> <span class="hl kwb">&lt;-</span> <span class="hl def">final_results_CT_female[,</span> <span class="hl kwd">colSums</span><span class="hl def">(</span><span class="hl kwd">is.na</span><span class="hl def">(final_results_CT_female))</span> <span class="hl opt">==</span> <span class="hl num">0</span><span class="hl def">]</span>


<span class="hl com"># Set up the color scale using RColorBrewer</span>
<span class="hl def">color_scale</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">colorRampPalette</span><span class="hl def">(</span><span class="hl kwd">brewer.pal</span><span class="hl def">(</span><span class="hl num">9</span><span class="hl def">,</span> <span class="hl sng">&quot;RdYlGn&quot;</span><span class="hl def">))</span>

<span class="hl com"># Plot the heatmap using heatmap.2</span>
<span class="hl kwd">heatmap.2</span><span class="hl def">(</span>
<span class="hl kwd">as.matrix</span><span class="hl def">(final_results_CT_female),</span>
<span class="hl kwc">Rowv</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>
<span class="hl kwc">Colv</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>
<span class="hl kwc">col</span> <span class="hl def">= color_scale,</span>
<span class="hl kwc">scale</span> <span class="hl def">=</span> <span class="hl sng">&quot;column&quot;</span><span class="hl def">,</span>  <span class="hl com"># Scale rows (genes) to Z-scores</span>
<span class="hl kwc">trace</span> <span class="hl def">=</span> <span class="hl sng">&quot;none&quot;</span><span class="hl def">,</span>  <span class="hl com"># Turn off row and column annotations</span>
<span class="hl kwc">margins</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl num">20</span><span class="hl def">,</span> <span class="hl num">20</span><span class="hl def">),</span>  <span class="hl com"># Set margins for row and column labels</span>
<span class="hl com">#col = cm.colors(256),  # Set the color map (256 colors)</span>
<span class="hl kwc">key</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>  <span class="hl com"># Include a color key</span>
<span class="hl kwc">keysize</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">,</span>  <span class="hl com"># Size of the color key</span>
<span class="hl kwc">cexCol</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">,</span>  <span class="hl com"># Set column label size</span>
<span class="hl kwc">cexRow</span> <span class="hl def">=</span> <span class="hl num">0.8</span><span class="hl def">,</span>  <span class="hl com"># Set row label size</span>
<span class="hl kwc">dendrogram</span> <span class="hl def">=</span> <span class="hl sng">&quot;both&quot;</span><span class="hl def">,</span>  <span class="hl com"># Show both row and column dendrograms</span>
<span class="hl kwc">main</span><span class="hl def">=</span><span class="hl sng">&quot;Pathway-specific Z score for CT Individuals (female)&quot;</span>
<span class="hl def">)</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-4-11.png" alt="plot of chunk unnamed-chunk-4" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">############################################################################################################################################################################</span>
<span class="hl com">############################################################################################################################################################################</span>
<span class="hl com">## CT_male</span>

<span class="hl com"># Function to process each file</span>
<span class="hl def">process_kegg_file</span> <span class="hl kwb">&lt;-</span> <span class="hl kwa">function</span><span class="hl def">(</span><span class="hl kwc">file_path</span><span class="hl def">,</span> <span class="hl kwc">zscore_data</span><span class="hl def">) {</span>
  <span class="hl com"># Read the KEGG pathway file</span>
  <span class="hl def">kegg_data</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">read.delim</span><span class="hl def">(file_path,</span> <span class="hl kwc">header</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span> <span class="hl kwc">sep</span> <span class="hl def">=</span> <span class="hl sng">&quot;\t&quot;</span><span class="hl def">)</span>

  <span class="hl com"># Merge with zscore_data</span>
   <span class="hl def">merged_data</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">merge</span><span class="hl def">(kegg_data, zscore_data,</span> <span class="hl kwc">by.x</span> <span class="hl def">=</span> <span class="hl sng">&quot;gene_symbol&quot;</span><span class="hl def">,</span> <span class="hl kwc">by.y</span> <span class="hl def">=</span> <span class="hl sng">&quot;Protein_name&quot;</span><span class="hl def">,</span> <span class="hl kwc">all.x</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>

  <span class="hl com"># Identify columns ending with &quot;_Zscore&quot;</span>
  <span class="hl def">zscore_columns</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">grep</span><span class="hl def">(</span><span class="hl sng">&quot;_Zscore$&quot;</span><span class="hl def">,</span> <span class="hl kwd">names</span><span class="hl def">(merged_data),</span> <span class="hl kwc">value</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>

  <span class="hl com"># Calculate Stouffer's combined Z-scores for each column</span>
  <span class="hl def">stouffer_zscores</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">apply</span><span class="hl def">(merged_data[, zscore_columns],</span> <span class="hl num">2</span><span class="hl def">,</span> <span class="hl kwa">function</span><span class="hl def">(</span><span class="hl kwc">z_scores</span><span class="hl def">) {</span>
    <span class="hl com"># Remove NA values</span>
    <span class="hl def">z_scores</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">na.omit</span><span class="hl def">(z_scores)</span>

    <span class="hl com"># Calculate the Stouffer's Z-score</span>
    <span class="hl def">stouffer_z</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">sum</span><span class="hl def">(z_scores)</span> <span class="hl opt">/</span> <span class="hl kwd">sqrt</span><span class="hl def">(</span><span class="hl kwd">length</span><span class="hl def">(z_scores))</span>

    <span class="hl kwd">return</span><span class="hl def">(stouffer_z)</span>
  <span class="hl def">})</span>

  <span class="hl com"># Remove &quot;_Zscore&quot; suffix from the Sample_ID names</span>
  <span class="hl def">sample_ids</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">sub</span><span class="hl def">(</span><span class="hl sng">&quot;_Zscore$&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;&quot;</span><span class="hl def">, zscore_columns)</span>

  <span class="hl com"># Create a new data frame with &quot;Sample_ID&quot; and the calculated Stouffer's Z-scores</span>
  <span class="hl def">pathway_id</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">unique</span><span class="hl def">(kegg_data</span><span class="hl opt">$</span><span class="hl def">pathway_id)[</span><span class="hl num">1</span><span class="hl def">]</span>
  <span class="hl def">colname</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">paste0</span><span class="hl def">(pathway_id,</span> <span class="hl sng">&quot;_Zscore&quot;</span><span class="hl def">)</span>
  <span class="hl def">results_df</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">data.frame</span><span class="hl def">(</span><span class="hl kwc">Sample_ID</span> <span class="hl def">= sample_ids)</span>
  <span class="hl def">results_df[[colname]]</span> <span class="hl kwb">&lt;-</span> <span class="hl def">stouffer_zscores</span>

  <span class="hl kwd">return</span><span class="hl def">(results_df)</span>
<span class="hl def">}</span>

<span class="hl com"># Path to the directory containing hsa*.txt files</span>
<span class="hl def">dir_path</span> <span class="hl kwb">&lt;-</span> <span class="hl sng">&quot;/Users/poddea/Desktop/ROSMAP_data_100623/TMT_proteomics/Round2/Pathway_sprcific_analysis/kegg_hsa_all&quot;</span>

<span class="hl com"># Read the zscore_AD_female data</span>
<span class="hl com">#zscore_AD_female &lt;- read.delim(&quot;zscore_AD_female.txt&quot;, header = TRUE, sep = &quot;\t&quot;)</span>

<span class="hl com"># List all hsa*.txt files in the directory</span>
<span class="hl def">kegg_files</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">list.files</span><span class="hl def">(</span><span class="hl kwc">path</span> <span class="hl def">= dir_path,</span> <span class="hl kwc">pattern</span> <span class="hl def">=</span> <span class="hl sng">&quot;^hsa.*\\.txt$&quot;</span><span class="hl def">,</span> <span class="hl kwc">full.names</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>

<span class="hl com"># Initialize an empty list to store the results</span>
<span class="hl def">all_results</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">list</span><span class="hl def">()</span>

<span class="hl com"># Process each KEGG file</span>
<span class="hl kwa">for</span> <span class="hl def">(file</span> <span class="hl kwa">in</span> <span class="hl def">kegg_files) {</span>
  <span class="hl def">result</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">process_kegg_file</span><span class="hl def">(file, zscore_CT_male)</span>
  <span class="hl def">all_results[[file]]</span> <span class="hl kwb">&lt;-</span> <span class="hl def">result</span>
<span class="hl def">}</span>

<span class="hl com"># Combine all results into one data frame by merging them on Sample_ID</span>
<span class="hl def">final_results_CT_male</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">Reduce</span><span class="hl def">(</span><span class="hl kwa">function</span><span class="hl def">(</span><span class="hl kwc">x</span><span class="hl def">,</span> <span class="hl kwc">y</span><span class="hl def">)</span> <span class="hl kwd">merge</span><span class="hl def">(x, y,</span> <span class="hl kwc">by</span> <span class="hl def">=</span> <span class="hl sng">&quot;Sample_ID&quot;</span><span class="hl def">,</span> <span class="hl kwc">all</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">), all_results)</span>

<span class="hl com"># Write the final results to a file</span>
<span class="hl kwd">write.table</span><span class="hl def">(final_results_CT_male,</span> <span class="hl sng">&quot;Pathway_zscore_CT_male_summary.txt&quot;</span><span class="hl def">,</span> <span class="hl kwc">sep</span> <span class="hl def">=</span> <span class="hl sng">&quot;\t&quot;</span><span class="hl def">,</span> <span class="hl kwc">quote</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">,</span> <span class="hl kwc">row.names</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">)</span>

<span class="hl com">###########################################################################################################################################################</span>
<span class="hl com"># Transpose the file to count 1st and 99th percentile of the Z score distributions in individual pathway</span>
<span class="hl def">final_results_CT_male_t</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">as.data.frame</span><span class="hl def">(</span><span class="hl kwd">t</span><span class="hl def">(final_results_CT_male[,</span><span class="hl opt">-</span><span class="hl num">1</span><span class="hl def">]))</span>

<span class="hl kwd">colnames</span><span class="hl def">(final_results_CT_male_t)</span> <span class="hl kwb">&lt;-</span> <span class="hl def">final_results_CT_male</span><span class="hl opt">$</span><span class="hl def">Sample_ID</span>

<span class="hl com"># Convert the row names to a new column named &quot;Pathway_ID&quot;</span>
<span class="hl def">final_results_CT_male_t</span><span class="hl opt">$</span><span class="hl def">Pathway_ID</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">rownames</span><span class="hl def">(final_results_CT_male_t)</span>

<span class="hl com"># Reorder the columns so that &quot;Pathway_ID&quot; comes first</span>
<span class="hl def">final_results_CT_male_t</span> <span class="hl kwb">&lt;-</span> <span class="hl def">final_results_CT_male_t[,</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;Pathway_ID&quot;</span><span class="hl def">,</span> <span class="hl kwd">names</span><span class="hl def">(final_results_CT_male_t)[</span><span class="hl opt">-</span><span class="hl kwd">ncol</span><span class="hl def">(final_results_CT_male_t)])]</span>

<span class="hl com"># Reset the row names as they are no longer needed</span>
<span class="hl kwd">rownames</span><span class="hl def">(final_results_CT_male_t)</span> <span class="hl kwb">&lt;-</span> <span class="hl kwa">NULL</span>


<span class="hl com"># Print the table using DT</span>
<span class="hl kwd">datatable</span><span class="hl def">(final_results_CT_male_t,</span> <span class="hl kwc">options</span> <span class="hl def">=</span> <span class="hl kwd">list</span><span class="hl def">(</span><span class="hl kwc">pageLength</span> <span class="hl def">=</span> <span class="hl num">5</span><span class="hl def">,</span> <span class="hl kwc">autoWidth</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">))</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-4-12.png" alt="plot of chunk unnamed-chunk-4" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com"># Function to calculate percentiles</span>
<span class="hl def">calc_percentiles</span> <span class="hl kwb">&lt;-</span> <span class="hl kwa">function</span><span class="hl def">(</span><span class="hl kwc">x</span><span class="hl def">) {</span>
  <span class="hl def">lower</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">quantile</span><span class="hl def">(x,</span> <span class="hl kwc">probs</span> <span class="hl def">=</span> <span class="hl num">0.01</span><span class="hl def">,</span> <span class="hl kwc">na.rm</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>
  <span class="hl def">upper</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">quantile</span><span class="hl def">(x,</span> <span class="hl kwc">probs</span> <span class="hl def">=</span> <span class="hl num">0.99</span><span class="hl def">,</span> <span class="hl kwc">na.rm</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>
  <span class="hl kwd">return</span><span class="hl def">(</span><span class="hl kwd">c</span><span class="hl def">(lower, upper))</span>
<span class="hl def">}</span>

<span class="hl com"># Apply the function to each row</span>
<span class="hl def">percentiles</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">t</span><span class="hl def">(</span><span class="hl kwd">apply</span><span class="hl def">(final_results_CT_male_t[,</span><span class="hl opt">-</span><span class="hl num">1</span><span class="hl def">],</span> <span class="hl num">1</span><span class="hl def">, calc_percentiles))</span>

<span class="hl com"># Add the percentiles to the data frame</span>
<span class="hl def">final_results_CT_male_t_percentile</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">cbind</span><span class="hl def">(final_results_CT_male_t,</span> <span class="hl kwc">lower_1st_percentile</span> <span class="hl def">= percentiles[,</span> <span class="hl num">1</span><span class="hl def">],</span> <span class="hl kwc">upper_99th_percentile</span> <span class="hl def">= percentiles[,</span> <span class="hl num">2</span><span class="hl def">])</span>

<span class="hl com"># Define a dataframe with thresold values for each gene within the control sample</span>
<span class="hl def">zscored_threshold_male</span> <span class="hl kwb">&lt;-</span> <span class="hl def">final_results_CT_male_t_percentile[,</span><span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;Pathway_ID&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;lower_1st_percentile&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;upper_99th_percentile&quot;</span><span class="hl def">)]</span>

<span class="hl com"># Write the z-scored and 5th and 95th Percentiles for Each Gene to a text file</span>
<span class="hl kwd">write.table</span><span class="hl def">(zscored_threshold_male,</span> <span class="hl sng">&quot;zscored_percentiles_pathway_data_DLPFC_contorl_male.txt&quot;</span><span class="hl def">,</span> <span class="hl kwc">sep</span><span class="hl def">=</span><span class="hl sng">&quot;\t&quot;</span><span class="hl def">,</span> <span class="hl kwc">quote</span><span class="hl def">=</span><span class="hl num">FALSE</span><span class="hl def">,</span> <span class="hl kwc">row.names</span><span class="hl def">=</span><span class="hl num">FALSE</span><span class="hl def">)</span>

<span class="hl com"># Print the table using DT</span>
<span class="hl kwd">datatable</span><span class="hl def">(zscored_threshold_male,</span> <span class="hl kwc">options</span> <span class="hl def">=</span> <span class="hl kwd">list</span><span class="hl def">(</span><span class="hl kwc">pageLength</span> <span class="hl def">=</span> <span class="hl num">5</span><span class="hl def">,</span> <span class="hl kwc">autoWidth</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">))</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-4-13.png" alt="plot of chunk unnamed-chunk-4" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">############################################################################################################################################################</span>
<span class="hl com"># Plot the data</span>

<span class="hl com"># Set rownames to Sample_ID and remove Sample_ID column</span>
<span class="hl kwd">rownames</span><span class="hl def">(final_results_CT_male)</span> <span class="hl kwb">&lt;-</span> <span class="hl def">final_results_CT_male</span><span class="hl opt">$</span><span class="hl def">Sample_ID</span>

<span class="hl def">final_results_CT_male</span> <span class="hl kwb">&lt;-</span> <span class="hl def">final_results_CT_male[,</span> <span class="hl opt">-</span><span class="hl num">1</span><span class="hl def">]</span>

<span class="hl com"># Remove columns with NA values</span>
<span class="hl def">final_results_CT_male</span> <span class="hl kwb">&lt;-</span> <span class="hl def">final_results_CT_male[,</span> <span class="hl kwd">colSums</span><span class="hl def">(</span><span class="hl kwd">is.na</span><span class="hl def">(final_results_CT_male))</span> <span class="hl opt">==</span> <span class="hl num">0</span><span class="hl def">]</span>


<span class="hl com"># Set up the color scale using RColorBrewer</span>
<span class="hl def">color_scale</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">colorRampPalette</span><span class="hl def">(</span><span class="hl kwd">brewer.pal</span><span class="hl def">(</span><span class="hl num">9</span><span class="hl def">,</span> <span class="hl sng">&quot;RdYlGn&quot;</span><span class="hl def">))</span>

<span class="hl com"># Plot the heatmap using heatmap.2</span>
<span class="hl kwd">heatmap.2</span><span class="hl def">(</span>
<span class="hl kwd">as.matrix</span><span class="hl def">(final_results_CT_male),</span>
<span class="hl kwc">Rowv</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>
<span class="hl kwc">Colv</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>
<span class="hl kwc">col</span> <span class="hl def">= color_scale,</span>
<span class="hl kwc">scale</span> <span class="hl def">=</span> <span class="hl sng">&quot;column&quot;</span><span class="hl def">,</span>  <span class="hl com"># Scale rows (genes) to Z-scores</span>
<span class="hl kwc">trace</span> <span class="hl def">=</span> <span class="hl sng">&quot;none&quot;</span><span class="hl def">,</span>  <span class="hl com"># Turn off row and column annotations</span>
<span class="hl kwc">margins</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl num">20</span><span class="hl def">,</span> <span class="hl num">20</span><span class="hl def">),</span>  <span class="hl com"># Set margins for row and column labels</span>
<span class="hl com">#col = cm.colors(256),  # Set the color map (256 colors)</span>
<span class="hl kwc">key</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>  <span class="hl com"># Include a color key</span>
<span class="hl kwc">keysize</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">,</span>  <span class="hl com"># Size of the color key</span>
<span class="hl kwc">cexCol</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">,</span>  <span class="hl com"># Set column label size</span>
<span class="hl kwc">cexRow</span> <span class="hl def">=</span> <span class="hl num">0.8</span><span class="hl def">,</span>  <span class="hl com"># Set row label size</span>
<span class="hl kwc">dendrogram</span> <span class="hl def">=</span> <span class="hl sng">&quot;both&quot;</span><span class="hl def">,</span>  <span class="hl com"># Show both row and column dendrograms</span>
<span class="hl kwc">main</span><span class="hl def">=</span><span class="hl sng">&quot;Pathway-specific Z score for CT Individuals (male)&quot;</span>
<span class="hl def">)</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-4-14.png" alt="plot of chunk unnamed-chunk-4" class="plot" /></div></div>

<p>Identfy significant pathways for AD and AsymAD (male and female) using the Z-score thresold (percentile upper and lower bound) from control samples</p>

<div class="chunk" id="unnamed-chunk-5"><div class="rcode"><div class="source"><pre class="knitr r"><span class="hl kwd">library</span><span class="hl def">(ggplot2)</span>
<span class="hl kwd">library</span><span class="hl def">(tidyr)</span>
<span class="hl kwd">library</span><span class="hl def">(ggdendro)</span>

<span class="hl com">############################################################################################################################################################################</span>


<span class="hl com">## AD_female</span>

<span class="hl com"># Merge both the data frame by Pathway_ID</span>
<span class="hl def">merged_data_female_AD</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">merge</span><span class="hl def">(zscored_threshold_female, final_results_AD_female_t,</span> <span class="hl kwc">by</span> <span class="hl def">=</span> <span class="hl sng">&quot;Pathway_ID&quot;</span><span class="hl def">)</span>

<span class="hl com"># Initialize an empty data frame to store results</span>
<span class="hl def">sig_pathway_AD_female</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">data.frame</span><span class="hl def">(</span><span class="hl kwc">Pathway_ID</span> <span class="hl def">= merged_data_female_AD</span><span class="hl opt">$</span><span class="hl def">Pathway_ID)</span>


<span class="hl com"># Iterate over each column in final_results_AD_fefemale_t (excluding the Pathway_ID)</span>
<span class="hl kwa">for</span> <span class="hl def">(col_name</span> <span class="hl kwa">in</span> <span class="hl kwd">colnames</span><span class="hl def">(final_results_AD_female_t)[</span><span class="hl opt">-</span><span class="hl num">1</span><span class="hl def">]) {</span>

  <span class="hl com"># Apply the conditions and store the actual value or 0 in a new column in sig_pathway_AD_female</span>
  <span class="hl def">sig_pathway_AD_female[[col_name]]</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">ifelse</span><span class="hl def">(</span>
    <span class="hl def">merged_data_female_AD[[col_name]]</span> <span class="hl opt">&lt;</span> <span class="hl def">merged_data_female_AD</span><span class="hl opt">$</span><span class="hl def">lower_1st_percentile,</span>
    <span class="hl def">merged_data_female_AD[[col_name]],</span>  <span class="hl com"># Retrieve the value if it's less than the lower 1st percentile</span>
    <span class="hl kwd">ifelse</span><span class="hl def">(</span>
      <span class="hl def">merged_data_female_AD[[col_name]]</span> <span class="hl opt">&gt;</span> <span class="hl def">merged_data_female_AD</span><span class="hl opt">$</span><span class="hl def">upper_99th_percentile,</span>
      <span class="hl def">merged_data_female_AD[[col_name]],</span>  <span class="hl com"># Retrieve the value if it's greater than the upper 99th percentile</span>
      <span class="hl num">0</span>  <span class="hl com"># Otherwise, set to 0</span>
    <span class="hl def">)</span>
  <span class="hl def">)</span>
<span class="hl def">}</span>

<span class="hl com"># Select only numeric columns (assuming all columns except Pathway_ID are numeric)</span>
<span class="hl def">numeric_columns</span> <span class="hl kwb">&lt;-</span> <span class="hl def">sig_pathway_AD_female[,</span> <span class="hl kwd">sapply</span><span class="hl def">(sig_pathway_AD_female, is.numeric)]</span>

<span class="hl com"># Calculate UP_count (count of positive values) for each row</span>
<span class="hl def">sig_pathway_AD_female</span><span class="hl opt">$</span><span class="hl def">UP_Count</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">rowSums</span><span class="hl def">(numeric_columns</span> <span class="hl opt">&gt;</span> <span class="hl num">0</span><span class="hl def">)</span>

<span class="hl com"># Calculate DOWN_count (count of negative values) for each row</span>
<span class="hl def">sig_pathway_AD_female</span><span class="hl opt">$</span><span class="hl def">DOWN_Count</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">rowSums</span><span class="hl def">(numeric_columns</span> <span class="hl opt">&lt;</span> <span class="hl num">0</span><span class="hl def">)</span>

<span class="hl com"># Write the output to a .txt file</span>
<span class="hl kwd">write.table</span><span class="hl def">(sig_pathway_AD_female,</span> <span class="hl sng">&quot;sig_pathway_AD_female.txt&quot;</span><span class="hl def">,</span> <span class="hl kwc">sep</span> <span class="hl def">=</span> <span class="hl sng">&quot;\t&quot;</span><span class="hl def">,</span> <span class="hl kwc">quote</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">,</span> <span class="hl kwc">row.names</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">)</span>

<span class="hl com"># Print the table using DT</span>
<span class="hl kwd">datatable</span><span class="hl def">(sig_pathway_AD_female,</span> <span class="hl kwc">options</span> <span class="hl def">=</span> <span class="hl kwd">list</span><span class="hl def">(</span><span class="hl kwc">pageLength</span> <span class="hl def">=</span> <span class="hl num">5</span><span class="hl def">,</span> <span class="hl kwc">autoWidth</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">))</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-5-1.png" alt="plot of chunk unnamed-chunk-5" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">###################################################################################################################################################</span>
<span class="hl com"># Plot the data</span>

<span class="hl com"># Remove the UP_Count and DOWN_Count columns</span>
<span class="hl def">heatmap_data_AD_female</span> <span class="hl kwb">&lt;-</span> <span class="hl def">sig_pathway_AD_female[,</span> <span class="hl opt">!</span><span class="hl def">(</span><span class="hl kwd">colnames</span><span class="hl def">(sig_pathway_AD_female)</span> <span class="hl opt">%in%</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;UP_Count&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;DOWN_Count&quot;</span><span class="hl def">))]</span>

<span class="hl com"># Set the row names to Pathway_ID and remove the Pathway_ID column</span>
<span class="hl kwd">row.names</span><span class="hl def">(heatmap_data_AD_female)</span> <span class="hl kwb">&lt;-</span> <span class="hl def">sig_pathway_AD_female</span><span class="hl opt">$</span><span class="hl def">Pathway_ID</span>
<span class="hl def">heatmap_data_AD_female</span><span class="hl opt">$</span><span class="hl def">Pathway_ID</span> <span class="hl kwb">&lt;-</span> <span class="hl kwa">NULL</span>

<span class="hl com"># Check and handle NA/NaN/Inf values</span>
<span class="hl def">heatmap_data_AD_female[</span><span class="hl kwd">is.na</span><span class="hl def">(heatmap_data_AD_female)]</span> <span class="hl kwb">&lt;-</span> <span class="hl num">0</span>  <span class="hl com"># Replace NA values with 0</span>
<span class="hl def">heatmap_data_AD_female[</span><span class="hl kwd">is.infinite</span><span class="hl def">(heatmap_data_AD_female)]</span> <span class="hl kwb">&lt;-</span> <span class="hl num">0</span>  <span class="hl com"># Replace Inf/-Inf values with 0</span>
</pre></div>
<div class="error"><pre class="knitr r">## Error in is.infinite(heatmap_data_AD_female): default method not implemented for type 'list'
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl com"># Convert the data to numeric type (if it's not already)</span>
<span class="hl def">heatmap_data_AD_female</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">as.matrix</span><span class="hl def">(heatmap_data_AD_female)</span>

<span class="hl com"># Define the color palette</span>
<span class="hl def">color_palette</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">colorRampPalette</span><span class="hl def">(</span><span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;red&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;white&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;blue&quot;</span><span class="hl def">))(</span><span class="hl num">100</span><span class="hl def">)</span>

<span class="hl com"># Generate the heatmap</span>
<span class="hl def">out_heatmap_AD_female</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">heatmap.2</span><span class="hl def">(</span>
  <span class="hl def">heatmap_data_AD_female,</span>              <span class="hl com"># Data matrix</span>
  <span class="hl kwc">Rowv</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>                 <span class="hl com"># Row dendrogram</span>
  <span class="hl kwc">Colv</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>                 <span class="hl com"># Column dendrogram</span>
  <span class="hl kwc">col</span> <span class="hl def">= color_palette,</span>         <span class="hl com"># Color palette</span>
  <span class="hl kwc">scale</span> <span class="hl def">=</span> <span class="hl sng">&quot;row&quot;</span><span class="hl def">,</span>              <span class="hl com"># No scaling</span>
  <span class="hl kwc">trace</span> <span class="hl def">=</span> <span class="hl sng">&quot;none&quot;</span><span class="hl def">,</span>              <span class="hl com"># Turn off trace lines</span>
  <span class="hl kwc">margins</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl num">10</span><span class="hl def">,</span> <span class="hl num">10</span><span class="hl def">),</span>         <span class="hl com"># Set margins for labels</span>
  <span class="hl kwc">key</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>                  <span class="hl com"># Include color key</span>
  <span class="hl kwc">key.title</span> <span class="hl def">=</span> <span class="hl sng">&quot;Z-Score&quot;</span><span class="hl def">,</span>       <span class="hl com"># Key title</span>
  <span class="hl kwc">keysize</span> <span class="hl def">=</span> <span class="hl num">1.5</span><span class="hl def">,</span>               <span class="hl com"># Size of the key</span>
  <span class="hl kwc">cexRow</span> <span class="hl def">=</span> <span class="hl num">0.8</span><span class="hl def">,</span>                <span class="hl com"># Size of row labels</span>
  <span class="hl kwc">cexCol</span> <span class="hl def">=</span> <span class="hl num">0.8</span><span class="hl def">,</span>                <span class="hl com"># Size of column labels</span>
  <span class="hl kwc">dendrogram</span> <span class="hl def">=</span> <span class="hl sng">&quot;both&quot;</span><span class="hl def">,</span>         <span class="hl com"># Show both row and column dendrograms</span>
  <span class="hl kwc">main</span> <span class="hl def">=</span> <span class="hl sng">&quot;Pathway Specific Z-Scores Distribution (AD female)&quot;</span>  <span class="hl com"># Main title</span>
<span class="hl def">)</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-5-2.png" alt="plot of chunk unnamed-chunk-5" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com"># Plot the dendogram</span>
<span class="hl kwd">ggdendrogram</span><span class="hl def">(out_heatmap_AD_female</span><span class="hl opt">$</span><span class="hl def">colDendrogram,</span> <span class="hl kwc">theme_dendro</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">)</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-5-3.png" alt="plot of chunk unnamed-chunk-5" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com"># Convert data from wide to long format</span>
<span class="hl def">long_sig_pathway_AD_female</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">gather</span><span class="hl def">(sig_pathway_AD_female,</span> <span class="hl kwc">key</span> <span class="hl def">=</span> <span class="hl sng">&quot;Direction&quot;</span><span class="hl def">,</span> <span class="hl kwc">value</span> <span class="hl def">=</span> <span class="hl sng">&quot;Count&quot;</span><span class="hl def">, UP_Count, DOWN_Count)</span>

<span class="hl com"># Create the line plot</span>
<span class="hl kwd">ggplot</span><span class="hl def">(long_sig_pathway_AD_female,</span> <span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">x</span> <span class="hl def">= Pathway_ID,</span> <span class="hl kwc">y</span> <span class="hl def">= Count,</span> <span class="hl kwc">color</span> <span class="hl def">= Direction,</span> <span class="hl kwc">group</span> <span class="hl def">= Direction))</span> <span class="hl opt">+</span>
  <span class="hl kwd">geom_line</span><span class="hl def">(</span><span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">)</span> <span class="hl opt">+</span>    <span class="hl com"># Add lines between points</span>
  <span class="hl kwd">geom_point</span><span class="hl def">(</span><span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">2</span><span class="hl def">)</span> <span class="hl opt">+</span>   <span class="hl com"># Add points at each Pathway_ID</span>
  <span class="hl kwd">theme_minimal</span><span class="hl def">()</span> <span class="hl opt">+</span>
  <span class="hl kwd">labs</span><span class="hl def">(</span><span class="hl kwc">title</span> <span class="hl def">=</span> <span class="hl sng">&quot;UP and DOWN Count for Each Pathway (AD female)&quot;</span><span class="hl def">,</span> <span class="hl kwc">x</span> <span class="hl def">=</span> <span class="hl sng">&quot;Pathway ID&quot;</span><span class="hl def">,</span> <span class="hl kwc">y</span> <span class="hl def">=</span> <span class="hl sng">&quot;Count&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl kwd">theme</span><span class="hl def">(</span><span class="hl kwc">axis.text.x</span> <span class="hl def">=</span> <span class="hl kwd">element_text</span><span class="hl def">(</span><span class="hl kwc">angle</span> <span class="hl def">=</span> <span class="hl num">90</span><span class="hl def">,</span> <span class="hl kwc">hjust</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">,</span> <span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">8</span><span class="hl def">))</span> <span class="hl opt">+</span>  <span class="hl com"># Rotate x-axis labels if needed</span>
  <span class="hl kwd">scale_color_manual</span><span class="hl def">(</span><span class="hl kwc">values</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;UP_Count&quot;</span> <span class="hl def">=</span> <span class="hl sng">&quot;lightblue&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;DOWN_Count&quot;</span> <span class="hl def">=</span> <span class="hl sng">&quot;pink&quot;</span><span class="hl def">))</span>  <span class="hl com"># Customize colors</span>
</pre></div>
<div class="warning"><pre class="knitr r">## Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
##  Please use `linewidth` instead.
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
## generated.
</pre></div>
<div class="warning"><pre class="knitr r">## Warning: Removed 16 rows containing missing values or values outside the scale range
## (`geom_point()`).
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-5-4.png" alt="plot of chunk unnamed-chunk-5" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">############################################################################################################################################################################</span>
<span class="hl com">############################################################################################################################################################################</span>

<span class="hl com">## AD_male</span>

<span class="hl com"># Merge both the data frame by Pathway_ID</span>
<span class="hl def">merged_data_male_AD</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">merge</span><span class="hl def">(zscored_threshold_male, final_results_AD_male_t,</span> <span class="hl kwc">by</span> <span class="hl def">=</span> <span class="hl sng">&quot;Pathway_ID&quot;</span><span class="hl def">)</span>

<span class="hl com"># Initialize an empty data frame to store results</span>
<span class="hl def">sig_pathway_AD_male</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">data.frame</span><span class="hl def">(</span><span class="hl kwc">Pathway_ID</span> <span class="hl def">= merged_data_male_AD</span><span class="hl opt">$</span><span class="hl def">Pathway_ID)</span>


<span class="hl com"># Iterate over each column in final_results_AD_female_t (excluding the Pathway_ID)</span>
<span class="hl kwa">for</span> <span class="hl def">(col_name</span> <span class="hl kwa">in</span> <span class="hl kwd">colnames</span><span class="hl def">(final_results_AD_male_t)[</span><span class="hl opt">-</span><span class="hl num">1</span><span class="hl def">]) {</span>

  <span class="hl com"># Apply the conditions and store the actual value or 0 in a new column in sig_pathway_AD_male</span>
  <span class="hl def">sig_pathway_AD_male[[col_name]]</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">ifelse</span><span class="hl def">(</span>
    <span class="hl def">merged_data_male_AD[[col_name]]</span> <span class="hl opt">&lt;</span> <span class="hl def">merged_data_male_AD</span><span class="hl opt">$</span><span class="hl def">lower_1st_percentile,</span>
    <span class="hl def">merged_data_male_AD[[col_name]],</span>  <span class="hl com"># Retrieve the value if it's less than the lower 1st percentile</span>
    <span class="hl kwd">ifelse</span><span class="hl def">(</span>
      <span class="hl def">merged_data_male_AD[[col_name]]</span> <span class="hl opt">&gt;</span> <span class="hl def">merged_data_male_AD</span><span class="hl opt">$</span><span class="hl def">upper_99th_percentile,</span>
      <span class="hl def">merged_data_male_AD[[col_name]],</span>  <span class="hl com"># Retrieve the value if it's greater than the upper 99th percentile</span>
      <span class="hl num">0</span>  <span class="hl com"># Otherwise, set to 0</span>
    <span class="hl def">)</span>
  <span class="hl def">)</span>
<span class="hl def">}</span>

<span class="hl com"># Select only numeric columns (assuming all columns except Pathway_ID are numeric)</span>
<span class="hl def">numeric_columns</span> <span class="hl kwb">&lt;-</span> <span class="hl def">sig_pathway_AD_male[,</span> <span class="hl kwd">sapply</span><span class="hl def">(sig_pathway_AD_male, is.numeric)]</span>

<span class="hl com"># Calculate UP_count (count of positive values) for each row</span>
<span class="hl def">sig_pathway_AD_male</span><span class="hl opt">$</span><span class="hl def">UP_Count</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">rowSums</span><span class="hl def">(numeric_columns</span> <span class="hl opt">&gt;</span> <span class="hl num">0</span><span class="hl def">)</span>

<span class="hl com"># Calculate DOWN_count (count of negative values) for each row</span>
<span class="hl def">sig_pathway_AD_male</span><span class="hl opt">$</span><span class="hl def">DOWN_Count</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">rowSums</span><span class="hl def">(numeric_columns</span> <span class="hl opt">&lt;</span> <span class="hl num">0</span><span class="hl def">)</span>

<span class="hl com"># Write the output to a .txt file</span>
<span class="hl kwd">write.table</span><span class="hl def">(sig_pathway_AD_male,</span> <span class="hl sng">&quot;sig_pathway_AD_male.txt&quot;</span><span class="hl def">,</span> <span class="hl kwc">sep</span> <span class="hl def">=</span> <span class="hl sng">&quot;\t&quot;</span><span class="hl def">,</span> <span class="hl kwc">quote</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">,</span> <span class="hl kwc">row.names</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">)</span>

<span class="hl com"># Print the table using DT</span>
<span class="hl kwd">datatable</span><span class="hl def">(sig_pathway_AD_male,</span> <span class="hl kwc">options</span> <span class="hl def">=</span> <span class="hl kwd">list</span><span class="hl def">(</span><span class="hl kwc">pageLength</span> <span class="hl def">=</span> <span class="hl num">5</span><span class="hl def">,</span> <span class="hl kwc">autoWidth</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">))</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-5-5.png" alt="plot of chunk unnamed-chunk-5" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">###################################################################################################################################################</span>
<span class="hl com"># Plot the data</span>

<span class="hl com"># Remove the UP_Count and DOWN_Count columns</span>
<span class="hl def">heatmap_data_AD_male</span> <span class="hl kwb">&lt;-</span> <span class="hl def">sig_pathway_AD_male[,</span> <span class="hl opt">!</span><span class="hl def">(</span><span class="hl kwd">colnames</span><span class="hl def">(sig_pathway_AD_male)</span> <span class="hl opt">%in%</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;UP_Count&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;DOWN_Count&quot;</span><span class="hl def">))]</span>

<span class="hl com"># Set the row names to Pathway_ID and remove the Pathway_ID column</span>
<span class="hl kwd">row.names</span><span class="hl def">(heatmap_data_AD_male)</span> <span class="hl kwb">&lt;-</span> <span class="hl def">sig_pathway_AD_male</span><span class="hl opt">$</span><span class="hl def">Pathway_ID</span>
<span class="hl def">heatmap_data_AD_male</span><span class="hl opt">$</span><span class="hl def">Pathway_ID</span> <span class="hl kwb">&lt;-</span> <span class="hl kwa">NULL</span>

<span class="hl com"># Check and handle NA/NaN/Inf values</span>
<span class="hl def">heatmap_data_AD_male[</span><span class="hl kwd">is.na</span><span class="hl def">(heatmap_data_AD_male)]</span> <span class="hl kwb">&lt;-</span> <span class="hl num">0</span>  <span class="hl com"># Replace NA values with 0</span>
<span class="hl def">heatmap_data_AD_male[</span><span class="hl kwd">is.infinite</span><span class="hl def">(heatmap_data_AD_male)]</span> <span class="hl kwb">&lt;-</span> <span class="hl num">0</span>  <span class="hl com"># Replace Inf/-Inf values with 0</span>
</pre></div>
<div class="error"><pre class="knitr r">## Error in is.infinite(heatmap_data_AD_male): default method not implemented for type 'list'
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl com"># Convert the data to numeric type (if it's not already)</span>
<span class="hl def">heatmap_data_AD_male</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">as.matrix</span><span class="hl def">(heatmap_data_AD_male)</span>

<span class="hl com"># Define the color palette</span>
<span class="hl def">color_palette</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">colorRampPalette</span><span class="hl def">(</span><span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;red&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;white&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;blue&quot;</span><span class="hl def">))(</span><span class="hl num">100</span><span class="hl def">)</span>

<span class="hl com"># Generate the heatmap</span>
<span class="hl def">out_heatmap_AD_male</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">heatmap.2</span><span class="hl def">(</span>
  <span class="hl def">heatmap_data_AD_male,</span>              <span class="hl com"># Data matrix</span>
  <span class="hl kwc">Rowv</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>                 <span class="hl com"># Row dendrogram</span>
  <span class="hl kwc">Colv</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>                 <span class="hl com"># Column dendrogram</span>
  <span class="hl kwc">col</span> <span class="hl def">= color_palette,</span>         <span class="hl com"># Color palette</span>
  <span class="hl kwc">scale</span> <span class="hl def">=</span> <span class="hl sng">&quot;row&quot;</span><span class="hl def">,</span>              <span class="hl com"># No scaling</span>
  <span class="hl kwc">trace</span> <span class="hl def">=</span> <span class="hl sng">&quot;none&quot;</span><span class="hl def">,</span>              <span class="hl com"># Turn off trace lines</span>
  <span class="hl kwc">margins</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl num">10</span><span class="hl def">,</span> <span class="hl num">10</span><span class="hl def">),</span>         <span class="hl com"># Set margins for labels</span>
  <span class="hl kwc">key</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>                  <span class="hl com"># Include color key</span>
  <span class="hl kwc">key.title</span> <span class="hl def">=</span> <span class="hl sng">&quot;Z-Score&quot;</span><span class="hl def">,</span>       <span class="hl com"># Key title</span>
  <span class="hl kwc">keysize</span> <span class="hl def">=</span> <span class="hl num">1.5</span><span class="hl def">,</span>               <span class="hl com"># Size of the key</span>
  <span class="hl kwc">cexRow</span> <span class="hl def">=</span> <span class="hl num">0.8</span><span class="hl def">,</span>                <span class="hl com"># Size of row labels</span>
  <span class="hl kwc">cexCol</span> <span class="hl def">=</span> <span class="hl num">0.8</span><span class="hl def">,</span>                <span class="hl com"># Size of column labels</span>
  <span class="hl kwc">dendrogram</span> <span class="hl def">=</span> <span class="hl sng">&quot;both&quot;</span><span class="hl def">,</span>         <span class="hl com"># Show both row and column dendrograms</span>
  <span class="hl kwc">main</span> <span class="hl def">=</span> <span class="hl sng">&quot;Pathway Specific Z-Scores Distribution (AD male)&quot;</span>  <span class="hl com"># Main title</span>
<span class="hl def">)</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-5-6.png" alt="plot of chunk unnamed-chunk-5" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com"># Plot the dendogram</span>
<span class="hl kwd">ggdendrogram</span><span class="hl def">(out_heatmap_AD_male</span><span class="hl opt">$</span><span class="hl def">colDendrogram,</span> <span class="hl kwc">theme_dendro</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">)</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-5-7.png" alt="plot of chunk unnamed-chunk-5" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com"># Convert data from wide to long format</span>
<span class="hl def">long_sig_pathway_AD_male</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">gather</span><span class="hl def">(sig_pathway_AD_male,</span> <span class="hl kwc">key</span> <span class="hl def">=</span> <span class="hl sng">&quot;Direction&quot;</span><span class="hl def">,</span> <span class="hl kwc">value</span> <span class="hl def">=</span> <span class="hl sng">&quot;Count&quot;</span><span class="hl def">, UP_Count, DOWN_Count)</span>

<span class="hl com"># Create the line plot</span>
<span class="hl kwd">ggplot</span><span class="hl def">(long_sig_pathway_AD_male,</span> <span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">x</span> <span class="hl def">= Pathway_ID,</span> <span class="hl kwc">y</span> <span class="hl def">= Count,</span> <span class="hl kwc">color</span> <span class="hl def">= Direction,</span> <span class="hl kwc">group</span> <span class="hl def">= Direction))</span> <span class="hl opt">+</span>
  <span class="hl kwd">geom_line</span><span class="hl def">(</span><span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">)</span> <span class="hl opt">+</span>    <span class="hl com"># Add lines between points</span>
  <span class="hl kwd">geom_point</span><span class="hl def">(</span><span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">2</span><span class="hl def">)</span> <span class="hl opt">+</span>   <span class="hl com"># Add points at each Pathway_ID</span>
  <span class="hl kwd">theme_minimal</span><span class="hl def">()</span> <span class="hl opt">+</span>
  <span class="hl kwd">labs</span><span class="hl def">(</span><span class="hl kwc">title</span> <span class="hl def">=</span> <span class="hl sng">&quot;UP and DOWN Count for Each Pathway (AD male)&quot;</span><span class="hl def">,</span> <span class="hl kwc">x</span> <span class="hl def">=</span> <span class="hl sng">&quot;Pathway ID&quot;</span><span class="hl def">,</span> <span class="hl kwc">y</span> <span class="hl def">=</span> <span class="hl sng">&quot;Count&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl kwd">theme</span><span class="hl def">(</span><span class="hl kwc">axis.text.x</span> <span class="hl def">=</span> <span class="hl kwd">element_text</span><span class="hl def">(</span><span class="hl kwc">angle</span> <span class="hl def">=</span> <span class="hl num">90</span><span class="hl def">,</span> <span class="hl kwc">hjust</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">,</span> <span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">8</span><span class="hl def">))</span> <span class="hl opt">+</span>  <span class="hl com"># Rotate x-axis labels if needed</span>
  <span class="hl kwd">scale_color_manual</span><span class="hl def">(</span><span class="hl kwc">values</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;UP_Count&quot;</span> <span class="hl def">=</span> <span class="hl sng">&quot;lightblue&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;DOWN_Count&quot;</span> <span class="hl def">=</span> <span class="hl sng">&quot;pink&quot;</span><span class="hl def">))</span>  <span class="hl com"># Customize colors</span>
</pre></div>
<div class="warning"><pre class="knitr r">## Warning: Removed 16 rows containing missing values or values outside the scale range
## (`geom_point()`).
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-5-8.png" alt="plot of chunk unnamed-chunk-5" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">############################################################################################################################################################################</span>
<span class="hl com">############################################################################################################################################################################</span>

<span class="hl com">## AsymAD_female</span>

<span class="hl com"># Merge both the data frame by Pathway_ID</span>
<span class="hl def">merged_data_female_AsymAD</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">merge</span><span class="hl def">(zscored_threshold_female, final_results_AsymAD_female_t,</span> <span class="hl kwc">by</span> <span class="hl def">=</span> <span class="hl sng">&quot;Pathway_ID&quot;</span><span class="hl def">)</span>

<span class="hl com"># Initialize an empty data frame to store results</span>
<span class="hl def">sig_pathway_AsymAD_female</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">data.frame</span><span class="hl def">(</span><span class="hl kwc">Pathway_ID</span> <span class="hl def">= merged_data_female_AsymAD</span><span class="hl opt">$</span><span class="hl def">Pathway_ID)</span>

<span class="hl com"># Iterate over each column in B (excluding the Pathway_ID)</span>
<span class="hl com">#for (col_name in colnames(final_results_AsymAD_female_t)[-1]) {</span>

  <span class="hl com"># Apply the conditions and store the result in a new column in Output</span>
  <span class="hl com">#sig_pathway_AsymAD_female[[col_name]] &lt;- ifelse(merged_data_female_AsymAD[[col_name]] &lt; merged_data_female_AsymAD$lower_1st_percentile, </span>
                               <span class="hl com">#&quot;DOWN&quot;, </span>
                               <span class="hl com">#ifelse(merged_data_female_AsymAD[[col_name]] &gt; merged_data_female_AsymAD$upper_99th_percentile, </span>
                                      <span class="hl com">#&quot;UP&quot;, </span>
                                      <span class="hl com">#&quot;0&quot;))</span>
<span class="hl com">#}</span>

<span class="hl com"># Add a column to count the number of &quot;UP&quot; occurrences in each row</span>
<span class="hl com">#sig_pathway_AsymAD_female$UP_Count &lt;- rowSums(sig_pathway_AsymAD_female == &quot;UP&quot;)</span>

<span class="hl com"># Add a column to count the number of &quot;DOWN&quot; occurrences in each row</span>
<span class="hl com">#sig_pathway_AsymAD_female$DOWN_Count &lt;- rowSums(sig_pathway_AsymAD_female == &quot;DOWN&quot;)</span>


<span class="hl com"># Iterate over each column in final_results_AsymAD_female_t (excluding the Pathway_ID)</span>
<span class="hl kwa">for</span> <span class="hl def">(col_name</span> <span class="hl kwa">in</span> <span class="hl kwd">colnames</span><span class="hl def">(final_results_AsymAD_female_t)[</span><span class="hl opt">-</span><span class="hl num">1</span><span class="hl def">]) {</span>

  <span class="hl com"># Apply the conditions and store the actual value or 0 in a new column in sig_pathway_AsymAD_female</span>
  <span class="hl def">sig_pathway_AsymAD_female[[col_name]]</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">ifelse</span><span class="hl def">(</span>
    <span class="hl def">merged_data_female_AsymAD[[col_name]]</span> <span class="hl opt">&lt;</span> <span class="hl def">merged_data_female_AsymAD</span><span class="hl opt">$</span><span class="hl def">lower_1st_percentile,</span>
    <span class="hl def">merged_data_female_AsymAD[[col_name]],</span>  <span class="hl com"># Retrieve the value if it's less than the lower 1st percentile</span>
    <span class="hl kwd">ifelse</span><span class="hl def">(</span>
      <span class="hl def">merged_data_female_AsymAD[[col_name]]</span> <span class="hl opt">&gt;</span> <span class="hl def">merged_data_female_AsymAD</span><span class="hl opt">$</span><span class="hl def">upper_99th_percentile,</span>
      <span class="hl def">merged_data_female_AsymAD[[col_name]],</span>  <span class="hl com"># Retrieve the value if it's greater than the upper 99th percentile</span>
      <span class="hl num">0</span>  <span class="hl com"># Otherwise, set to 0</span>
    <span class="hl def">)</span>
  <span class="hl def">)</span>
<span class="hl def">}</span>

<span class="hl com"># Select only numeric columns (assuming all columns except Pathway_ID are numeric)</span>
<span class="hl def">numeric_columns</span> <span class="hl kwb">&lt;-</span> <span class="hl def">sig_pathway_AsymAD_female[,</span> <span class="hl kwd">sapply</span><span class="hl def">(sig_pathway_AsymAD_female, is.numeric)]</span>

<span class="hl com"># Calculate UP_count (count of positive values) for each row</span>
<span class="hl def">sig_pathway_AsymAD_female</span><span class="hl opt">$</span><span class="hl def">UP_Count</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">rowSums</span><span class="hl def">(numeric_columns</span> <span class="hl opt">&gt;</span> <span class="hl num">0</span><span class="hl def">)</span>

<span class="hl com"># Calculate DOWN_count (count of negative values) for each row</span>
<span class="hl def">sig_pathway_AsymAD_female</span><span class="hl opt">$</span><span class="hl def">DOWN_Count</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">rowSums</span><span class="hl def">(numeric_columns</span> <span class="hl opt">&lt;</span> <span class="hl num">0</span><span class="hl def">)</span>

<span class="hl com"># Write the output to a .txt file</span>
<span class="hl kwd">write.table</span><span class="hl def">(sig_pathway_AsymAD_female,</span> <span class="hl sng">&quot;sig_pathway_AsymAD_female.txt&quot;</span><span class="hl def">,</span> <span class="hl kwc">sep</span> <span class="hl def">=</span> <span class="hl sng">&quot;\t&quot;</span><span class="hl def">,</span> <span class="hl kwc">quote</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">,</span> <span class="hl kwc">row.names</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">)</span>

<span class="hl com"># Print the table using DT</span>
<span class="hl kwd">datatable</span><span class="hl def">(sig_pathway_AsymAD_female,</span> <span class="hl kwc">options</span> <span class="hl def">=</span> <span class="hl kwd">list</span><span class="hl def">(</span><span class="hl kwc">pageLength</span> <span class="hl def">=</span> <span class="hl num">5</span><span class="hl def">,</span> <span class="hl kwc">autoWidth</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">))</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-5-9.png" alt="plot of chunk unnamed-chunk-5" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">###################################################################################################################################################</span>
<span class="hl com"># Plot the data</span>

<span class="hl com"># Remove the UP_Count and DOWN_Count columns</span>
<span class="hl def">heatmap_data_AsymAD_female</span> <span class="hl kwb">&lt;-</span> <span class="hl def">sig_pathway_AsymAD_female[,</span> <span class="hl opt">!</span><span class="hl def">(</span><span class="hl kwd">colnames</span><span class="hl def">(sig_pathway_AsymAD_female)</span> <span class="hl opt">%in%</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;UP_Count&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;DOWN_Count&quot;</span><span class="hl def">))]</span>

<span class="hl com"># Set the row names to Pathway_ID and remove the Pathway_ID column</span>
<span class="hl kwd">row.names</span><span class="hl def">(heatmap_data_AsymAD_female)</span> <span class="hl kwb">&lt;-</span> <span class="hl def">sig_pathway_AsymAD_female</span><span class="hl opt">$</span><span class="hl def">Pathway_ID</span>
<span class="hl def">heatmap_data_AsymAD_female</span><span class="hl opt">$</span><span class="hl def">Pathway_ID</span> <span class="hl kwb">&lt;-</span> <span class="hl kwa">NULL</span>

<span class="hl com"># Check and handle NA/NaN/Inf values</span>
<span class="hl def">heatmap_data_AsymAD_female[</span><span class="hl kwd">is.na</span><span class="hl def">(heatmap_data_AsymAD_female)]</span> <span class="hl kwb">&lt;-</span> <span class="hl num">0</span>  <span class="hl com"># Replace NA values with 0</span>
<span class="hl def">heatmap_data_AsymAD_female[</span><span class="hl kwd">is.infinite</span><span class="hl def">(heatmap_data_AsymAD_female)]</span> <span class="hl kwb">&lt;-</span> <span class="hl num">0</span>  <span class="hl com"># Replace Inf/-Inf values with 0</span>
</pre></div>
<div class="error"><pre class="knitr r">## Error in is.infinite(heatmap_data_AsymAD_female): default method not implemented for type 'list'
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl com"># Convert the data to numeric type (if it's not already)</span>
<span class="hl def">heatmap_data_AsymAD_female</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">as.matrix</span><span class="hl def">(heatmap_data_AsymAD_female)</span>

<span class="hl com"># Define the color palette</span>
<span class="hl def">color_palette</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">colorRampPalette</span><span class="hl def">(</span><span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;red&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;white&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;blue&quot;</span><span class="hl def">))(</span><span class="hl num">100</span><span class="hl def">)</span>

<span class="hl com"># Generate the heatmap</span>
<span class="hl def">out_heatmap_AsymAD_female</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">heatmap.2</span><span class="hl def">(</span>
  <span class="hl def">heatmap_data_AsymAD_female,</span>              <span class="hl com"># Data matrix</span>
  <span class="hl kwc">Rowv</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>                 <span class="hl com"># Row dendrogram</span>
  <span class="hl kwc">Colv</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>                 <span class="hl com"># Column dendrogram</span>
  <span class="hl kwc">col</span> <span class="hl def">= color_palette,</span>         <span class="hl com"># Color palette</span>
  <span class="hl kwc">scale</span> <span class="hl def">=</span> <span class="hl sng">&quot;row&quot;</span><span class="hl def">,</span>              <span class="hl com"># No scaling</span>
  <span class="hl kwc">trace</span> <span class="hl def">=</span> <span class="hl sng">&quot;none&quot;</span><span class="hl def">,</span>              <span class="hl com"># Turn off trace lines</span>
  <span class="hl kwc">margins</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl num">10</span><span class="hl def">,</span> <span class="hl num">10</span><span class="hl def">),</span>         <span class="hl com"># Set margins for labels</span>
  <span class="hl kwc">key</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>                  <span class="hl com"># Include color key</span>
  <span class="hl kwc">key.title</span> <span class="hl def">=</span> <span class="hl sng">&quot;Z-Score&quot;</span><span class="hl def">,</span>       <span class="hl com"># Key title</span>
  <span class="hl kwc">keysize</span> <span class="hl def">=</span> <span class="hl num">1.5</span><span class="hl def">,</span>               <span class="hl com"># Size of the key</span>
  <span class="hl kwc">cexRow</span> <span class="hl def">=</span> <span class="hl num">0.8</span><span class="hl def">,</span>                <span class="hl com"># Size of row labels</span>
  <span class="hl kwc">cexCol</span> <span class="hl def">=</span> <span class="hl num">0.8</span><span class="hl def">,</span>                <span class="hl com"># Size of column labels</span>
  <span class="hl kwc">dendrogram</span> <span class="hl def">=</span> <span class="hl sng">&quot;both&quot;</span><span class="hl def">,</span>         <span class="hl com"># Show both row and column dendrograms</span>
  <span class="hl kwc">main</span> <span class="hl def">=</span> <span class="hl sng">&quot;Pathway Specific Z-Scores Distribution (AsymAD female)&quot;</span>  <span class="hl com"># Main title</span>
<span class="hl def">)</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-5-10.png" alt="plot of chunk unnamed-chunk-5" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com"># Plot the dendogram</span>
<span class="hl kwd">ggdendrogram</span><span class="hl def">(out_heatmap_AsymAD_female</span><span class="hl opt">$</span><span class="hl def">colDendrogram,</span> <span class="hl kwc">theme_dendro</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-5-11.png" alt="plot of chunk unnamed-chunk-5" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com"># Convert data from wide to long format</span>
<span class="hl def">long_sig_pathway_AsymAD_female</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">gather</span><span class="hl def">(sig_pathway_AsymAD_female,</span> <span class="hl kwc">key</span> <span class="hl def">=</span> <span class="hl sng">&quot;Direction&quot;</span><span class="hl def">,</span> <span class="hl kwc">value</span> <span class="hl def">=</span> <span class="hl sng">&quot;Count&quot;</span><span class="hl def">, UP_Count, DOWN_Count)</span>

<span class="hl com"># Create the bar plot</span>
<span class="hl kwd">ggplot</span><span class="hl def">(long_sig_pathway_AsymAD_female,</span> <span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">x</span> <span class="hl def">= Pathway_ID,</span> <span class="hl kwc">y</span> <span class="hl def">= Count,</span> <span class="hl kwc">color</span> <span class="hl def">= Direction,</span> <span class="hl kwc">group</span> <span class="hl def">= Direction))</span> <span class="hl opt">+</span>
  <span class="hl kwd">geom_line</span><span class="hl def">(</span><span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">)</span> <span class="hl opt">+</span>    <span class="hl com"># Add lines between points</span>
  <span class="hl kwd">geom_point</span><span class="hl def">(</span><span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">2</span><span class="hl def">)</span> <span class="hl opt">+</span>   <span class="hl com"># Add points at each Pathway_ID</span>
  <span class="hl kwd">theme_minimal</span><span class="hl def">()</span> <span class="hl opt">+</span>
  <span class="hl kwd">labs</span><span class="hl def">(</span><span class="hl kwc">title</span> <span class="hl def">=</span> <span class="hl sng">&quot;UP and DOWN Count for Each Pathway (AsymAD female)&quot;</span><span class="hl def">,</span> <span class="hl kwc">x</span> <span class="hl def">=</span> <span class="hl sng">&quot;Pathway ID&quot;</span><span class="hl def">,</span> <span class="hl kwc">y</span> <span class="hl def">=</span> <span class="hl sng">&quot;Count&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl kwd">theme</span><span class="hl def">(</span><span class="hl kwc">axis.text.x</span> <span class="hl def">=</span> <span class="hl kwd">element_text</span><span class="hl def">(</span><span class="hl kwc">angle</span> <span class="hl def">=</span> <span class="hl num">90</span><span class="hl def">,</span> <span class="hl kwc">hjust</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">,</span> <span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">8</span><span class="hl def">))</span> <span class="hl opt">+</span>  <span class="hl com"># Rotate x-axis labels if needed</span>
  <span class="hl kwd">scale_color_manual</span><span class="hl def">(</span><span class="hl kwc">values</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;UP_Count&quot;</span> <span class="hl def">=</span> <span class="hl sng">&quot;lightblue&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;DOWN_Count&quot;</span> <span class="hl def">=</span> <span class="hl sng">&quot;pink&quot;</span><span class="hl def">))</span>  <span class="hl com"># Customize colors</span>
</pre></div>
<div class="warning"><pre class="knitr r">## Warning: Removed 16 rows containing missing values or values outside the scale range
## (`geom_point()`).
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-5-12.png" alt="plot of chunk unnamed-chunk-5" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">############################################################################################################################################################################</span>
<span class="hl com">############################################################################################################################################################################</span>
<span class="hl com">## AsymAD_male</span>

<span class="hl com"># Merge both the data frame by Pathway_ID</span>
<span class="hl def">merged_data_male_AsymAD</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">merge</span><span class="hl def">(zscored_threshold_male, final_results_AsymAD_male_t,</span> <span class="hl kwc">by</span> <span class="hl def">=</span> <span class="hl sng">&quot;Pathway_ID&quot;</span><span class="hl def">)</span>

<span class="hl com"># Initialize an empty data frame to store results</span>
<span class="hl def">sig_pathway_AsymAD_male</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">data.frame</span><span class="hl def">(</span><span class="hl kwc">Pathway_ID</span> <span class="hl def">= merged_data_male_AsymAD</span><span class="hl opt">$</span><span class="hl def">Pathway_ID)</span>


<span class="hl com"># Iterate over each column in final_results_AsymAD_female_t (excluding the Pathway_ID)</span>
<span class="hl kwa">for</span> <span class="hl def">(col_name</span> <span class="hl kwa">in</span> <span class="hl kwd">colnames</span><span class="hl def">(final_results_AsymAD_male_t)[</span><span class="hl opt">-</span><span class="hl num">1</span><span class="hl def">]) {</span>

  <span class="hl com"># Apply the conditions and store the actual value or 0 in a new column in sig_pathway_AsymAD_male</span>
  <span class="hl def">sig_pathway_AsymAD_male[[col_name]]</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">ifelse</span><span class="hl def">(</span>
    <span class="hl def">merged_data_male_AsymAD[[col_name]]</span> <span class="hl opt">&lt;</span> <span class="hl def">merged_data_male_AsymAD</span><span class="hl opt">$</span><span class="hl def">lower_1st_percentile,</span>
    <span class="hl def">merged_data_male_AsymAD[[col_name]],</span>  <span class="hl com"># Retrieve the value if it's less than the lower 1st percentile</span>
    <span class="hl kwd">ifelse</span><span class="hl def">(</span>
      <span class="hl def">merged_data_male_AsymAD[[col_name]]</span> <span class="hl opt">&gt;</span> <span class="hl def">merged_data_male_AsymAD</span><span class="hl opt">$</span><span class="hl def">upper_99th_percentile,</span>
      <span class="hl def">merged_data_male_AsymAD[[col_name]],</span>  <span class="hl com"># Retrieve the value if it's greater than the upper 99th percentile</span>
      <span class="hl num">0</span>  <span class="hl com"># Otherwise, set to 0</span>
    <span class="hl def">)</span>
  <span class="hl def">)</span>
<span class="hl def">}</span>

<span class="hl com"># Select only numeric columns (assuming all columns except Pathway_ID are numeric)</span>
<span class="hl def">numeric_columns</span> <span class="hl kwb">&lt;-</span> <span class="hl def">sig_pathway_AsymAD_male[,</span> <span class="hl kwd">sapply</span><span class="hl def">(sig_pathway_AsymAD_male, is.numeric)]</span>

<span class="hl com"># Calculate UP_count (count of positive values) for each row</span>
<span class="hl def">sig_pathway_AsymAD_male</span><span class="hl opt">$</span><span class="hl def">UP_Count</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">rowSums</span><span class="hl def">(numeric_columns</span> <span class="hl opt">&gt;</span> <span class="hl num">0</span><span class="hl def">)</span>

<span class="hl com"># Calculate DOWN_count (count of negative values) for each row</span>
<span class="hl def">sig_pathway_AsymAD_male</span><span class="hl opt">$</span><span class="hl def">DOWN_Count</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">rowSums</span><span class="hl def">(numeric_columns</span> <span class="hl opt">&lt;</span> <span class="hl num">0</span><span class="hl def">)</span>

<span class="hl com"># Write the output to a .txt file</span>
<span class="hl kwd">write.table</span><span class="hl def">(sig_pathway_AsymAD_male,</span> <span class="hl sng">&quot;sig_pathway_AsymAD_male.txt&quot;</span><span class="hl def">,</span> <span class="hl kwc">sep</span> <span class="hl def">=</span> <span class="hl sng">&quot;\t&quot;</span><span class="hl def">,</span> <span class="hl kwc">quote</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">,</span> <span class="hl kwc">row.names</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">)</span>

<span class="hl com"># Print the table using DT</span>
<span class="hl kwd">datatable</span><span class="hl def">(sig_pathway_AsymAD_male,</span> <span class="hl kwc">options</span> <span class="hl def">=</span> <span class="hl kwd">list</span><span class="hl def">(</span><span class="hl kwc">pageLength</span> <span class="hl def">=</span> <span class="hl num">5</span><span class="hl def">,</span> <span class="hl kwc">autoWidth</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">))</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-5-13.png" alt="plot of chunk unnamed-chunk-5" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">###################################################################################################################################################</span>
<span class="hl com"># Plot the data</span>

<span class="hl com"># Remove the UP_Count and DOWN_Count columns</span>
<span class="hl def">heatmap_data_AsymAD_male</span> <span class="hl kwb">&lt;-</span> <span class="hl def">sig_pathway_AsymAD_male[,</span> <span class="hl opt">!</span><span class="hl def">(</span><span class="hl kwd">colnames</span><span class="hl def">(sig_pathway_AsymAD_male)</span> <span class="hl opt">%in%</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;UP_Count&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;DOWN_Count&quot;</span><span class="hl def">))]</span>

<span class="hl com"># Set the row names to Pathway_ID and remove the Pathway_ID column</span>
<span class="hl kwd">row.names</span><span class="hl def">(heatmap_data_AsymAD_male)</span> <span class="hl kwb">&lt;-</span> <span class="hl def">sig_pathway_AsymAD_male</span><span class="hl opt">$</span><span class="hl def">Pathway_ID</span>
<span class="hl def">heatmap_data_AsymAD_male</span><span class="hl opt">$</span><span class="hl def">Pathway_ID</span> <span class="hl kwb">&lt;-</span> <span class="hl kwa">NULL</span>

<span class="hl com"># Check and handle NA/NaN/Inf values</span>
<span class="hl def">heatmap_data_AsymAD_male[</span><span class="hl kwd">is.na</span><span class="hl def">(heatmap_data_AsymAD_male)]</span> <span class="hl kwb">&lt;-</span> <span class="hl num">0</span>  <span class="hl com"># Replace NA values with 0</span>
<span class="hl def">heatmap_data_AsymAD_male[</span><span class="hl kwd">is.infinite</span><span class="hl def">(heatmap_data_AsymAD_male)]</span> <span class="hl kwb">&lt;-</span> <span class="hl num">0</span>  <span class="hl com"># Replace Inf/-Inf values with 0</span>
</pre></div>
<div class="error"><pre class="knitr r">## Error in is.infinite(heatmap_data_AsymAD_male): default method not implemented for type 'list'
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl com"># Convert the data to numeric type (if it's not already)</span>
<span class="hl def">heatmap_data_AsymAD_male</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">as.matrix</span><span class="hl def">(heatmap_data_AsymAD_male)</span>

<span class="hl com"># Define the color palette</span>
<span class="hl def">color_palette</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">colorRampPalette</span><span class="hl def">(</span><span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;red&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;white&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;blue&quot;</span><span class="hl def">))(</span><span class="hl num">100</span><span class="hl def">)</span>

<span class="hl com"># Generate the heatmap</span>
<span class="hl def">out_heatmap_AsymAD_male</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">heatmap.2</span><span class="hl def">(</span>
  <span class="hl def">heatmap_data_AsymAD_male,</span>              <span class="hl com"># Data matrix</span>
  <span class="hl kwc">Rowv</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>                 <span class="hl com"># Row dendrogram</span>
  <span class="hl kwc">Colv</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>                 <span class="hl com"># Column dendrogram</span>
  <span class="hl kwc">col</span> <span class="hl def">= color_palette,</span>         <span class="hl com"># Color palette</span>
  <span class="hl kwc">scale</span> <span class="hl def">=</span> <span class="hl sng">&quot;row&quot;</span><span class="hl def">,</span>              <span class="hl com"># No scaling</span>
  <span class="hl kwc">trace</span> <span class="hl def">=</span> <span class="hl sng">&quot;none&quot;</span><span class="hl def">,</span>              <span class="hl com"># Turn off trace lines</span>
  <span class="hl kwc">margins</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl num">10</span><span class="hl def">,</span> <span class="hl num">10</span><span class="hl def">),</span>         <span class="hl com"># Set margins for labels</span>
  <span class="hl kwc">key</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>                  <span class="hl com"># Include color key</span>
  <span class="hl kwc">key.title</span> <span class="hl def">=</span> <span class="hl sng">&quot;Z-Score&quot;</span><span class="hl def">,</span>       <span class="hl com"># Key title</span>
  <span class="hl kwc">keysize</span> <span class="hl def">=</span> <span class="hl num">1.5</span><span class="hl def">,</span>               <span class="hl com"># Size of the key</span>
  <span class="hl kwc">cexRow</span> <span class="hl def">=</span> <span class="hl num">0.8</span><span class="hl def">,</span>                <span class="hl com"># Size of row labels</span>
  <span class="hl kwc">cexCol</span> <span class="hl def">=</span> <span class="hl num">0.8</span><span class="hl def">,</span>                <span class="hl com"># Size of column labels</span>
  <span class="hl kwc">dendrogram</span> <span class="hl def">=</span> <span class="hl sng">&quot;both&quot;</span><span class="hl def">,</span>         <span class="hl com"># Show both row and column dendrograms</span>
  <span class="hl kwc">main</span> <span class="hl def">=</span> <span class="hl sng">&quot;Pathway Specific Z-Scores Distribution (AsymAD male)&quot;</span>  <span class="hl com"># Main title</span>
<span class="hl def">)</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-5-14.png" alt="plot of chunk unnamed-chunk-5" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com"># Plot the dendogram</span>
<span class="hl kwd">ggdendrogram</span><span class="hl def">(out_heatmap_AsymAD_male</span><span class="hl opt">$</span><span class="hl def">colDendrogram,</span> <span class="hl kwc">theme_dendro</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">)</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-5-15.png" alt="plot of chunk unnamed-chunk-5" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com"># Convert data from wide to long format</span>
<span class="hl def">long_sig_pathway_AsymAD_male</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">gather</span><span class="hl def">(sig_pathway_AsymAD_male,</span> <span class="hl kwc">key</span> <span class="hl def">=</span> <span class="hl sng">&quot;Direction&quot;</span><span class="hl def">,</span> <span class="hl kwc">value</span> <span class="hl def">=</span> <span class="hl sng">&quot;Count&quot;</span><span class="hl def">, UP_Count, DOWN_Count)</span>

<span class="hl com"># Create the line plot</span>
<span class="hl kwd">ggplot</span><span class="hl def">(long_sig_pathway_AsymAD_male,</span> <span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">x</span> <span class="hl def">= Pathway_ID,</span> <span class="hl kwc">y</span> <span class="hl def">= Count,</span> <span class="hl kwc">color</span> <span class="hl def">= Direction,</span> <span class="hl kwc">group</span> <span class="hl def">= Direction))</span> <span class="hl opt">+</span>
  <span class="hl kwd">geom_line</span><span class="hl def">(</span><span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">)</span> <span class="hl opt">+</span>    <span class="hl com"># AsymADd lines between points</span>
  <span class="hl kwd">geom_point</span><span class="hl def">(</span><span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">2</span><span class="hl def">)</span> <span class="hl opt">+</span>   <span class="hl com"># Add points at each Pathway_ID</span>
  <span class="hl kwd">theme_minimal</span><span class="hl def">()</span> <span class="hl opt">+</span>
  <span class="hl kwd">labs</span><span class="hl def">(</span><span class="hl kwc">title</span> <span class="hl def">=</span> <span class="hl sng">&quot;UP and DOWN Count for Each Pathway (AsymAD male)&quot;</span><span class="hl def">,</span> <span class="hl kwc">x</span> <span class="hl def">=</span> <span class="hl sng">&quot;Pathway ID&quot;</span><span class="hl def">,</span> <span class="hl kwc">y</span> <span class="hl def">=</span> <span class="hl sng">&quot;Count&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl kwd">theme</span><span class="hl def">(</span><span class="hl kwc">axis.text.x</span> <span class="hl def">=</span> <span class="hl kwd">element_text</span><span class="hl def">(</span><span class="hl kwc">angle</span> <span class="hl def">=</span> <span class="hl num">90</span><span class="hl def">,</span> <span class="hl kwc">hjust</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">,</span> <span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">8</span><span class="hl def">))</span> <span class="hl opt">+</span>  <span class="hl com"># Rotate x-axis labels if needed</span>
  <span class="hl kwd">scale_color_manual</span><span class="hl def">(</span><span class="hl kwc">values</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;UP_Count&quot;</span> <span class="hl def">=</span> <span class="hl sng">&quot;lightblue&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;DOWN_Count&quot;</span> <span class="hl def">=</span> <span class="hl sng">&quot;pink&quot;</span><span class="hl def">))</span>  <span class="hl com"># Customize colors</span>
</pre></div>
<div class="warning"><pre class="knitr r">## Warning: Removed 14 rows containing missing values or values outside the scale range
## (`geom_point()`).
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-5-16.png" alt="plot of chunk unnamed-chunk-5" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">############################################################################################################################################################################</span>
<span class="hl com">############################################################################################################################################################################</span>

<span class="hl com">## CT_female</span>

<span class="hl com"># Merge both data frames by Pathway_ID</span>
<span class="hl def">merged_data_female_CT</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">merge</span><span class="hl def">(zscored_threshold_female, final_results_CT_female_t,</span> <span class="hl kwc">by</span> <span class="hl def">=</span> <span class="hl sng">&quot;Pathway_ID&quot;</span><span class="hl def">,</span> <span class="hl kwc">all.x</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>

<span class="hl com"># Initialize an empty data frame to store results</span>
<span class="hl def">sig_pathway_CT_female</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">data.frame</span><span class="hl def">(</span><span class="hl kwc">Pathway_ID</span> <span class="hl def">= merged_data_female_CT</span><span class="hl opt">$</span><span class="hl def">Pathway_ID)</span>

<span class="hl com"># Iterate over each column in final_results_CT_female_t (excluding the Pathway_ID)</span>
<span class="hl kwa">for</span> <span class="hl def">(col_name</span> <span class="hl kwa">in</span> <span class="hl kwd">colnames</span><span class="hl def">(final_results_CT_female_t)[</span><span class="hl opt">-</span><span class="hl num">1</span><span class="hl def">]) {</span>

  <span class="hl com"># Apply the conditions and store the actual value or 0 in a new column in sig_pathway_CT_female</span>
  <span class="hl def">sig_pathway_CT_female[[col_name]]</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">ifelse</span><span class="hl def">(</span>
    <span class="hl def">merged_data_female_CT[[col_name]]</span> <span class="hl opt">&lt;</span> <span class="hl def">merged_data_female_CT</span><span class="hl opt">$</span><span class="hl def">lower_1st_percentile,</span>
    <span class="hl def">merged_data_female_CT[[col_name]],</span>  <span class="hl com"># Retrieve the value if it's less than the lower 1st percentile</span>
    <span class="hl kwd">ifelse</span><span class="hl def">(</span>
      <span class="hl def">merged_data_female_CT[[col_name]]</span> <span class="hl opt">&gt;</span> <span class="hl def">merged_data_female_CT</span><span class="hl opt">$</span><span class="hl def">upper_99th_percentile,</span>
      <span class="hl def">merged_data_female_CT[[col_name]],</span>  <span class="hl com"># Retrieve the value if it's greater than the upper 99th percentile</span>
      <span class="hl num">0</span>  <span class="hl com"># Otherwise, set to 0</span>
    <span class="hl def">)</span>
  <span class="hl def">)</span>
<span class="hl def">}</span>

<span class="hl com"># Select only numeric columns (assuming all columns except Pathway_ID are numeric)</span>
<span class="hl def">numeric_columns</span> <span class="hl kwb">&lt;-</span> <span class="hl def">sig_pathway_CT_female[,</span> <span class="hl kwd">sapply</span><span class="hl def">(sig_pathway_CT_female, is.numeric)]</span>

<span class="hl com"># Calculate UP_count (count of positive values) for each row</span>
<span class="hl def">sig_pathway_CT_female</span><span class="hl opt">$</span><span class="hl def">UP_Count</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">rowSums</span><span class="hl def">(numeric_columns</span> <span class="hl opt">&gt;</span> <span class="hl num">0</span><span class="hl def">)</span>

<span class="hl com"># Calculate DOWN_count (count of negative values) for each row</span>
<span class="hl def">sig_pathway_CT_female</span><span class="hl opt">$</span><span class="hl def">DOWN_Count</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">rowSums</span><span class="hl def">(numeric_columns</span> <span class="hl opt">&lt;</span> <span class="hl num">0</span><span class="hl def">)</span>

<span class="hl com"># Write the output to a .txt file</span>
<span class="hl kwd">write.table</span><span class="hl def">(sig_pathway_CT_female,</span> <span class="hl sng">&quot;sig_pathway_CT_female.txt&quot;</span><span class="hl def">,</span> <span class="hl kwc">sep</span> <span class="hl def">=</span> <span class="hl sng">&quot;\t&quot;</span><span class="hl def">,</span> <span class="hl kwc">quote</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">,</span> <span class="hl kwc">row.names</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">)</span>

<span class="hl com"># Print the table using DT</span>
<span class="hl kwd">datatable</span><span class="hl def">(sig_pathway_CT_female,</span> <span class="hl kwc">options</span> <span class="hl def">=</span> <span class="hl kwd">list</span><span class="hl def">(</span><span class="hl kwc">pageLength</span> <span class="hl def">=</span> <span class="hl num">5</span><span class="hl def">,</span> <span class="hl kwc">autoWidth</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">))</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-5-17.png" alt="plot of chunk unnamed-chunk-5" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">###################################################################################################################################################</span>
<span class="hl com"># Plot the data</span>

<span class="hl com"># Remove the UP_Count and DOWN_Count columns</span>
<span class="hl def">heatmap_data_CT_female</span> <span class="hl kwb">&lt;-</span> <span class="hl def">sig_pathway_CT_female[,</span> <span class="hl opt">!</span><span class="hl def">(</span><span class="hl kwd">colnames</span><span class="hl def">(sig_pathway_CT_female)</span> <span class="hl opt">%in%</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;UP_Count&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;DOWN_Count&quot;</span><span class="hl def">))]</span>

<span class="hl com"># Set the row names to Pathway_ID and remove the Pathway_ID column</span>
<span class="hl kwd">row.names</span><span class="hl def">(heatmap_data_CT_female)</span> <span class="hl kwb">&lt;-</span> <span class="hl def">sig_pathway_CT_female</span><span class="hl opt">$</span><span class="hl def">Pathway_ID</span>
<span class="hl def">heatmap_data_CT_female</span><span class="hl opt">$</span><span class="hl def">Pathway_ID</span> <span class="hl kwb">&lt;-</span> <span class="hl kwa">NULL</span>

<span class="hl com"># Check and handle NA/NaN/Inf values</span>
<span class="hl def">heatmap_data_CT_female[</span><span class="hl kwd">is.na</span><span class="hl def">(heatmap_data_CT_female)]</span> <span class="hl kwb">&lt;-</span> <span class="hl num">0</span>  <span class="hl com"># Replace NA values with 0</span>
<span class="hl def">heatmap_data_CT_female[</span><span class="hl kwd">is.infinite</span><span class="hl def">(heatmap_data_CT_female)]</span> <span class="hl kwb">&lt;-</span> <span class="hl num">0</span>  <span class="hl com"># Replace Inf/-Inf values with 0</span>
</pre></div>
<div class="error"><pre class="knitr r">## Error in is.infinite(heatmap_data_CT_female): default method not implemented for type 'list'
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl com"># Convert the data to numeric type (if it's not already)</span>
<span class="hl def">heatmap_data_CT_female</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">as.matrix</span><span class="hl def">(heatmap_data_CT_female)</span>

<span class="hl com"># Define the color palette</span>
<span class="hl def">color_palette</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">colorRampPalette</span><span class="hl def">(</span><span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;red&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;white&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;blue&quot;</span><span class="hl def">))(</span><span class="hl num">100</span><span class="hl def">)</span>

<span class="hl com"># Generate the heatmap</span>
<span class="hl kwd">heatmap.2</span><span class="hl def">(</span>
  <span class="hl def">heatmap_data_CT_female,</span>              <span class="hl com"># Data matrix</span>
  <span class="hl kwc">Rowv</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>                 <span class="hl com"># Row dendrogram</span>
  <span class="hl kwc">Colv</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>                 <span class="hl com"># Column dendrogram</span>
  <span class="hl kwc">col</span> <span class="hl def">= color_palette,</span>         <span class="hl com"># Color palette</span>
  <span class="hl kwc">scale</span> <span class="hl def">=</span> <span class="hl sng">&quot;row&quot;</span><span class="hl def">,</span>              <span class="hl com"># No scaling</span>
  <span class="hl kwc">trace</span> <span class="hl def">=</span> <span class="hl sng">&quot;none&quot;</span><span class="hl def">,</span>              <span class="hl com"># Turn off trace lines</span>
  <span class="hl kwc">margins</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl num">10</span><span class="hl def">,</span> <span class="hl num">10</span><span class="hl def">),</span>         <span class="hl com"># Set margins for labels</span>
  <span class="hl kwc">key</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>                  <span class="hl com"># Include color key</span>
  <span class="hl kwc">key.title</span> <span class="hl def">=</span> <span class="hl sng">&quot;Z-Score&quot;</span><span class="hl def">,</span>       <span class="hl com"># Key title</span>
  <span class="hl kwc">keysize</span> <span class="hl def">=</span> <span class="hl num">1.5</span><span class="hl def">,</span>               <span class="hl com"># Size of the key</span>
  <span class="hl kwc">cexRow</span> <span class="hl def">=</span> <span class="hl num">0.8</span><span class="hl def">,</span>                <span class="hl com"># Size of row labels</span>
  <span class="hl kwc">cexCol</span> <span class="hl def">=</span> <span class="hl num">0.8</span><span class="hl def">,</span>                <span class="hl com"># Size of column labels</span>
  <span class="hl kwc">dendrogram</span> <span class="hl def">=</span> <span class="hl sng">&quot;both&quot;</span><span class="hl def">,</span>         <span class="hl com"># Show both row and column dendrograms</span>
  <span class="hl kwc">main</span> <span class="hl def">=</span> <span class="hl sng">&quot;Pathway Specific Z-Scores Distribution (CT female)&quot;</span>  <span class="hl com"># Main title</span>
<span class="hl def">)</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-5-18.png" alt="plot of chunk unnamed-chunk-5" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com"># Convert data from wide to long format</span>
<span class="hl def">long_sig_pathway_CT_female</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">gather</span><span class="hl def">(sig_pathway_CT_female,</span> <span class="hl kwc">key</span> <span class="hl def">=</span> <span class="hl sng">&quot;Direction&quot;</span><span class="hl def">,</span> <span class="hl kwc">value</span> <span class="hl def">=</span> <span class="hl sng">&quot;Count&quot;</span><span class="hl def">, UP_Count, DOWN_Count)</span>

<span class="hl com"># Create the bar plot</span>
<span class="hl kwd">ggplot</span><span class="hl def">(long_sig_pathway_CT_female,</span> <span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">x</span> <span class="hl def">= Pathway_ID,</span> <span class="hl kwc">y</span> <span class="hl def">= Count,</span> <span class="hl kwc">fill</span> <span class="hl def">= Direction))</span> <span class="hl opt">+</span>
  <span class="hl kwd">geom_bar</span><span class="hl def">(</span><span class="hl kwc">stat</span> <span class="hl def">=</span> <span class="hl sng">&quot;identity&quot;</span><span class="hl def">,</span> <span class="hl kwc">position</span> <span class="hl def">=</span> <span class="hl sng">&quot;dodge&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl kwd">theme_minimal</span><span class="hl def">()</span> <span class="hl opt">+</span>
  <span class="hl kwd">labs</span><span class="hl def">(</span><span class="hl kwc">title</span> <span class="hl def">=</span> <span class="hl sng">&quot;UP and DOWN Count for Each Pathway (CT female)&quot;</span><span class="hl def">,</span> <span class="hl kwc">x</span> <span class="hl def">=</span> <span class="hl sng">&quot;Pathway ID&quot;</span><span class="hl def">,</span> <span class="hl kwc">y</span> <span class="hl def">=</span> <span class="hl sng">&quot;Count&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl kwd">theme</span><span class="hl def">(</span><span class="hl kwc">axis.text.x</span> <span class="hl def">=</span> <span class="hl kwd">element_text</span><span class="hl def">(</span><span class="hl kwc">angle</span> <span class="hl def">=</span> <span class="hl num">90</span><span class="hl def">,</span> <span class="hl kwc">hjust</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">,</span> <span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">0.5</span><span class="hl def">))</span> <span class="hl opt">+</span>  <span class="hl com"># Rotate x-axis labels if needed</span>
  <span class="hl kwd">scale_fill_manual</span><span class="hl def">(</span><span class="hl kwc">values</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;UP_Count&quot;</span> <span class="hl def">=</span> <span class="hl sng">&quot;lightblue&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;DOWN_Count&quot;</span> <span class="hl def">=</span> <span class="hl sng">&quot;pink&quot;</span><span class="hl def">))</span>  <span class="hl com"># Customize colors</span>
</pre></div>
<div class="warning"><pre class="knitr r">## Warning: Removed 16 rows containing missing values or values outside the scale range
## (`geom_bar()`).
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-5-19.png" alt="plot of chunk unnamed-chunk-5" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">############################################################################################################################################################################</span>
<span class="hl com">############################################################################################################################################################################</span>

<span class="hl com">## CT_male</span>

<span class="hl com"># Merge both data frames by Pathway_ID</span>
<span class="hl def">merged_data_male_CT</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">merge</span><span class="hl def">(zscored_threshold_male, final_results_CT_male_t,</span> <span class="hl kwc">by</span> <span class="hl def">=</span> <span class="hl sng">&quot;Pathway_ID&quot;</span><span class="hl def">,</span> <span class="hl kwc">all.x</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>

<span class="hl com"># Initialize an empty data frame to store results</span>
<span class="hl def">sig_pathway_CT_male</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">data.frame</span><span class="hl def">(</span><span class="hl kwc">Pathway_ID</span> <span class="hl def">= merged_data_male_CT</span><span class="hl opt">$</span><span class="hl def">Pathway_ID)</span>

<span class="hl com"># Iterate over each column in final_results_CT_female_t (excluding the Pathway_ID)</span>
<span class="hl kwa">for</span> <span class="hl def">(col_name</span> <span class="hl kwa">in</span> <span class="hl kwd">colnames</span><span class="hl def">(final_results_CT_male_t)[</span><span class="hl opt">-</span><span class="hl num">1</span><span class="hl def">]) {</span>

  <span class="hl com"># Apply the conditions and store the actual value or 0 in a new column in sig_pathway_CT_male</span>
  <span class="hl def">sig_pathway_CT_male[[col_name]]</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">ifelse</span><span class="hl def">(</span>
    <span class="hl def">merged_data_male_CT[[col_name]]</span> <span class="hl opt">&lt;</span> <span class="hl def">merged_data_male_CT</span><span class="hl opt">$</span><span class="hl def">lower_1st_percentile,</span>
    <span class="hl def">merged_data_male_CT[[col_name]],</span>  <span class="hl com"># Retrieve the value if it's less than the lower 1st percentile</span>
    <span class="hl kwd">ifelse</span><span class="hl def">(</span>
      <span class="hl def">merged_data_male_CT[[col_name]]</span> <span class="hl opt">&gt;</span> <span class="hl def">merged_data_male_CT</span><span class="hl opt">$</span><span class="hl def">upper_99th_percentile,</span>
      <span class="hl def">merged_data_male_CT[[col_name]],</span>  <span class="hl com"># Retrieve the value if it's greater than the upper 99th percentile</span>
      <span class="hl num">0</span>  <span class="hl com"># Otherwise, set to 0</span>
    <span class="hl def">)</span>
  <span class="hl def">)</span>
<span class="hl def">}</span>

<span class="hl com"># Select only numeric columns (assuming all columns except Pathway_ID are numeric)</span>
<span class="hl def">numeric_columns</span> <span class="hl kwb">&lt;-</span> <span class="hl def">sig_pathway_CT_male[,</span> <span class="hl kwd">sapply</span><span class="hl def">(sig_pathway_CT_male, is.numeric)]</span>

<span class="hl com"># Calculate UP_count (count of positive values) for each row</span>
<span class="hl def">sig_pathway_CT_male</span><span class="hl opt">$</span><span class="hl def">UP_Count</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">rowSums</span><span class="hl def">(numeric_columns</span> <span class="hl opt">&gt;</span> <span class="hl num">0</span><span class="hl def">)</span>

<span class="hl com"># Calculate DOWN_count (count of negative values) for each row</span>
<span class="hl def">sig_pathway_CT_male</span><span class="hl opt">$</span><span class="hl def">DOWN_Count</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">rowSums</span><span class="hl def">(numeric_columns</span> <span class="hl opt">&lt;</span> <span class="hl num">0</span><span class="hl def">)</span>

<span class="hl com"># Write the output to a .txt file</span>
<span class="hl kwd">write.table</span><span class="hl def">(sig_pathway_CT_male,</span> <span class="hl sng">&quot;sig_pathway_CT_male.txt&quot;</span><span class="hl def">,</span> <span class="hl kwc">sep</span> <span class="hl def">=</span> <span class="hl sng">&quot;\t&quot;</span><span class="hl def">,</span> <span class="hl kwc">quote</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">,</span> <span class="hl kwc">row.names</span> <span class="hl def">=</span> <span class="hl num">FALSE</span><span class="hl def">)</span>

<span class="hl com"># Print the table using DT</span>
<span class="hl kwd">datatable</span><span class="hl def">(sig_pathway_CT_male,</span> <span class="hl kwc">options</span> <span class="hl def">=</span> <span class="hl kwd">list</span><span class="hl def">(</span><span class="hl kwc">pageLength</span> <span class="hl def">=</span> <span class="hl num">5</span><span class="hl def">,</span> <span class="hl kwc">autoWidth</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">))</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-5-20.png" alt="plot of chunk unnamed-chunk-5" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">###################################################################################################################################################</span>
<span class="hl com"># Plot the data</span>

<span class="hl com"># Remove the UP_Count and DOWN_Count columns</span>
<span class="hl def">heatmap_data_CT_male</span> <span class="hl kwb">&lt;-</span> <span class="hl def">sig_pathway_CT_male[,</span> <span class="hl opt">!</span><span class="hl def">(</span><span class="hl kwd">colnames</span><span class="hl def">(sig_pathway_CT_male)</span> <span class="hl opt">%in%</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;UP_Count&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;DOWN_Count&quot;</span><span class="hl def">))]</span>

<span class="hl com"># Set the row names to Pathway_ID and remove the Pathway_ID column</span>
<span class="hl kwd">row.names</span><span class="hl def">(heatmap_data_CT_male)</span> <span class="hl kwb">&lt;-</span> <span class="hl def">sig_pathway_CT_male</span><span class="hl opt">$</span><span class="hl def">Pathway_ID</span>
<span class="hl def">heatmap_data_CT_male</span><span class="hl opt">$</span><span class="hl def">Pathway_ID</span> <span class="hl kwb">&lt;-</span> <span class="hl kwa">NULL</span>

<span class="hl com"># Check and handle NA/NaN/Inf values</span>
<span class="hl def">heatmap_data_CT_male[</span><span class="hl kwd">is.na</span><span class="hl def">(heatmap_data_CT_male)]</span> <span class="hl kwb">&lt;-</span> <span class="hl num">0</span>  <span class="hl com"># Replace NA values with 0</span>
<span class="hl def">heatmap_data_CT_male[</span><span class="hl kwd">is.infinite</span><span class="hl def">(heatmap_data_CT_male)]</span> <span class="hl kwb">&lt;-</span> <span class="hl num">0</span>  <span class="hl com"># Replace Inf/-Inf values with 0</span>
</pre></div>
<div class="error"><pre class="knitr r">## Error in is.infinite(heatmap_data_CT_male): default method not implemented for type 'list'
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl com"># Convert the data to numeric type (if it's not already)</span>
<span class="hl def">heatmap_data_CT_male</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">as.matrix</span><span class="hl def">(heatmap_data_CT_male)</span>

<span class="hl com"># Define the color palette</span>
<span class="hl def">color_palette</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">colorRampPalette</span><span class="hl def">(</span><span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;red&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;white&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;blue&quot;</span><span class="hl def">))(</span><span class="hl num">100</span><span class="hl def">)</span>

<span class="hl com"># Generate the heatmap</span>
<span class="hl kwd">heatmap.2</span><span class="hl def">(</span>
  <span class="hl def">heatmap_data_CT_male,</span>              <span class="hl com"># Data matrix</span>
  <span class="hl kwc">Rowv</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>                 <span class="hl com"># Row dendrogram</span>
  <span class="hl kwc">Colv</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>                 <span class="hl com"># Column dendrogram</span>
  <span class="hl kwc">col</span> <span class="hl def">= color_palette,</span>         <span class="hl com"># Color palette</span>
  <span class="hl kwc">scale</span> <span class="hl def">=</span> <span class="hl sng">&quot;row&quot;</span><span class="hl def">,</span>              <span class="hl com"># No scaling</span>
  <span class="hl kwc">trace</span> <span class="hl def">=</span> <span class="hl sng">&quot;none&quot;</span><span class="hl def">,</span>              <span class="hl com"># Turn off trace lines</span>
  <span class="hl kwc">margins</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl num">10</span><span class="hl def">,</span> <span class="hl num">10</span><span class="hl def">),</span>         <span class="hl com"># Set margins for labels</span>
  <span class="hl kwc">key</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span>                  <span class="hl com"># Include color key</span>
  <span class="hl kwc">key.title</span> <span class="hl def">=</span> <span class="hl sng">&quot;Z-Score&quot;</span><span class="hl def">,</span>       <span class="hl com"># Key title</span>
  <span class="hl kwc">keysize</span> <span class="hl def">=</span> <span class="hl num">1.5</span><span class="hl def">,</span>               <span class="hl com"># Size of the key</span>
  <span class="hl kwc">cexRow</span> <span class="hl def">=</span> <span class="hl num">0.8</span><span class="hl def">,</span>                <span class="hl com"># Size of row labels</span>
  <span class="hl kwc">cexCol</span> <span class="hl def">=</span> <span class="hl num">0.8</span><span class="hl def">,</span>                <span class="hl com"># Size of column labels</span>
  <span class="hl kwc">dendrogram</span> <span class="hl def">=</span> <span class="hl sng">&quot;both&quot;</span><span class="hl def">,</span>         <span class="hl com"># Show both row and column dendrograms</span>
  <span class="hl kwc">main</span> <span class="hl def">=</span> <span class="hl sng">&quot;Pathway Specific Z-Scores Distribution (CT male)&quot;</span>  <span class="hl com"># Main title</span>
<span class="hl def">)</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-5-21.png" alt="plot of chunk unnamed-chunk-5" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com"># Convert data from wide to long format</span>
<span class="hl def">long_sig_pathway_CT_male</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">gather</span><span class="hl def">(sig_pathway_CT_male,</span> <span class="hl kwc">key</span> <span class="hl def">=</span> <span class="hl sng">&quot;Direction&quot;</span><span class="hl def">,</span> <span class="hl kwc">value</span> <span class="hl def">=</span> <span class="hl sng">&quot;Count&quot;</span><span class="hl def">, UP_Count, DOWN_Count)</span>

<span class="hl com"># Create the bar plot</span>
<span class="hl kwd">ggplot</span><span class="hl def">(long_sig_pathway_CT_male,</span> <span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">x</span> <span class="hl def">= Pathway_ID,</span> <span class="hl kwc">y</span> <span class="hl def">= Count,</span> <span class="hl kwc">fill</span> <span class="hl def">= Direction))</span> <span class="hl opt">+</span>
  <span class="hl kwd">geom_bar</span><span class="hl def">(</span><span class="hl kwc">stat</span> <span class="hl def">=</span> <span class="hl sng">&quot;identity&quot;</span><span class="hl def">,</span> <span class="hl kwc">position</span> <span class="hl def">=</span> <span class="hl sng">&quot;dodge&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl kwd">theme_minimal</span><span class="hl def">()</span> <span class="hl opt">+</span>
  <span class="hl kwd">labs</span><span class="hl def">(</span><span class="hl kwc">title</span> <span class="hl def">=</span> <span class="hl sng">&quot;UP and DOWN Count for Each Pathway (CT male)&quot;</span><span class="hl def">,</span> <span class="hl kwc">x</span> <span class="hl def">=</span> <span class="hl sng">&quot;Pathway ID&quot;</span><span class="hl def">,</span> <span class="hl kwc">y</span> <span class="hl def">=</span> <span class="hl sng">&quot;Count&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl kwd">theme</span><span class="hl def">(</span><span class="hl kwc">axis.text.x</span> <span class="hl def">=</span> <span class="hl kwd">element_text</span><span class="hl def">(</span><span class="hl kwc">angle</span> <span class="hl def">=</span> <span class="hl num">90</span><span class="hl def">,</span> <span class="hl kwc">hjust</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">,</span> <span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">0.5</span><span class="hl def">))</span> <span class="hl opt">+</span>  <span class="hl com"># Rotate x-axis labels if needed</span>
  <span class="hl kwd">scale_fill_manual</span><span class="hl def">(</span><span class="hl kwc">values</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;UP_Count&quot;</span> <span class="hl def">=</span> <span class="hl sng">&quot;lightblue&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;DOWN_Count&quot;</span> <span class="hl def">=</span> <span class="hl sng">&quot;pink&quot;</span><span class="hl def">))</span>  <span class="hl com"># Customize colors</span>
</pre></div>
<div class="warning"><pre class="knitr r">## Warning: Removed 14 rows containing missing values or values outside the scale range
## (`geom_bar()`).
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-5-22.png" alt="plot of chunk unnamed-chunk-5" class="plot" /></div></div>

<p>Compare Pathway-specific Zscore Distributions for CT and AD Individuals across Glycerophospholipid metabolism :</p>

<div class="chunk" id="unnamed-chunk-6"><div class="rcode"><div class="source"><pre class="knitr r"><span class="hl kwd">library</span><span class="hl def">(ggplot2)</span>
<span class="hl kwd">library</span><span class="hl def">(ggridges)</span>
<span class="hl kwd">library</span><span class="hl def">(reshape2)</span>
</pre></div>
<div class="message"><pre class="knitr r">## 
## Attaching package: 'reshape2'
</pre></div>
<div class="message"><pre class="knitr r">## The following object is masked from 'package:tidyr':
## 
##     smiths
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl kwd">library</span><span class="hl def">(ggbreak)</span>
</pre></div>
<div class="message"><pre class="knitr r">## ggbreak v0.1.6 Learn more at https://yulab-smu.top/
</pre></div>
<div class="message"><pre class="knitr r">## If you use ggbreak in published research, please cite the following
## paper:
## 
## S Xu, M Chen, T Feng, L Zhan, L Zhou, G Yu. Use ggbreak to effectively
## utilize plotting space to deal with large datasets and outliers.
## Frontiers in Genetics. 2021, 12:774846. doi: 10.3389/fgene.2021.774846
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl com"># Select pathway-specific (eg: hsa00564) Zscore from all four groups</span>

<span class="hl com"># Find the maximum length of the vectors</span>
<span class="hl def">max_length</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">max</span><span class="hl def">(</span>
  <span class="hl kwd">length</span><span class="hl def">(final_results_AD_female</span><span class="hl opt">$</span><span class="hl def">hsa00564_Zscore),</span>
  <span class="hl kwd">length</span><span class="hl def">(final_results_AD_male</span><span class="hl opt">$</span><span class="hl def">hsa00564_Zscore),</span>
  <span class="hl kwd">length</span><span class="hl def">(final_results_AsymAD_female</span><span class="hl opt">$</span><span class="hl def">hsa00564_Zscore),</span>
  <span class="hl kwd">length</span><span class="hl def">(final_results_AsymAD_male</span><span class="hl opt">$</span><span class="hl def">hsa00564_Zscore),</span>
  <span class="hl kwd">length</span><span class="hl def">(final_results_CT_female</span><span class="hl opt">$</span><span class="hl def">hsa00564_Zscore),</span>
  <span class="hl kwd">length</span><span class="hl def">(final_results_CT_male</span><span class="hl opt">$</span><span class="hl def">hsa00564_Zscore)</span>
<span class="hl def">)</span>

<span class="hl com"># Function to pad a vector with NA values</span>
<span class="hl def">pad_vector</span> <span class="hl kwb">&lt;-</span> <span class="hl kwa">function</span><span class="hl def">(</span><span class="hl kwc">vec</span><span class="hl def">,</span> <span class="hl kwc">length_out</span><span class="hl def">) {</span>
  <span class="hl kwd">c</span><span class="hl def">(vec,</span> <span class="hl kwd">rep</span><span class="hl def">(</span><span class="hl num">NA</span><span class="hl def">, length_out</span> <span class="hl opt">-</span> <span class="hl kwd">length</span><span class="hl def">(vec)))</span>
<span class="hl def">}</span>

<span class="hl com"># Pad each vector to the maximum length</span>
<span class="hl def">hsa00564_AD_female_padded</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">pad_vector</span><span class="hl def">(final_results_AD_female</span><span class="hl opt">$</span><span class="hl def">hsa00564_Zscore, max_length)</span>
<span class="hl def">hsa00564_AD_male_padded</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">pad_vector</span><span class="hl def">(final_results_AD_male</span><span class="hl opt">$</span><span class="hl def">hsa00564_Zscore, max_length)</span>
<span class="hl def">hsa00564_AsymAD_female_padded</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">pad_vector</span><span class="hl def">(final_results_AsymAD_female</span><span class="hl opt">$</span><span class="hl def">hsa00564_Zscore, max_length)</span>
<span class="hl def">hsa00564_AsymAD_male_padded</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">pad_vector</span><span class="hl def">(final_results_AsymAD_male</span><span class="hl opt">$</span><span class="hl def">hsa00564_Zscore, max_length)</span>
<span class="hl def">hsa00564_CT_female_padded</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">pad_vector</span><span class="hl def">(final_results_CT_female</span><span class="hl opt">$</span><span class="hl def">hsa00564_Zscore, max_length)</span>
<span class="hl def">hsa00564_CT_male_padded</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">pad_vector</span><span class="hl def">(final_results_CT_male</span><span class="hl opt">$</span><span class="hl def">hsa00564_Zscore, max_length)</span>

<span class="hl com"># Combine into a single data frame</span>
<span class="hl def">hsa00564_Zscore_summary</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">data.frame</span><span class="hl def">(</span>
  <span class="hl kwc">hsa00564_AD_female</span> <span class="hl def">= hsa00564_AD_female_padded,</span>
  <span class="hl kwc">hsa00564_AD_male</span> <span class="hl def">= hsa00564_AD_male_padded,</span>
  <span class="hl kwc">hsa00564_AsymAD_female</span> <span class="hl def">= hsa00564_AsymAD_female_padded,</span>
  <span class="hl kwc">hsa00564_AsymAD_male</span> <span class="hl def">= hsa00564_AsymAD_male_padded,</span>
  <span class="hl kwc">hsa00564_CT_female</span> <span class="hl def">= hsa00564_CT_female_padded,</span>
  <span class="hl kwc">hsa00564_CT_male</span> <span class="hl def">= hsa00564_CT_male_padded</span>
<span class="hl def">)</span>

<span class="hl com"># View the combined data frame</span>
<span class="hl com">#head(hsa00564_Zscore_summary)</span>

<span class="hl com"># Print the table using DT</span>
<span class="hl kwd">datatable</span><span class="hl def">(hsa00564_Zscore_summary,</span> <span class="hl kwc">options</span> <span class="hl def">=</span> <span class="hl kwd">list</span><span class="hl def">(</span><span class="hl kwc">pageLength</span> <span class="hl def">=</span> <span class="hl num">5</span><span class="hl def">,</span> <span class="hl kwc">autoWidth</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">))</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-6-1.png" alt="plot of chunk unnamed-chunk-6" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">#write.table(hsa00564_Zscore_summary, &quot;/Users/poddea/Desktop/hsa00564_Zscore_summary.txt&quot;, sep = &quot;\t&quot;, quote = FALSE, row.names = FALSE)</span>

<span class="hl com"># Transform the data to long format</span>
<span class="hl def">hsa00564_Zscore_summary_long</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">melt</span><span class="hl def">(hsa00564_Zscore_summary,</span> <span class="hl kwc">variable.name</span> <span class="hl def">=</span> <span class="hl sng">&quot;Condition&quot;</span><span class="hl def">,</span> <span class="hl kwc">value.name</span> <span class="hl def">=</span> <span class="hl sng">&quot;Value&quot;</span><span class="hl def">)</span>
</pre></div>
<div class="message"><pre class="knitr r">## No id variables; using all as measure variables
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl com"># Remove NA values (if any)</span>
<span class="hl def">hsa00564_Zscore_summary_long</span> <span class="hl kwb">&lt;-</span> <span class="hl def">hsa00564_Zscore_summary_long[</span><span class="hl kwd">complete.cases</span><span class="hl def">(hsa00564_Zscore_summary_long), ]</span>

<span class="hl com"># Create the Ridgeline Plot:</span>
<span class="hl kwd">ggplot</span><span class="hl def">(hsa00564_Zscore_summary_long,</span> <span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">x</span> <span class="hl def">= Value,</span> <span class="hl kwc">y</span> <span class="hl def">= Condition,</span> <span class="hl kwc">fill</span> <span class="hl def">= Condition))</span> <span class="hl opt">+</span>
  <span class="hl kwd">geom_density_ridges</span><span class="hl def">(</span><span class="hl kwc">scale</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl kwd">theme_ridges</span><span class="hl def">()</span> <span class="hl opt">+</span>
  <span class="hl kwd">theme</span><span class="hl def">(</span><span class="hl kwc">legend.position</span> <span class="hl def">=</span> <span class="hl sng">&quot;none&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl kwd">labs</span><span class="hl def">(</span><span class="hl kwc">title</span> <span class="hl def">=</span> <span class="hl sng">&quot;Comparison of Pathway-specific Zscore Distributions for CT, AD and AsymAD Individuals across Glycerophospholipid metabolism &quot;</span><span class="hl def">,</span> <span class="hl kwc">x</span> <span class="hl def">=</span> <span class="hl sng">&quot;Value&quot;</span><span class="hl def">,</span> <span class="hl kwc">y</span> <span class="hl def">=</span> <span class="hl sng">&quot;Condition&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl kwd">scale_fill_manual</span><span class="hl def">(</span><span class="hl kwc">values</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;tomato&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;blue&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;coral&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;cornflowerblue&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;pink&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;lightblue&quot;</span><span class="hl def">))</span>
</pre></div>
<div class="message"><pre class="knitr r">## Picking joint bandwidth of 0.452
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-6-2.png" alt="plot of chunk unnamed-chunk-6" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">##############################################################################################################################################################</span>

<span class="hl com"># Filter the data for Female_CT and Female_AD</span>
<span class="hl def">data_filtered_AD_mf</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">subset</span><span class="hl def">(hsa00564_Zscore_summary_long, Condition</span> <span class="hl opt">%in%</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;hsa00564_AD_female&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;hsa00564_AD_male&quot;</span><span class="hl def">))</span>

<span class="hl def">data_filtered_CTAD_m</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">subset</span><span class="hl def">(hsa00564_Zscore_summary_long, Condition</span> <span class="hl opt">%in%</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;hsa00564_CT_male&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;hsa00564_AD_male&quot;</span><span class="hl def">))</span>

<span class="hl def">data_filtered_CTAD_f</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">subset</span><span class="hl def">(hsa00564_Zscore_summary_long, Condition</span> <span class="hl opt">%in%</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;hsa00564_CT_female&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;hsa00564_AD_female&quot;</span><span class="hl def">))</span>

<span class="hl com"># Create the overlapping density plot</span>
<span class="hl kwd">ggplot</span><span class="hl def">(data_filtered_AD_mf,</span> <span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">x</span> <span class="hl def">= Value,</span> <span class="hl kwc">fill</span> <span class="hl def">= Condition))</span> <span class="hl opt">+</span>
  <span class="hl kwd">geom_density</span><span class="hl def">(</span><span class="hl kwc">alpha</span> <span class="hl def">=</span> <span class="hl num">0.5</span><span class="hl def">)</span> <span class="hl opt">+</span>  <span class="hl com"># Set transparency for overlapping densities</span>
  <span class="hl kwd">theme_minimal</span><span class="hl def">()</span> <span class="hl opt">+</span>
  <span class="hl kwd">labs</span><span class="hl def">(</span><span class="hl kwc">title</span> <span class="hl def">=</span> <span class="hl sng">&quot;Density Plot for AD_male and AD_female across Glycerophospholipid metabolism   pathway&quot;</span><span class="hl def">,</span>
       <span class="hl kwc">x</span> <span class="hl def">=</span> <span class="hl sng">&quot;Z score&quot;</span><span class="hl def">,</span>
       <span class="hl kwc">y</span> <span class="hl def">=</span> <span class="hl sng">&quot;Density&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl kwd">scale_fill_manual</span><span class="hl def">(</span><span class="hl kwc">values</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;hsa00564_AD_female&quot;</span> <span class="hl def">=</span> <span class="hl sng">&quot;red&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;hsa00564_AD_male&quot;</span> <span class="hl def">=</span> <span class="hl sng">&quot;blue&quot;</span><span class="hl def">))</span> <span class="hl opt">+</span>
  <span class="hl kwd">theme_minimal</span><span class="hl def">(</span><span class="hl kwc">base_size</span> <span class="hl def">=</span> <span class="hl num">14</span><span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl kwd">theme</span><span class="hl def">(</span>
    <span class="hl kwc">text</span> <span class="hl def">=</span> <span class="hl kwd">element_text</span><span class="hl def">(</span><span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">20</span><span class="hl def">),</span>
    <span class="hl kwc">legend.position</span> <span class="hl def">=</span> <span class="hl sng">&quot;right&quot;</span>
  <span class="hl def">)</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-6-3.png" alt="plot of chunk unnamed-chunk-6" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl def">result_AD_mf</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">wilcox.test</span><span class="hl def">(hsa00564_Zscore_summary</span><span class="hl opt">$</span><span class="hl def">hsa00564_AD_male, hsa00564_Zscore_summary</span><span class="hl opt">$</span><span class="hl def">hsa00564_AD_female,</span> <span class="hl kwc">paired</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span> <span class="hl kwc">na.rm</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>

<span class="hl kwd">print</span><span class="hl def">(result_AD_mf)</span>
</pre></div>
<div class="output"><pre class="knitr r">## 
## 	Wilcoxon signed rank exact test
## 
## data:  hsa00564_Zscore_summary$hsa00564_AD_male and hsa00564_Zscore_summary$hsa00564_AD_female
## V = 669, p-value = 0.2715
## alternative hypothesis: true location shift is not equal to 0
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl kwd">ggplot</span><span class="hl def">(data_filtered_CTAD_m,</span> <span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">x</span> <span class="hl def">= Value,</span> <span class="hl kwc">fill</span> <span class="hl def">= Condition))</span> <span class="hl opt">+</span>
  <span class="hl kwd">geom_density</span><span class="hl def">(</span><span class="hl kwc">alpha</span> <span class="hl def">=</span> <span class="hl num">0.5</span><span class="hl def">)</span> <span class="hl opt">+</span>  <span class="hl com"># Set transparency for overlapping densities</span>
  <span class="hl kwd">theme_minimal</span><span class="hl def">()</span> <span class="hl opt">+</span>
  <span class="hl kwd">labs</span><span class="hl def">(</span><span class="hl kwc">title</span> <span class="hl def">=</span> <span class="hl sng">&quot;Density Plot for CT_male and AD_male across Glycerophospholipid metabolism   pathway&quot;</span><span class="hl def">,</span>
       <span class="hl kwc">x</span> <span class="hl def">=</span> <span class="hl sng">&quot;Z score&quot;</span><span class="hl def">,</span>
       <span class="hl kwc">y</span> <span class="hl def">=</span> <span class="hl sng">&quot;Density&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl kwd">scale_fill_manual</span><span class="hl def">(</span><span class="hl kwc">values</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;hsa00564_CT_male&quot;</span> <span class="hl def">=</span> <span class="hl sng">&quot;lightblue&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;hsa00564_AD_male&quot;</span> <span class="hl def">=</span> <span class="hl sng">&quot;blue&quot;</span><span class="hl def">))</span> <span class="hl opt">+</span>
  <span class="hl kwd">theme</span><span class="hl def">(</span><span class="hl kwc">legend.title</span> <span class="hl def">=</span> <span class="hl kwd">element_blank</span><span class="hl def">())</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-6-4.png" alt="plot of chunk unnamed-chunk-6" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl def">result_CTAD_m</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">wilcox.test</span><span class="hl def">(hsa00564_Zscore_summary</span><span class="hl opt">$</span><span class="hl def">hsa00564_CT_male, hsa00564_Zscore_summary</span><span class="hl opt">$</span><span class="hl def">hsa00564_AD_male,</span> <span class="hl kwc">paired</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span> <span class="hl kwc">na.rm</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>

<span class="hl kwd">print</span><span class="hl def">(result_CTAD_m)</span>
</pre></div>
<div class="output"><pre class="knitr r">## 
## 	Wilcoxon signed rank exact test
## 
## data:  hsa00564_Zscore_summary$hsa00564_CT_male and hsa00564_Zscore_summary$hsa00564_AD_male
## V = 757, p-value = 0.01727
## alternative hypothesis: true location shift is not equal to 0
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl kwd">ggplot</span><span class="hl def">(data_filtered_CTAD_f,</span> <span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">x</span> <span class="hl def">= Value,</span> <span class="hl kwc">fill</span> <span class="hl def">= Condition))</span> <span class="hl opt">+</span>
  <span class="hl kwd">geom_density</span><span class="hl def">(</span><span class="hl kwc">alpha</span> <span class="hl def">=</span> <span class="hl num">0.5</span><span class="hl def">)</span> <span class="hl opt">+</span>  <span class="hl com"># Set transparency for overlapping densities</span>
  <span class="hl kwd">theme_minimal</span><span class="hl def">()</span> <span class="hl opt">+</span>
  <span class="hl kwd">labs</span><span class="hl def">(</span><span class="hl kwc">title</span> <span class="hl def">=</span> <span class="hl sng">&quot;Density Plot for CT_female and AD_female across Glycerophospholipid metabolism &quot;</span><span class="hl def">,</span>
       <span class="hl kwc">x</span> <span class="hl def">=</span> <span class="hl sng">&quot;Z score&quot;</span><span class="hl def">,</span>
       <span class="hl kwc">y</span> <span class="hl def">=</span> <span class="hl sng">&quot;Density&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl kwd">scale_fill_manual</span><span class="hl def">(</span><span class="hl kwc">values</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;hsa00564_CT_female&quot;</span> <span class="hl def">=</span> <span class="hl sng">&quot;pink&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;hsa00564_AD_female&quot;</span> <span class="hl def">=</span> <span class="hl sng">&quot;red&quot;</span><span class="hl def">))</span> <span class="hl opt">+</span>
  <span class="hl kwd">theme</span><span class="hl def">(</span><span class="hl kwc">legend.title</span> <span class="hl def">=</span> <span class="hl kwd">element_blank</span><span class="hl def">())</span>
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-6-5.png" alt="plot of chunk unnamed-chunk-6" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl def">result_CTAD_f</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">wilcox.test</span><span class="hl def">(hsa00564_Zscore_summary</span><span class="hl opt">$</span><span class="hl def">hsa00564_CT_female, hsa00564_Zscore_summary</span><span class="hl opt">$</span><span class="hl def">hsa00564_AD_female,</span> <span class="hl kwc">paired</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">,</span> <span class="hl kwc">na.rm</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>

<span class="hl kwd">print</span><span class="hl def">(result_CTAD_f)</span>
</pre></div>
<div class="output"><pre class="knitr r">## 
## 	Wilcoxon signed rank test with continuity correction
## 
## data:  hsa00564_Zscore_summary$hsa00564_CT_female and hsa00564_Zscore_summary$hsa00564_AD_female
## V = 1631, p-value = 0.005182
## alternative hypothesis: true location shift is not equal to 0
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl com">########################################################################################################################################################################</span>

<span class="hl com"># Calculate the 1st and 99th percentiles of pathway-specific Z scores for the hsa00564 pathway within the Control individuals</span>
<span class="hl def">hsa00564_CT_female_1st</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">quantile</span><span class="hl def">(hsa00564_Zscore_summary</span><span class="hl opt">$</span><span class="hl def">hsa00564_CT_female,</span> <span class="hl num">0.01</span><span class="hl def">,</span> <span class="hl kwc">na.rm</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>
<span class="hl def">hsa00564_CT_female_99th</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">quantile</span><span class="hl def">(hsa00564_Zscore_summary</span><span class="hl opt">$</span><span class="hl def">hsa00564_CT_female,</span> <span class="hl num">0.99</span><span class="hl def">,</span> <span class="hl kwc">na.rm</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>

<span class="hl def">hsa00564_CT_male_1st</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">quantile</span><span class="hl def">(hsa00564_Zscore_summary</span><span class="hl opt">$</span><span class="hl def">hsa00564_CT_male,</span> <span class="hl num">0.01</span><span class="hl def">,</span> <span class="hl kwc">na.rm</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>
<span class="hl def">hsa00564_CT_male_99th</span> <span class="hl kwb">&lt;-</span> <span class="hl kwd">quantile</span><span class="hl def">(hsa00564_Zscore_summary</span><span class="hl opt">$</span><span class="hl def">hsa00564_CT_male,</span> <span class="hl num">0.99</span><span class="hl def">,</span> <span class="hl kwc">na.rm</span> <span class="hl def">=</span> <span class="hl num">TRUE</span><span class="hl def">)</span>


<span class="hl kwd">ggplot</span><span class="hl def">(hsa00564_Zscore_summary,</span> <span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">x</span> <span class="hl def">= hsa00564_CT_female))</span> <span class="hl opt">+</span>
  <span class="hl kwd">geom_density</span><span class="hl def">(</span><span class="hl kwc">fill</span> <span class="hl def">=</span> <span class="hl sng">&quot;pink&quot;</span><span class="hl def">,</span> <span class="hl kwc">alpha</span> <span class="hl def">=</span> <span class="hl num">0.5</span><span class="hl def">)</span> <span class="hl opt">+</span>  <span class="hl com"># Density plot for Female_CT</span>
  <span class="hl kwd">geom_point</span><span class="hl def">(</span><span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">x</span> <span class="hl def">= hsa00564_AD_female,</span> <span class="hl kwc">y</span> <span class="hl def">=</span> <span class="hl num">0</span><span class="hl def">),</span> <span class="hl kwc">position</span> <span class="hl def">=</span> <span class="hl kwd">position_jitter</span><span class="hl def">(</span><span class="hl kwc">height</span> <span class="hl def">=</span> <span class="hl num">0.02</span><span class="hl def">),</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;red&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>  <span class="hl com"># Dots for Female_AD</span>
  <span class="hl kwd">labs</span><span class="hl def">(</span><span class="hl kwc">title</span> <span class="hl def">=</span> <span class="hl sng">&quot;Density Plot for Female_CT with Female_AD as Dots acrossGlycerophospholipid metabolism &quot;</span><span class="hl def">,</span>
       <span class="hl kwc">x</span> <span class="hl def">=</span> <span class="hl sng">&quot;Z score&quot;</span><span class="hl def">,</span>
       <span class="hl kwc">y</span> <span class="hl def">=</span> <span class="hl sng">&quot;Density&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl kwd">theme_minimal</span><span class="hl def">()</span> <span class="hl opt">+</span>
  <span class="hl kwd">geom_vline</span><span class="hl def">(</span><span class="hl kwc">xintercept</span> <span class="hl def">= hsa00564_CT_female_1st,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;black&quot;</span><span class="hl def">,</span> <span class="hl kwc">linetype</span> <span class="hl def">=</span> <span class="hl sng">&quot;dashed&quot;</span><span class="hl def">,</span> <span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl kwd">geom_vline</span><span class="hl def">(</span><span class="hl kwc">xintercept</span> <span class="hl def">= hsa00564_CT_female_99th,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;black&quot;</span><span class="hl def">,</span> <span class="hl kwc">linetype</span> <span class="hl def">=</span> <span class="hl sng">&quot;dashed&quot;</span><span class="hl def">,</span> <span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl kwd">theme</span><span class="hl def">(</span><span class="hl kwc">text</span> <span class="hl def">=</span> <span class="hl kwd">element_text</span><span class="hl def">(</span><span class="hl kwc">size</span><span class="hl def">=</span><span class="hl num">20</span><span class="hl def">))</span>
</pre></div>
<div class="warning"><pre class="knitr r">## Warning: Removed 91 rows containing non-finite outside the scale range
## (`stat_density()`).
</pre></div>
<div class="warning"><pre class="knitr r">## Warning: Removed 32 rows containing missing values or values outside the scale range
## (`geom_point()`).
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-6-6.png" alt="plot of chunk unnamed-chunk-6" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">###############################################################################################################################</span>

<span class="hl kwd">ggplot</span><span class="hl def">()</span> <span class="hl opt">+</span>
  <span class="hl com"># Density plot for Female_CT</span>
  <span class="hl kwd">geom_density</span><span class="hl def">(</span>
    <span class="hl kwc">data</span> <span class="hl def">= hsa00564_Zscore_summary,</span>
    <span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">x</span> <span class="hl def">= hsa00564_CT_female,</span> <span class="hl kwc">fill</span> <span class="hl def">=</span> <span class="hl sng">&quot;Female_CT&quot;</span><span class="hl def">),</span>
    <span class="hl kwc">alpha</span> <span class="hl def">=</span> <span class="hl num">0.5</span>
  <span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl com"># Density plot for Female_AD</span>
  <span class="hl kwd">geom_density</span><span class="hl def">(</span>
    <span class="hl kwc">data</span> <span class="hl def">= hsa00564_Zscore_summary,</span>
    <span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">x</span> <span class="hl def">= hsa00564_AD_female,</span> <span class="hl kwc">fill</span> <span class="hl def">=</span> <span class="hl sng">&quot;Female_AD&quot;</span><span class="hl def">),</span>
    <span class="hl kwc">alpha</span> <span class="hl def">=</span> <span class="hl num">0.5</span>
  <span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl com"># Mimic rug plot for Female_CT</span>
  <span class="hl kwd">geom_point</span><span class="hl def">(</span>
    <span class="hl kwc">data</span> <span class="hl def">= hsa00564_Zscore_summary,</span>
    <span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">x</span> <span class="hl def">= hsa00564_CT_female,</span> <span class="hl kwc">y</span> <span class="hl def">=</span> <span class="hl opt">-</span><span class="hl num">0.02</span><span class="hl def">,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;Female_CT&quot;</span><span class="hl def">),</span>
    <span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">10</span><span class="hl def">,</span> <span class="hl kwc">shape</span> <span class="hl def">=</span> <span class="hl sng">&quot;|&quot;</span>
  <span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl com"># Mimic rug plot for Female_AsymAD</span>
  <span class="hl kwd">geom_point</span><span class="hl def">(</span>
    <span class="hl kwc">data</span> <span class="hl def">= hsa00564_Zscore_summary,</span>
    <span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">x</span> <span class="hl def">= hsa00564_AsymAD_female,</span> <span class="hl kwc">y</span> <span class="hl def">=</span> <span class="hl opt">-</span><span class="hl num">0.05</span><span class="hl def">,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;Female_AsymAD&quot;</span><span class="hl def">),</span>
    <span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">10</span><span class="hl def">,</span> <span class="hl kwc">shape</span> <span class="hl def">=</span> <span class="hl sng">&quot;|&quot;</span>
  <span class="hl def">)</span> <span class="hl opt">+</span>
    <span class="hl com"># Mimic rug plot for Female_AD</span>
  <span class="hl kwd">geom_point</span><span class="hl def">(</span>
    <span class="hl kwc">data</span> <span class="hl def">= hsa00564_Zscore_summary,</span>
    <span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">x</span> <span class="hl def">= hsa00564_AD_female,</span> <span class="hl kwc">y</span> <span class="hl def">=</span> <span class="hl opt">-</span><span class="hl num">0.08</span><span class="hl def">,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;Female_AD&quot;</span><span class="hl def">),</span>
    <span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">10</span><span class="hl def">,</span> <span class="hl kwc">shape</span> <span class="hl def">=</span> <span class="hl sng">&quot;|&quot;</span>
  <span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl com"># Labels and themes</span>
  <span class="hl kwd">labs</span><span class="hl def">(</span>
    <span class="hl kwc">title</span> <span class="hl def">=</span> <span class="hl sng">&quot;Distributions of Pathway-specific Z scores across Glycerophospholipid metabolism &quot;</span><span class="hl def">,</span>
    <span class="hl kwc">x</span> <span class="hl def">=</span> <span class="hl sng">&quot;Z score&quot;</span><span class="hl def">,</span>
    <span class="hl kwc">y</span> <span class="hl def">=</span> <span class="hl sng">&quot;Density&quot;</span><span class="hl def">,</span>
    <span class="hl kwc">fill</span> <span class="hl def">=</span> <span class="hl sng">&quot;Z Score Value&quot;</span><span class="hl def">,</span>
    <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;Sample Category&quot;</span>
  <span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl kwd">theme_minimal</span><span class="hl def">(</span><span class="hl kwc">base_size</span> <span class="hl def">=</span> <span class="hl num">14</span><span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl kwd">theme</span><span class="hl def">(</span>
    <span class="hl kwc">text</span> <span class="hl def">=</span> <span class="hl kwd">element_text</span><span class="hl def">(</span><span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">20</span><span class="hl def">),</span>
    <span class="hl kwc">legend.position</span> <span class="hl def">=</span> <span class="hl sng">&quot;right&quot;</span>
  <span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl com"># Vertical dashed lines for Female_CT</span>
  <span class="hl kwd">geom_vline</span><span class="hl def">(</span><span class="hl kwc">xintercept</span> <span class="hl def">= hsa00564_CT_female_1st,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;black&quot;</span><span class="hl def">,</span> <span class="hl kwc">linetype</span> <span class="hl def">=</span> <span class="hl sng">&quot;dashed&quot;</span><span class="hl def">,</span> <span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl kwd">geom_vline</span><span class="hl def">(</span><span class="hl kwc">xintercept</span> <span class="hl def">= hsa00564_CT_female_99th,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;black&quot;</span><span class="hl def">,</span> <span class="hl kwc">linetype</span> <span class="hl def">=</span> <span class="hl sng">&quot;dashed&quot;</span><span class="hl def">,</span> <span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl com"># Manually specify the color scales for the legends</span>
  <span class="hl kwd">scale_fill_manual</span><span class="hl def">(</span>
    <span class="hl kwc">values</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;Female_CT&quot;</span> <span class="hl def">=</span> <span class="hl sng">&quot;pink&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;Female_AD&quot;</span> <span class="hl def">=</span> <span class="hl sng">&quot;tomato&quot;</span><span class="hl def">),</span>
    <span class="hl kwc">labels</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;Female_AD&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;Female_CT&quot;</span><span class="hl def">)</span>
  <span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl kwd">scale_color_manual</span><span class="hl def">(</span>
    <span class="hl kwc">values</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;Female_CT&quot;</span> <span class="hl def">=</span> <span class="hl sng">&quot;pink&quot;</span><span class="hl def">,</span><span class="hl sng">&quot;Female_AsymAD&quot;</span> <span class="hl def">=</span> <span class="hl sng">&quot;coral&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;Female_AD&quot;</span> <span class="hl def">=</span> <span class="hl sng">&quot;tomato&quot;</span><span class="hl def">),</span>
    <span class="hl kwc">labels</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;Female_CT&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;Female_AsymAD&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;Female_AD&quot;</span><span class="hl def">)</span>
  <span class="hl def">)</span> <span class="hl com">#+</span>
</pre></div>
<div class="warning"><pre class="knitr r">## Warning: Removed 91 rows containing non-finite outside the scale range
## (`stat_density()`).
</pre></div>
<div class="warning"><pre class="knitr r">## Warning: Removed 32 rows containing non-finite outside the scale range
## (`stat_density()`).
</pre></div>
<div class="warning"><pre class="knitr r">## Warning: Removed 91 rows containing missing values or values outside the scale range
## (`geom_point()`).
</pre></div>
<div class="warning"><pre class="knitr r">## Warning: Removed 32 rows containing missing values or values outside the scale range
## (`geom_point()`).
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-6-7.png" alt="plot of chunk unnamed-chunk-6" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r">     <span class="hl com"># Break the X axis manually to make the plot more visible </span>
 <span class="hl com">#scale_x_break(c(7.5, 17.5), scales = 0.3, ticklabels = c(7.5, 17.5))</span>


<span class="hl com">###############################################################################################################################</span>

<span class="hl kwd">ggplot</span><span class="hl def">(hsa00564_Zscore_summary,</span> <span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">x</span> <span class="hl def">= hsa00564_CT_male))</span> <span class="hl opt">+</span>
  <span class="hl kwd">geom_density</span><span class="hl def">(</span><span class="hl kwc">fill</span> <span class="hl def">=</span> <span class="hl sng">&quot;lightblue&quot;</span><span class="hl def">,</span> <span class="hl kwc">alpha</span> <span class="hl def">=</span> <span class="hl num">0.5</span><span class="hl def">)</span> <span class="hl opt">+</span>  <span class="hl com"># Density plot for Female_CT</span>
  <span class="hl kwd">geom_point</span><span class="hl def">(</span><span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">x</span> <span class="hl def">= hsa00564_AD_male,</span> <span class="hl kwc">y</span> <span class="hl def">=</span> <span class="hl num">0</span><span class="hl def">),</span> <span class="hl kwc">position</span> <span class="hl def">=</span> <span class="hl kwd">position_jitter</span><span class="hl def">(</span><span class="hl kwc">height</span> <span class="hl def">=</span> <span class="hl num">0.02</span><span class="hl def">),</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;blue&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>  <span class="hl com"># Dots for Male_AD</span>
  <span class="hl kwd">labs</span><span class="hl def">(</span><span class="hl kwc">title</span> <span class="hl def">=</span> <span class="hl sng">&quot;Density Plot for Male_CT with Male_AD as Dots across Glycerophospholipid metabolism &quot;</span><span class="hl def">,</span>
       <span class="hl kwc">x</span> <span class="hl def">=</span> <span class="hl sng">&quot;Z score&quot;</span><span class="hl def">,</span>
       <span class="hl kwc">y</span> <span class="hl def">=</span> <span class="hl sng">&quot;Density&quot;</span><span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl kwd">theme_minimal</span><span class="hl def">()</span> <span class="hl opt">+</span>
  <span class="hl kwd">geom_vline</span><span class="hl def">(</span><span class="hl kwc">xintercept</span> <span class="hl def">= hsa00564_CT_male_1st,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;black&quot;</span><span class="hl def">,</span> <span class="hl kwc">linetype</span> <span class="hl def">=</span> <span class="hl sng">&quot;dashed&quot;</span><span class="hl def">,</span> <span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl kwd">geom_vline</span><span class="hl def">(</span><span class="hl kwc">xintercept</span> <span class="hl def">= hsa00564_CT_male_99th,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;black&quot;</span><span class="hl def">,</span> <span class="hl kwc">linetype</span> <span class="hl def">=</span> <span class="hl sng">&quot;dashed&quot;</span><span class="hl def">,</span> <span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl kwd">theme</span><span class="hl def">(</span><span class="hl kwc">text</span> <span class="hl def">=</span> <span class="hl kwd">element_text</span><span class="hl def">(</span><span class="hl kwc">size</span><span class="hl def">=</span><span class="hl num">20</span><span class="hl def">))</span>
</pre></div>
<div class="warning"><pre class="knitr r">## Warning: Removed 113 rows containing non-finite outside the scale range
## (`stat_density()`).
</pre></div>
<div class="warning"><pre class="knitr r">## Warning: Removed 112 rows containing missing values or values outside the scale range
## (`geom_point()`).
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-6-8.png" alt="plot of chunk unnamed-chunk-6" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">###############################################################################################################################</span>

<span class="hl kwd">ggplot</span><span class="hl def">()</span> <span class="hl opt">+</span>
  <span class="hl com"># Density plot for Male_CT</span>
  <span class="hl kwd">geom_density</span><span class="hl def">(</span>
    <span class="hl kwc">data</span> <span class="hl def">= hsa00564_Zscore_summary,</span>
    <span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">x</span> <span class="hl def">= hsa00564_CT_male,</span> <span class="hl kwc">fill</span> <span class="hl def">=</span> <span class="hl sng">&quot;Male_CT&quot;</span><span class="hl def">),</span>
    <span class="hl kwc">alpha</span> <span class="hl def">=</span> <span class="hl num">0.5</span>
  <span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl com"># Density plot for Male_AD</span>
  <span class="hl kwd">geom_density</span><span class="hl def">(</span>
    <span class="hl kwc">data</span> <span class="hl def">= hsa00564_Zscore_summary,</span>
    <span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">x</span> <span class="hl def">= hsa00564_AD_male,</span> <span class="hl kwc">fill</span> <span class="hl def">=</span> <span class="hl sng">&quot;Male_AD&quot;</span><span class="hl def">),</span>
    <span class="hl kwc">alpha</span> <span class="hl def">=</span> <span class="hl num">0.5</span>
  <span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl com"># Mimic rug plot for Male_CT</span>
  <span class="hl kwd">geom_point</span><span class="hl def">(</span>
    <span class="hl kwc">data</span> <span class="hl def">= hsa00564_Zscore_summary,</span>
    <span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">x</span> <span class="hl def">= hsa00564_CT_male,</span> <span class="hl kwc">y</span> <span class="hl def">=</span> <span class="hl opt">-</span><span class="hl num">0.02</span><span class="hl def">,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;Male_CT&quot;</span><span class="hl def">),</span>
    <span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">10</span><span class="hl def">,</span> <span class="hl kwc">shape</span> <span class="hl def">=</span> <span class="hl sng">&quot;|&quot;</span>
  <span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl com"># Mimic rug plot for Male_AsymAD</span>
  <span class="hl kwd">geom_point</span><span class="hl def">(</span>
    <span class="hl kwc">data</span> <span class="hl def">= hsa00564_Zscore_summary,</span>
    <span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">x</span> <span class="hl def">= hsa00564_AsymAD_male,</span> <span class="hl kwc">y</span> <span class="hl def">=</span> <span class="hl opt">-</span><span class="hl num">0.05</span><span class="hl def">,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;Male_AsymAD&quot;</span><span class="hl def">),</span>
    <span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">10</span><span class="hl def">,</span> <span class="hl kwc">shape</span> <span class="hl def">=</span> <span class="hl sng">&quot;|&quot;</span>
  <span class="hl def">)</span> <span class="hl opt">+</span>
   <span class="hl com"># Mimic rug plot for Male_AD</span>
  <span class="hl kwd">geom_point</span><span class="hl def">(</span>
    <span class="hl kwc">data</span> <span class="hl def">= hsa00564_Zscore_summary,</span>
    <span class="hl kwd">aes</span><span class="hl def">(</span><span class="hl kwc">x</span> <span class="hl def">= hsa00564_AD_male,</span> <span class="hl kwc">y</span> <span class="hl def">=</span> <span class="hl opt">-</span><span class="hl num">0.08</span><span class="hl def">,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;Male_AD&quot;</span><span class="hl def">),</span>
    <span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">10</span><span class="hl def">,</span> <span class="hl kwc">shape</span> <span class="hl def">=</span> <span class="hl sng">&quot;|&quot;</span>
  <span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl com"># Labels and themes</span>
  <span class="hl kwd">labs</span><span class="hl def">(</span>
    <span class="hl kwc">title</span> <span class="hl def">=</span> <span class="hl sng">&quot;Distributions of Pathway-specific Z scores across Glycerophospholipid metabolism &quot;</span><span class="hl def">,</span>
    <span class="hl kwc">x</span> <span class="hl def">=</span> <span class="hl sng">&quot;Z score&quot;</span><span class="hl def">,</span>
    <span class="hl kwc">y</span> <span class="hl def">=</span> <span class="hl sng">&quot;Density&quot;</span><span class="hl def">,</span>
    <span class="hl kwc">fill</span> <span class="hl def">=</span> <span class="hl sng">&quot;Z Score Value&quot;</span><span class="hl def">,</span>
    <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;Sample Category&quot;</span>
  <span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl kwd">theme_minimal</span><span class="hl def">(</span><span class="hl kwc">base_size</span> <span class="hl def">=</span> <span class="hl num">14</span><span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl kwd">theme</span><span class="hl def">(</span>
    <span class="hl kwc">text</span> <span class="hl def">=</span> <span class="hl kwd">element_text</span><span class="hl def">(</span><span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">20</span><span class="hl def">),</span>
    <span class="hl kwc">legend.position</span> <span class="hl def">=</span> <span class="hl sng">&quot;right&quot;</span>
  <span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl com"># Vertical dashed lines for Male_CT</span>
  <span class="hl kwd">geom_vline</span><span class="hl def">(</span><span class="hl kwc">xintercept</span> <span class="hl def">= hsa00564_CT_male_1st,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;black&quot;</span><span class="hl def">,</span> <span class="hl kwc">linetype</span> <span class="hl def">=</span> <span class="hl sng">&quot;dashed&quot;</span><span class="hl def">,</span> <span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl kwd">geom_vline</span><span class="hl def">(</span><span class="hl kwc">xintercept</span> <span class="hl def">= hsa00564_CT_male_99th,</span> <span class="hl kwc">color</span> <span class="hl def">=</span> <span class="hl sng">&quot;black&quot;</span><span class="hl def">,</span> <span class="hl kwc">linetype</span> <span class="hl def">=</span> <span class="hl sng">&quot;dashed&quot;</span><span class="hl def">,</span> <span class="hl kwc">size</span> <span class="hl def">=</span> <span class="hl num">1</span><span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl com"># Manually specify the color scales for the legends</span>
  <span class="hl kwd">scale_fill_manual</span><span class="hl def">(</span>
    <span class="hl kwc">values</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;Male_CT&quot;</span> <span class="hl def">=</span> <span class="hl sng">&quot;lightblue&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;Male_AD&quot;</span> <span class="hl def">=</span> <span class="hl sng">&quot;blue&quot;</span><span class="hl def">),</span>
    <span class="hl kwc">labels</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;Male_AD&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;Male_CT&quot;</span><span class="hl def">)</span>
  <span class="hl def">)</span> <span class="hl opt">+</span>
  <span class="hl kwd">scale_color_manual</span><span class="hl def">(</span>
    <span class="hl kwc">values</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;Male_CT&quot;</span> <span class="hl def">=</span> <span class="hl sng">&quot;lightblue&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;Male_AsymAD&quot;</span> <span class="hl def">=</span> <span class="hl sng">&quot;cornflowerblue&quot;</span><span class="hl def">,</span>  <span class="hl sng">&quot;Male_AD&quot;</span> <span class="hl def">=</span> <span class="hl sng">&quot;blue&quot;</span><span class="hl def">),</span>
    <span class="hl kwc">labels</span> <span class="hl def">=</span> <span class="hl kwd">c</span><span class="hl def">(</span><span class="hl sng">&quot;Male_CT&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;Male_AsymAD&quot;</span><span class="hl def">,</span> <span class="hl sng">&quot;Male_AD&quot;</span><span class="hl def">)</span>
  <span class="hl def">)</span> <span class="hl com">#+</span>
</pre></div>
<div class="warning"><pre class="knitr r">## Warning: Removed 113 rows containing non-finite outside the scale range
## (`stat_density()`).
</pre></div>
<div class="warning"><pre class="knitr r">## Warning: Removed 112 rows containing non-finite outside the scale range
## (`stat_density()`).
</pre></div>
<div class="warning"><pre class="knitr r">## Warning: Removed 113 rows containing missing values or values outside the scale range
## (`geom_point()`).
</pre></div>
<div class="warning"><pre class="knitr r">## Warning: Removed 98 rows containing missing values or values outside the scale range
## (`geom_point()`).
</pre></div>
<div class="warning"><pre class="knitr r">## Warning: Removed 112 rows containing missing values or values outside the scale range
## (`geom_point()`).
</pre></div>
</div><div class="rimage default"><img src="figure/unnamed-chunk-6-9.png" alt="plot of chunk unnamed-chunk-6" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r">   <span class="hl com"># Break the X axis manually to make the plot more visible </span>
 <span class="hl com">#scale_x_break(c(10, 20), scales = 0.3, ticklabels = c(10, 20))</span>
</pre></div>
</div></div>



</body>
</html>
